---
title: "16S ANCOMBC2 workup"
author: "Conner Mertz"
date: "2023-07-02"
output: html_document
---

Script for differential abundance analyses identifying ASVs (or bacterial families) associated with different diets, protein treatments, and gut sections using ANCOM-BC2.

ANCOM
Analysis of Compositions of Microbiomes with Bias Correction (ANCOM-BC) (Lin and Peddada 2020) is a methodology of differential abundance (DA) analysis for microbial absolute abundances. ANCOM-BC estimates the unknown sampling fractions, corrects the bias induced by their differences through a log linear regression model including the estimated sampling fraction as an offset terms, and identifies taxa that are differentially abundant according to the variable of interest. For more details, please refer to the ANCOM-BC paper.
official tutorial: https://bioconductor.org/packages/release/bioc/vignettes/ANCOMBC/inst/doc/ANCOMBC2.html

https://bioconductor.org/packages/devel/bioc/vignettes/ANCOMBC/inst/doc/ANCOMBC.html

a helpful tutorial: https://www.yanh.org/2021/01/01/microbiome-r/#differential-abundance-analysis

ANCOM manual: https://bioconductor.org/packages/release/bioc/manuals/ANCOMBC/man/ANCOMBC.pdf

```{r setup, include=FALSE}
knitr::opts_chunk$set(message = FALSE, warning = FALSE, comment = NA, 
                      fig.width = 6.25, fig.height = 5)
library(ANCOMBC)
library(tidyverse)
library(caret)
library(DT)
options(DT.options = list(
  initComplete = JS("function(settings, json) {",
  "$(this.api().table().header()).css({'background-color': 
  '#000', 'color': '#fff'});","}")))
```

#LOAD OTHER PACKAGES THAT MIGHT BE USEFUL
```{r}
#load packages
# = this package is definitely needed
library(BiocManager)
library(phyloseq) #
library(ggplot2) #
library(randomForest)
library(vegan)
library(caret)
library(gridExtra) #
library(ggpubr) 
library(ggthemes) #
library(vegan) #
library(MASS)
library(cluster)
library(RColorBrewer) #
library(grid)
library(cooccur)
library(igraph)
library(Hmisc)
#library(minerva)
library(parallel)
library(corrplot)
library(tidyverse) #
library(plyr);library(dplyr) #
library(tidyr) #
library(usdm) #
library(decontam) #
library(ecodist) #
library(betapart) #
#library(psych)
#library(ggbiplot)
library(ANCOMBC) #
library(DT) #
library(gplots) #
library(remotes)
library(metagMisc)
library(iNEXT)
library("DESeq2")
packageVersion("DESeq2")
library("reshape2")
library("remotes")
library("phyloseq.extended")
#library(ggvegan) 
#library(ggpubfigs)
#library(agricolae) #this package is not compatable with this version of R, if we really need it we will need to install new version of package
library(RColorBrewer)
library(scales)
library("devtools")

theme_set(theme_bw()) #sets ggplot theme
```


#load metadata
```{r}
Meta <- read.csv("/Users/connermertz/Dropbox (vlab)/Conner Mertz/UNM biology/Chapter 1 E1 vs E2/Methods/16S rRNA gene sequencing/Sequencing_workup/E1_E2_Meta_ALL.csv") #you should have 333 total samples
row.names(Meta) <- Meta$Sample_ID
```

#read in ps objects
```{r}
ps_raw <- readRDS("/Users/connermertz/Dropbox (vlab)/Conner Mertz/UNM biology/Chapter 1 E1 vs E2/Methods/16S rRNA gene sequencing/Sequencing_workup/DADA2_output/clean_ps.rds") #raw but with choloro,mito, low reads (less than 2000), and sample duplicates removed
ps_all <- readRDS("/Users/connermertz/Dropbox (vlab)/Conner Mertz/UNM biology/Chapter 1 E1 vs E2/Methods/16S rRNA gene sequencing/Sequencing_workup/RarefyOutput/ps_rare.rds") #this is my phyloseq object I will be using for the majority of my analysis. It contains all samples from E1 and E2 (duplicates not included) and has been rarefied at lib size of 8000.
sample_data(ps_raw) <- Meta #If you beed to update your meta sheet, this is how to do it without dunning dada2 again to create a whole new ps
sample_data(ps_all) <- Meta #If you beed to update your meta sheet, this is how to do it without dunning dada2 again to create a whole new ps
```


```{r}
#read in un-normalized- unrarefied phyloseq object from rds (ANCOM normalizes for you, and if you give it normalized data it will give bias)
#make relevant subsets

### Make new phyloseq object of just E1, no duplicates
E1_ps_raw <- subset_samples(ps_raw, Experiment=="E1")


#Subset by E1 2.5P,5P, 10P only to compare to E2; no duplicates
ps_raw_E1_forcomparison <- subset_samples(E1_ps_raw, Percent_Protein %in% c("2.5P", "5P", "10P"))
ps_raw_E1_forcomparison <- subset_samples(ps_raw_E1_forcomparison, Location %in% c("Duodenum", "Ileum", "Cecum"))
ps_raw_E1_forcomparison

#Subset by E1 and E2 2.5P,5P, 10P; no duplicates
ps_raw_E1_E2_forcomparison <- subset_samples(ps_raw, Percent_Protein %in% c("2.5P", "5P", "10P"))
ps_raw_E1_E2_forcomparison <- subset_samples(ps_raw_E1_E2_forcomparison, Location %in% c("Duodenum", "Ileum", "Cecum"))
ps_raw_E1_E2_forcomparison


### Make new phyloseq object of just E2, no duplicates
E2_ps_raw <- subset_samples(ps_raw, Experiment=="E2")   


#Subset by E1_E2_Location
ps_raw_E1_E2_duod <- subset_samples(ps_raw_E1_E2_forcomparison, Location %in% c("Duodenum"))
ps_raw_E1_E2_ile <- subset_samples(ps_raw_E1_E2_forcomparison, Location %in% c("Ileum"))
ps_raw_E1_E2_cec <- subset_samples(ps_raw_E1_E2_forcomparison, Location %in% c("Cecum"))
ps_raw_E1_E2_cec
###
```
#colors
```{r}
#to make Phylum always the same color:

phyColor <- c(Acidobacteriota = "#D95F02",
  Actinobacteriota = "#1B9E77", 
  Bacteroidota = "#A6CEE3", 
  Campylobacterota = "#E7298A",
  Deferribacterota= "orchid1", 
  Desulfobacterota= "darkturquoise",
  Firmicutes = "darkviolet",
  Fusobacteriota = "gold2",
  Gemmatimonadota = "#66A61E", 
  Nitrospirota = "maroon", 
  Planctomycetota = "#1F78B4" ,
  Proteobacteria = "#FB9A99",
  Verrucomicrobiota = "#B2DF8A")
```


#ANCOMBC2 by Location- E1 vs E2 for all locations samples
```{r}
set.seed(123)
output = ancombc2(ps_raw_E1_E2_forcomparison, assay_name = NULL, tax_level = "family",
                  fix_formula = "Percent_Protein + Exp_Location", rand_formula = NULL,
                  p_adj_method = "holm", pseudo = 0, pseudo_sens = TRUE,
                  prv_cut = 0.10, lib_cut = 1000, s0_perc = 0.05,
                  group = "Exp_Location", struc_zero = TRUE, neg_lb = TRUE,
                  alpha = 0.05, n_cl = 2, verbose = TRUE,
                  global = TRUE, pairwise = TRUE, dunnet = TRUE, trend = TRUE,
                  iter_control = list(tol = 1e-2, max_iter = 20, 
                                      verbose = TRUE),
                  em_control = list(tol = 1e-5, max_iter = 100),
                  lme_control = lme4::lmerControl(),
                  mdfdr_control = list(fwer_ctrl_method = "holm", B = 100),
                  trend_control = list(contrast = list(matrix(c(1, 0, 0, 0, 0,
                                                               -1, 1, 0, 0, 0,
                                                                0, -1, 1, 0, 0, 
                                                                0, 0, -1, 1, 0,
                                                                0, 0, 0, -1, 1),
                                                              nrow = 5, 
                                                              byrow = TRUE),
                                                       matrix(c(1, 0, 0, 0, 0,
                                                               -1, 1, 0, 0, 0,
                                                                0, -1, 1, 0, 0, 
                                                                0, 0, -1, 1, 0,
                                                                0, 0, 0, -1, 1),
                                                              nrow = 5, 
                                                              byrow = TRUE)),
                                       node = list(2, 2),
                                       solver = "ECOS",
                                       B = 100))
```
Save the output
```{r}
res_prim = output$res
saveRDS(res_prim, "/Users/connermertz/Dropbox (vlab)/Conner Mertz/UNM biology/Chapter 1 E1 vs E2/Methods/16S rRNA gene sequencing/Sequencing_workup/ANCOMoutput/codeexports/E1_E2_forcomparison/E1vsE2_LOCATION_res_prim.rds")
#read in the ANCOMBC output if you want to see E1 vs E2 by Location
E1vsE2_LOCATION_res_prim <- readRDS("/Users/connermertz/Dropbox (vlab)/Conner Mertz/UNM biology/Chapter 1 E1 vs E2/Methods/16S rRNA gene sequencing/Sequencing_workup/ANCOMoutput/codeexports/E1_E2_forcomparison/E1vsE2_LOCATION_res_prim.rds")

res_pair = output$res_pair
saveRDS(res_pair, "/Users/connermertz/Dropbox (vlab)/Conner Mertz/UNM biology/Chapter 1 E1 vs E2/Methods/16S rRNA gene sequencing/Sequencing_workup/ANCOMoutput/codeexports/E1_E2_forcomparison/E1vsE2_LOCATION_res_pair.rds")
E1vsE2_LOCATION_res_pair <- readRDS("/Users/connermertz/Dropbox (vlab)/Conner Mertz/UNM biology/Chapter 1 E1 vs E2/Methods/16S rRNA gene sequencing/Sequencing_workup/ANCOMoutput/codeexports/E1_E2_forcomparison/E1vsE2_LOCATION_res_pair.rds")

res_trend = output$res_trend
saveRDS(res_trend, "/Users/connermertz/Dropbox (vlab)/Conner Mertz/UNM biology/Chapter 1 E1 vs E2/Methods/16S rRNA gene sequencing/Sequencing_workup/ANCOMoutput/codeexports/E1_E2_forcomparison/E1vsE2_LOCATION_res_trend.rds")
E1vsE2_LOCATION_res_trend <- readRDS("/Users/connermertz/Dropbox (vlab)/Conner Mertz/UNM biology/Chapter 1 E1 vs E2/Methods/16S rRNA gene sequencing/Sequencing_workup/ANCOMoutput/codeexports/E1_E2_forcomparison/E1vsE2_LOCATION_res_trend.rds")
```

ANCOM-BC2 primary analysis
```{r}
E1vsE2_LOCATION_res_prim
```

##ANCOM-BC2 pairwise directional test
ANCOM-BC2 pairwise directional test aims to determine taxa that are differentially abundant between any pair of two groups across three or more experimental groups, while controlling the mdFDR.

In this example, we want to identify taxa that are differentially abundant between any pair of two groups across “lean”, “overweight”, and “obese”. The result contains: 1) log fold changes, 2) standard errors, 3) test statistics, 4) p-values, 5) adjusted p-values, 6) indicators of whether the taxon is differentially abundant (TRUE) or not (FALSE).
```{r}
E1vsE2_LOCATION_res_pair
write.csv(E1vsE2_LOCATION_res_pair, "/Users/connermertz/Dropbox (vlab)/Conner Mertz/UNM biology/Chapter 1 E1 vs E2/Methods/16S rRNA gene sequencing/Sequencing_workup/ANCOMoutput/codeexports/E1_E2_forcomparison/E1vsE2_LOCATION_res_pair_edit.csv")

#read in your res_pair csv: 
E1vsE2_LOCATION_res_pair2 <- read.csv("/Users/connermertz/Dropbox (vlab)/Conner Mertz/UNM biology/Chapter 1 E1 vs E2/Methods/16S rRNA gene sequencing/Sequencing_workup/ANCOMoutput/codeexports/E1_E2_forcomparison/E1vsE2_LOCATION_res_pair_edit.csv")
#change name so target_ids match on both files
names(E1vsE2_LOCATION_res_pair2)[2] ="taxon" 

#read in your taxonomy classification file 
Taxonomy_classification <- read.csv("/Users/connermertz/Dropbox (vlab)/Conner Mertz/UNM biology/Chapter 1 E1 vs E2/Methods/16S rRNA gene sequencing/Sequencing_workup/DADA2_output/ALL_taxa.csv") #if this doesn't load be sure to download the file from the cloud, then it will work
#change name so target_ids match on both files
names(Taxonomy_classification)[1] ="taxon"

#now we need to get our ANCOMBC results and join it with our Taxonomy results from DADA2 (combining two data frames)
#we only want to keep the ASVs that showed up with ANCOMBC, we don't need the other ASVs that showed up
Part_1 <- Taxonomy_classification %>% 
  inner_join (E1vsE2_LOCATION_res_pair2, by = "taxon")

write.csv(Part_1,"/Users/connermertz/Dropbox (vlab)/Conner Mertz/UNM biology/Chapter 1 E1 vs E2/Methods/16S rRNA gene sequencing/Sequencing_workup/ANCOMoutput/codeexports/E1_E2_forcomparison/E1vsE2_LOCATION_res_pair_AND_Taxonomy.csv") # went in here and edited file to my liking
```


#Plot ANCOM-BC2 pairwise directional test

```{r}
#this makes a heatmap
df_fig_pair = E1vsE2_LOCATION_res_pair %>%
    filter(diff_Exp_LocationE1_DUOD == 1 | 
             diff_Exp_LocationE1_IL == 1 |
             diff_Exp_LocationE2_CEC == 1 |
             diff_Exp_LocationE2_DUOD == 1 |
             diff_Exp_LocationE2_IL == 1 |
             diff_Exp_LocationE1_IL_Exp_LocationE1_DUOD == 1 |
             diff_Exp_LocationE2_CEC_Exp_LocationE1_DUOD == 1 |
             diff_Exp_LocationE2_DUOD_Exp_LocationE1_DUOD == 1 |
             diff_Exp_LocationE2_IL_Exp_LocationE1_DUOD == 1 |
             diff_Exp_LocationE2_CEC_Exp_LocationE1_IL == 1 |
             diff_Exp_LocationE2_DUOD_Exp_LocationE1_IL == 1 |
             diff_Exp_LocationE2_IL_Exp_LocationE1_IL == 1 |
             diff_Exp_LocationE2_DUOD_Exp_LocationE2_CEC == 1 |
             diff_Exp_LocationE2_IL_Exp_LocationE2_CEC == 1 |
             diff_Exp_LocationE2_IL_Exp_LocationE2_DUOD == 1) %>%
    mutate(lfc_E1_DUOD = ifelse(diff_Exp_LocationE1_DUOD == 1, 
                             lfc_Exp_LocationE1_DUOD, 0),
           lfc_E1_IL = ifelse(diff_Exp_LocationE1_IL == 1, 
                                   lfc_Exp_LocationE1_IL, 0),
           lfc_E2_CEC = ifelse(diff_Exp_LocationE2_CEC == 1, 
                                   lfc_Exp_LocationE2_CEC, 0),
           lfc_E2_DUOD = ifelse(diff_Exp_LocationE2_DUOD == 1, 
                                   lfc_Exp_LocationE2_DUOD, 0),
           lfc_E2_IL = ifelse(diff_Exp_LocationE2_IL == 1, 
                                   lfc_Exp_LocationE2_IL, 0),
           lfc_E1_IL_E1_DUOD = ifelse(diff_Exp_LocationE1_IL_Exp_LocationE1_DUOD == 1, 
                                   lfc_Exp_LocationE1_IL_Exp_LocationE1_DUOD, 0),
           lfc_E2_CEC_E1_DUOD = ifelse(diff_Exp_LocationE2_CEC_Exp_LocationE1_DUOD == 1, 
                                   lfc_Exp_LocationE2_CEC_Exp_LocationE1_DUOD, 0),
           lfc_E2_DUOD_E1_DUOD = ifelse(diff_Exp_LocationE2_DUOD_Exp_LocationE1_DUOD == 1, 
                                   lfc_Exp_LocationE2_DUOD_Exp_LocationE1_DUOD, 0),
           lfc_E2_IL_E1_DUOD = ifelse(diff_Exp_LocationE2_IL_Exp_LocationE1_DUOD == 1, 
                                   lfc_Exp_LocationE2_IL_Exp_LocationE1_DUOD, 0),
           lfc_E2_CEC_E1_IL = ifelse(diff_Exp_LocationE2_CEC_Exp_LocationE1_IL == 1, 
                                   lfc_Exp_LocationE2_CEC_Exp_LocationE1_IL, 0),
           lfc_E2_DUOD_E1_IL = ifelse(diff_Exp_LocationE2_DUOD_Exp_LocationE1_IL == 1, 
                                   lfc_Exp_LocationE2_DUOD_Exp_LocationE1_IL, 0),
           lfc_E2_IL_E1_IL = ifelse(diff_Exp_LocationE2_IL_Exp_LocationE1_IL == 1, 
                                   lfc_Exp_LocationE2_IL_Exp_LocationE1_IL, 0),
           lfc_E2_DUOD_E2_CEC = ifelse(diff_Exp_LocationE2_DUOD_Exp_LocationE2_CEC == 1, 
                                   lfc_Exp_LocationE2_DUOD_Exp_LocationE2_CEC, 0),
           lfc_E2_IL_E2_CEC = ifelse(diff_Exp_LocationE2_IL_Exp_LocationE2_CEC == 1, 
                                   lfc_Exp_LocationE2_IL_Exp_LocationE2_CEC, 0),
           lfc_E2_IL_E2_DUOD = ifelse(diff_Exp_LocationE2_IL_Exp_LocationE2_DUOD == 1, 
                                        lfc_Exp_LocationE2_IL_Exp_LocationE2_DUOD, 0)) %>%
    transmute(taxon, 
              `E1_DUOD vs. E1_CEC` = round(lfc_E1_DUOD, 2), 
              `E1_IL vs. E1_CEC` = round(lfc_E1_IL, 2),
              `E2_CEC vs. E1_CEC` = round(lfc_E2_CEC, 2),
              `E2_DUOD vs. E1_CEC` = round(lfc_E2_DUOD, 2),
              `E2_IL vs. E1_CEC` = round(lfc_E2_IL, 2),
              `E1_IL vs. E1_DUOD` = round(lfc_E1_IL_E1_DUOD, 2),
              `E2_CEC vs. E1_DUOD` = round(lfc_E2_CEC_E1_DUOD, 2),
              `E2_DUOD vs. E1_DUOD` = round(lfc_E2_DUOD_E1_DUOD, 2),
              `E2_IL vs. E1_DUOD` = round(lfc_E2_IL_E1_DUOD, 2),
              `E2_CEC vs. E1_IL` = round(lfc_E2_CEC_E1_IL, 2),
              `E2_DUOD vs. E1_IL` = round(lfc_E2_DUOD_E1_IL, 2),
              `E2_IL vs. E1_IL` = round(lfc_E2_IL_E1_IL, 2),
              `E2_DUOD vs. E2_CEC` = round(lfc_E2_DUOD_E2_CEC, 2),
              `E2_IL vs. E2_CEC` = round(lfc_E2_IL_E2_CEC, 2),
              `E2_IL vs. E2_DUOD` = round(lfc_E2_IL_E2_DUOD, 2)
              ) %>%
    pivot_longer(cols = `E1_DUOD vs. E1_CEC`:`E2_IL vs. E2_DUOD`, 
                 names_to = "group", values_to = "value") %>%
    arrange(taxon)
df_fig_pair$group = factor(df_fig_pair$group, 
                           levels = c("E1_DUOD vs. E1_CEC",
                                      "E1_IL vs. E1_CEC",
                                      "E2_CEC vs. E1_CEC",
                                      "E2_DUOD vs. E1_CEC",
                                      "E2_IL vs. E1_CEC",
                                      "E1_IL vs. E1_DUOD",
                                      "E2_CEC vs. E1_DUOD",
                                      "E2_DUOD vs. E1_DUOD",
                                      "E2_IL vs. E1_DUOD",
                                      "E2_CEC vs. E1_IL",
                                      "E2_DUOD vs. E1_IL",
                                      "E2_IL vs. E1_IL",
                                      "E2_DUOD vs. E2_CEC",
                                      "E2_IL vs. E2_CEC",
                                      "E2_IL vs. E2_DUOD"))
  
lo = floor(min(df_fig_pair$value))
up = ceiling(max(df_fig_pair$value))
mid = (lo + up)/2
fig_pair = df_fig_pair %>%
  ggplot(aes(x = group, y = taxon, fill = value)) + 
  geom_tile(color = "black") +
  scale_fill_gradient2(low = "blue", high = "red", mid = "white", 
                       na.value = "white", midpoint = mid, limit = c(lo, up),
                       name = NULL) +
  geom_text(aes(group, taxon, label = value), color = "black", size = 4) +
  labs(x = NULL, y = NULL, title = "Log fold change of pairwise comparisons in E1 vs E2 by Location") +
  theme_bw() +
  theme(
    plot.title = element_text(hjust = 1),
    title =element_text(size=22, face='bold'),
    axis.text.x = element_text(color="black", size=12, angle = 90, hjust = 1),
    axis.text.y = element_text(color="black", size=12),
    strip.text.y = element_text(color="black", size=12, angle=90),
    #legend.title = element_text(color="black", size=20),
    legend.text = element_text(color="black", size=12),
    legend.key.size = unit(1, 'cm'),
    #legend.position = "right",
    strip.text = element_text(size = 16),
    axis.title.x = element_text(color="black", size=14),
    axis.title.y = element_text(color="black", size=14, angle=90),
    panel.grid.major = element_blank(), # switch off major gridlines
    panel.grid.minor = element_blank(), # switch off minor gridlines
  )
#CREATE A PDF AND SAVES IT TO THE RIGHT DIRECTORY/FOLDER. 
pdf("/Users/connermertz/Dropbox (vlab)/Conner Mertz/UNM biology/Chapter 1 E1 vs E2/Methods/16S rRNA gene sequencing/Sequencing_workup/ANCOMoutput/E1vsE2_Location_LFC_Pairwise.pdf", height = 12, width = 14)
fig_pair
dev.off()
  
```



#lets make the cool bar plot!- but careful this makes  the LFC and SE for the reference group (“obese” in this example) set to be 0s.
```{r}
E1vsE2_LOCATION_res_trend
```


```{r}
#code works but only identifies 2ASVs- we should use the pairwise comparisons instead
df_fig_trend = E1vsE2_LOCATION_res_trend %>%
    dplyr::filter(diff_abn) %>%
    transmute(taxon,
              lfc = lfc_Exp_LocationE1_DUOD,
              se = se_Exp_LocationE1_DUOD,
              p_val,
              group = "E1_DUOD - E1_CEC") %>%
    bind_rows(
        E1vsE2_LOCATION_res_trend %>%
            dplyr::filter(diff_abn) %>%
            transmute(taxon,
                      lfc = lfc_Exp_LocationE1_IL,
                      se = se_Exp_LocationE1_IL,
                      p_val,
                      group = "E1_IL - E1_CEC")
    ) %>%
    bind_rows(
        E1vsE2_LOCATION_res_trend %>%
            dplyr::filter(diff_abn) %>%
            transmute(taxon,
                      lfc = lfc_Exp_LocationE2_CEC,
                      se = se_Exp_LocationE2_CEC,
                      p_val,
                      group = "E2_CEC - E1_CEC")
          
    ) %>%
  bind_rows(
        E1vsE2_LOCATION_res_trend %>%
            dplyr::filter(diff_abn) %>%
            transmute(taxon,
                      lfc = lfc_Exp_LocationE2_DUOD,
                      se = se_Exp_LocationE2_DUOD,
                      p_val,
                      group = "E2_DUOD - E1_CEC")
          
    ) %>%
  bind_rows(
        E1vsE2_LOCATION_res_trend %>%
            dplyr::filter(diff_abn) %>%
            transmute(taxon,
                      lfc = lfc_Exp_LocationE2_IL,
                      se = se_Exp_LocationE2_IL,
                      p_val,
                      group = "E2_IL - E1_CEC")
          
    )
df_fig_trend$group = factor(df_fig_trend$group, 
                           levels = c("E1_DUOD - E1_CEC", "E1_IL - E1_CEC","E2_CEC - E1_CEC","E2_DUOD - E1_CEC", "E2_IL - E1_CEC"))
  
fig_trend = df_fig_trend %>% 
  ggplot(aes(x = group, y = lfc, fill = group)) + 
  geom_bar(stat = "identity", position = position_dodge(), color = "black") +
  geom_errorbar(aes(ymin = lfc - se, ymax = lfc + se), width = .2,
                position = position_dodge(.9)) +
  facet_wrap(vars(taxon), nrow = 2, scales = "free") +
  labs(x = NULL, y = NULL, title = "Log fold change as compared to E1_CEC") +
  scale_fill_brewer(palette = "Set2", name = NULL) +
  theme_bw() +
  theme(
    plot.title = element_text(hjust = 1),
    title =element_text(size=22, face='bold'),
    axis.text.x = element_text(color="black", size=12, angle = 90, hjust = 1),
    axis.text.y = element_text(color="black", size=12),
    strip.text.y = element_text(color="black", size=12, angle=90),
    #legend.title = element_text(color="black", size=20),
    legend.text = element_text(color="black", size=12),
    legend.key.size = unit(0.5, 'cm'),
    #legend.position = "right",
    strip.text = element_text(size = 16),
    axis.title.x = element_text(color="black", size=14),
    axis.title.y = element_text(color="black", size=14, angle=90),
    panel.grid.major = element_blank(), # switch off major gridlines
    panel.grid.minor = element_blank(), # switch off minor gridlines
  )
#CREATE A PDF AND SAVES IT TO THE RIGHT DIRECTORY/FOLDER. 
pdf("/Users/connermertz/Documents/Conner Documents/UNM biology/Chapter 1 E1 vs E2/ANCOMoutput/E1vsE2_Location_Trend.fig.pdf", height = 14, width = 24)
fig_trend
dev.off()


```

#E1_E2_Location
ANCOMBC2
pairwise comparisons by experiment and Location
```{r}
#create your data frame for log fold change of pairwise comparisons for dot plot! We will use this to graph dot plots of bargraphs
df_fig_trend = E1vsE2_LOCATION_res_pair %>%
  dplyr::filter(diff_Exp_LocationE1_DUOD == 1) %>%
  mutate(lfc_E1_DUOD = ifelse(diff_Exp_LocationE1_DUOD == 1, 
                             lfc_Exp_LocationE1_DUOD, 0)) %>%
              transmute(taxon,
              lfc = lfc_E1_DUOD,
              se = se_Exp_LocationE1_DUOD,
              p_val= p_Exp_LocationE1_DUOD,
              group = "E1_DUOD - E1_CEC") %>%
              
  bind_rows(
  E1vsE2_LOCATION_res_pair %>%
  dplyr::filter(diff_Exp_LocationE1_IL == 1) %>%
  mutate(lfc_E1_IL = ifelse(diff_Exp_LocationE1_IL == 1, 
                             lfc_Exp_LocationE1_IL, 0)) %>%
            transmute(taxon,
                      lfc = lfc_E1_IL,
                      se = se_Exp_LocationE1_IL,
                      p_val = p_Exp_LocationE1_IL,
                      group = "E1_IL - E1_CEC")
  ) %>%
    bind_rows(
    E1vsE2_LOCATION_res_pair %>%
    dplyr::filter(diff_Exp_LocationE2_CEC == 1) %>%
    mutate(lfc_E2_CEC = ifelse(diff_Exp_LocationE2_CEC == 1, 
                                   lfc_Exp_LocationE2_CEC, 0)) %>%
            transmute(taxon,
                      lfc = lfc_E2_CEC,
                      se = se_Exp_LocationE2_CEC,
                      p_val= p_Exp_LocationE2_CEC,
                      group = "E2_CEC - E1_CEC")
    ) %>%
   bind_rows(
    E1vsE2_LOCATION_res_pair %>%
    dplyr::filter(diff_Exp_LocationE2_DUOD == 1) %>%
    mutate(lfc_E2_DUOD = ifelse(diff_Exp_LocationE2_DUOD == 1, 
                                   lfc_Exp_LocationE2_DUOD, 0)) %>%
                    transmute(taxon,
                      lfc = lfc_E2_DUOD,
                      se = se_Exp_LocationE2_DUOD,
                      p_val = p_Exp_LocationE2_DUOD,
                      group = "E2_DUOD - E1_CEC")
    ) %>%  
   bind_rows(
    E1vsE2_LOCATION_res_pair %>%
    dplyr::filter(diff_Exp_LocationE2_IL == 1) %>%
    mutate(lfc_E2_IL = ifelse(diff_Exp_LocationE2_IL == 1, 
                                   lfc_Exp_LocationE2_IL, 0)) %>%  
                    transmute(taxon,
                      lfc = lfc_E2_IL,
                      se = se_Exp_LocationE2_IL,
                      p_val = p_Exp_LocationE2_IL,
                      group = "E2_IL - E1_CEC") 
    ) %>%                 
    bind_rows(
    E1vsE2_LOCATION_res_pair %>%
    dplyr::filter(diff_Exp_LocationE1_IL_Exp_LocationE1_DUOD == 1) %>%
    mutate(lfc_E1_IL_E1_DUOD = ifelse(diff_Exp_LocationE1_IL_Exp_LocationE1_DUOD == 1, 
                                   lfc_Exp_LocationE1_IL_Exp_LocationE1_DUOD, 0)) %>%
                    transmute(taxon,
                      lfc = lfc_E1_IL_E1_DUOD,
                      se = se_Exp_LocationE1_IL_Exp_LocationE1_DUOD,
                      p_val = p_Exp_LocationE1_IL_Exp_LocationE1_DUOD,
                      group = "E1_IL - E1_DUOD")
    ) %>%              
    bind_rows(
    E1vsE2_LOCATION_res_pair %>%
    dplyr::filter(diff_Exp_LocationE2_CEC_Exp_LocationE1_DUOD == 1) %>%
    mutate(lfc_E2_CEC_E1_DUOD = ifelse(diff_Exp_LocationE2_CEC_Exp_LocationE1_DUOD == 1, 
                                   lfc_Exp_LocationE2_CEC_Exp_LocationE1_DUOD, 0)) %>%
                    transmute(taxon,
                      lfc = lfc_E2_CEC_E1_DUOD,
                      se = se_Exp_LocationE2_CEC_Exp_LocationE1_DUOD,
                      p_val = p_Exp_LocationE2_CEC_Exp_LocationE1_DUOD,
                      group = "E2_CEC - E1_DUOD") 
    ) %>%              
    bind_rows(
    E1vsE2_LOCATION_res_pair %>%
    dplyr::filter(diff_Exp_LocationE2_DUOD_Exp_LocationE1_DUOD == 1) %>%
    mutate(lfc_E2_DUOD_E1_DUOD = ifelse(diff_Exp_LocationE2_DUOD_Exp_LocationE1_DUOD == 1, 
                                   lfc_Exp_LocationE2_DUOD_Exp_LocationE1_DUOD, 0)) %>%
                    transmute(taxon,
                      lfc = lfc_E2_DUOD_E1_DUOD,
                      se = se_Exp_LocationE2_DUOD_Exp_LocationE1_DUOD,
                      p_val = p_Exp_LocationE2_DUOD_Exp_LocationE1_DUOD,
                      group = "E2_DUOD - E1_DUOD")
    ) %>%
    bind_rows(
    E1vsE2_LOCATION_res_pair %>%
    dplyr::filter(diff_Exp_LocationE2_IL_Exp_LocationE1_DUOD == 1) %>%
    mutate(lfc_E2_IL_E1_DUOD = ifelse(diff_Exp_LocationE2_IL_Exp_LocationE1_DUOD == 1, 
                                   lfc_Exp_LocationE2_IL_Exp_LocationE1_DUOD, 0)) %>%
                    transmute(taxon,
                      lfc = lfc_E2_IL_E1_DUOD,
                      se = se_Exp_LocationE2_IL_Exp_LocationE1_DUOD,
                      p_val = p_Exp_LocationE2_IL_Exp_LocationE1_DUOD,
                      group = "E2_IL - E1_DUOD")
    ) %>%
    bind_rows(
    E1vsE2_LOCATION_res_pair %>%
    dplyr::filter(diff_Exp_LocationE2_CEC_Exp_LocationE1_IL == 1) %>%
    mutate(lfc_E2_CEC_E1_IL = ifelse(diff_Exp_LocationE2_CEC_Exp_LocationE1_IL == 1, 
                                   lfc_Exp_LocationE2_CEC_Exp_LocationE1_IL, 0)) %>%
                transmute(taxon,
                      lfc = lfc_E2_CEC_E1_IL,
                      se = se_Exp_LocationE2_CEC_Exp_LocationE1_IL,
                      p_val = p_Exp_LocationE2_CEC_Exp_LocationE1_IL,
                      group = "E2_CEC - E1_IL")
    ) %>%
 bind_rows(
    E1vsE2_LOCATION_res_pair %>%
    dplyr::filter(diff_Exp_LocationE2_DUOD_Exp_LocationE1_IL == 1) %>%
    mutate(lfc_E2_DUOD_E1_IL = ifelse(diff_Exp_LocationE2_DUOD_Exp_LocationE1_IL == 1, 
                                   lfc_Exp_LocationE2_DUOD_Exp_LocationE1_IL, 0)) %>% 
                    transmute(taxon,
                      lfc = lfc_E2_DUOD_E1_IL,
                      se = se_Exp_LocationE2_DUOD_Exp_LocationE1_IL,
                      p_val = p_Exp_LocationE2_DUOD_Exp_LocationE1_IL,
                      group = "E2_DUOD - E1_IL")
    ) %>%
bind_rows(
    E1vsE2_LOCATION_res_pair %>%
    dplyr::filter(diff_Exp_LocationE2_IL_Exp_LocationE1_IL == 1) %>%
    mutate(lfc_E2_IL_E1_IL = ifelse(diff_Exp_LocationE2_IL_Exp_LocationE1_IL == 1, 
                                   lfc_Exp_LocationE2_IL_Exp_LocationE1_IL, 0)) %>% 
                    transmute(taxon,
                      lfc = lfc_E2_IL_E1_IL,
                      se = se_Exp_LocationE2_IL_Exp_LocationE1_IL,
                      p_val = p_Exp_LocationE2_IL_Exp_LocationE1_IL,
                      group = "E2_IL - E1_IL")
    ) %>%    
bind_rows(
    E1vsE2_LOCATION_res_pair %>%
    dplyr::filter(diff_Exp_LocationE2_DUOD_Exp_LocationE2_CEC == 1) %>%
    mutate(lfc_E2_DUOD_E2_CEC = ifelse(diff_Exp_LocationE2_DUOD_Exp_LocationE2_CEC == 1, 
                                   lfc_Exp_LocationE2_DUOD_Exp_LocationE2_CEC, 0)) %>% 
                    transmute(taxon,
                      lfc = lfc_E2_DUOD_E2_CEC,
                      se = se_Exp_LocationE2_DUOD_Exp_LocationE2_CEC,
                      p_val= p_Exp_LocationE2_DUOD_Exp_LocationE2_CEC,
                      group = "E2_DUOD - E2_CEC")
) %>%
bind_rows(
    E1vsE2_LOCATION_res_pair %>%
    dplyr::filter(diff_Exp_LocationE2_IL_Exp_LocationE2_CEC == 1) %>%
    mutate(lfc_E2_IL_E2_CEC = ifelse(diff_Exp_LocationE2_IL_Exp_LocationE2_CEC == 1, 
                                   lfc_Exp_LocationE2_IL_Exp_LocationE2_CEC, 0)) %>%
                    transmute(taxon,
                      lfc = lfc_E2_IL_E2_CEC,
                      se = se_Exp_LocationE2_IL_Exp_LocationE2_CEC,
                      p_val = p_Exp_LocationE2_IL_Exp_LocationE2_CEC,
                      group = "E2_IL - E2_CEC")
    ) %>%  
bind_rows(
    E1vsE2_LOCATION_res_pair %>%
    dplyr::filter(diff_Exp_LocationE2_IL_Exp_LocationE2_DUOD == 1) %>%
    mutate(lfc_E2_IL_E2_DUOD = ifelse(diff_Exp_LocationE2_IL_Exp_LocationE2_DUOD == 1, 
                                        lfc_Exp_LocationE2_IL_Exp_LocationE2_DUOD, 0)) %>%
                    transmute(taxon,
                      lfc = lfc_E2_IL_E2_DUOD,
                      se = se_Exp_LocationE2_IL_Exp_LocationE2_DUOD,
                      p_val = p_Exp_LocationE2_IL_Exp_LocationE2_DUOD,
                      group = "E2_IL - E2_DUOD")
    )
      df_fig_trend$group = factor(df_fig_trend$group, 
                            levels = c("E1_DUOD - E1_CEC",
                                      "E1_IL - E1_CEC",
                                      "E2_CEC - E1_CEC",
                                      "E2_DUOD - E1_CEC",
                                      "E2_IL - E1_CEC",
                                      "E1_IL - E1_DUOD",
                                      "E2_CEC - E1_DUOD",
                                      "E2_DUOD - E1_DUOD",
                                      "E2_IL - E1_DUOD",
                                      "E2_CEC - E1_IL",
                                      "E2_DUOD - E1_IL",
                                      "E2_IL - E1_IL",
                                      "E2_DUOD - E2_CEC",
                                      "E2_IL - E2_CEC",
                                      "E2_IL - E2_DUOD"))


```
Save the df you created
```{r}
write.csv(df_fig_trend, "/Users/connermertz/Dropbox (vlab)/Conner Mertz/UNM biology/Chapter 1 E1 vs E2/Methods/16S rRNA gene sequencing/Sequencing_workup/ANCOMoutput/codeexports/E1_E2_forcomparison/E1vsE2_LOCATION_df_fig_trend.csv") #ADD IN OTHER CATEGORIES YOU WANT TO GRAPH 

#read in your EDITED DF csv: 
E1vsE2_LOCATION_df_fig_trend <- read.csv("/Users/connermertz/Dropbox (vlab)/Conner Mertz/UNM biology/Chapter 1 E1 vs E2/Methods/16S rRNA gene sequencing/Sequencing_workup/ANCOMoutput/codeexports/E1_E2_forcomparison/E1vsE2_LOCATION_df_fig_trend.csv")
#change name so target_ids match on both files
names(E1vsE2_LOCATION_df_fig_trend)[2] ="taxon" 

#read in your TAXA classification file 
Taxonomy_classification <- read.csv("/Users/connermertz/Dropbox (vlab)/Conner Mertz/UNM biology/Chapter 1 E1 vs E2/Methods/16S rRNA gene sequencing/Sequencing_workup/DADA2_output/ALL_taxa.csv") #if this doesn't load be sure to dowload the file from the cloud, then it will work
#change name so target_ids match on both files
names(Taxonomy_classification)[1] ="taxon"

#now we need to get our DF results and join it with our Taxonomy results from CAT (combining two data frames)
#we only want to keep the genes that showed up with gapmind, we don't need the other genes that showed up
Part_DF <- Taxonomy_classification %>% 
  inner_join (E1vsE2_LOCATION_df_fig_trend, by = "taxon")

write.csv(Part_DF,"/Users/connermertz/Dropbox (vlab)/Conner Mertz/UNM biology/Chapter 1 E1 vs E2/Methods/16S rRNA gene sequencing/Sequencing_workup/ANCOMoutput/codeexports/E1_E2_forcomparison/E1vsE2_LOCATION_DF_FIG_TREND_AND_Taxonomy.csv") # went in here and edited file to my liking. I saved the new version as edit_E1vsE2_LOCATION_DF_FIG_TREND_AND_Taxonomy.csv. That is what I will be using for the figures. I basically switched the signs for the differential abundant taxa to be positive for E1 and negative for E2 but consistent with whatever category it was being compared to. 

```


Just keep it simple 
#trying to compare just by experiment and location so:
#E1_duod vs E2_duod
#E1_ile vs E2_ile
#E1_cec vs E2_cec

#read in the edited df
```{r}
E1vsE2_df <- read.csv("/Users/connermertz/Dropbox (vlab)/Conner Mertz/UNM biology/Chapter 1 E1 vs E2/Methods/16S rRNA gene sequencing/Sequencing_workup/ANCOMoutput/codeexports/E1_E2_forcomparison/edit_E1vsE2_LOCATION_DF_FIG_TREND_AND_Taxonomy.csv")
```

#filter out to just include: 
#E1_duod vs E2_duod
#E1_ile vs E2_ile
#E1_cec vs E2_cec

AND

#filter to keep comparisons of interest and mutate so that location and ASV's are in the correct order

```{r}
E1vsE2_LOCATION_df1 <- read.csv("/Users/connermertz/Dropbox (vlab)/Conner Mertz/UNM biology/Chapter 1 E1 vs E2/Methods/16S rRNA gene sequencing/Sequencing_workup/ANCOMoutput/codeexports/E1_E2_forcomparison/edit_E1vsE2_LOCATION_DF_FIG_TREND_AND_Taxonomy.csv") %>%
  filter(group_in_E1vsE2_order ==  "E1_DUODvsE2_DUOD" |
           group_in_E1vsE2_order ==  "E1_ILvsE2_IL" |
           group_in_E1vsE2_order ==  "E1_CECvsE2_CEC" )%>%
 mutate(Differentially_Abundant_Group = factor(Differentially_Abundant_Group, levels = c("E1_DUOD","E1_IL", "E1_CEC", "E2_DUOD", "E2_IL", "E2_CEC"))) %>%
 mutate(Differentially_Abundant_Location = factor(Differentially_Abundant_Location, levels = c("DUOD","IL", "CEC"))) %>%
 mutate(taxon = factor(taxon, levels = c("ASV8", "ASV58", "ASV80", "ASV6", "ASV78", "ASV1", "ASV21", "ASV72", "ASV56", "ASV13", "ASV44", "ASV16", "ASV37", "ASV118")))

#save df
write.csv(E1vsE2_LOCATION_df1,"/Users/connermertz/Dropbox (vlab)/Conner Mertz/UNM biology/Chapter 1 E1 vs E2/Methods/16S rRNA gene sequencing/Sequencing_workup/ANCOMoutput/codeexports/E1_E2_forcomparison/E1vsE2_LOCATION_only.csv")

```

change color of location
```{r}
#first you want to factor your items. These are the names of the Location in the order I want them to be every time. 
E1vsE2_LOCATION_df1$Differentially_Abundant_Location <-factor(E1vsE2_LOCATION_df1$Differentially_Abundant_Location, levels=c("DUOD","IL", "CEC"))

#now create an object that is a list of colors in a certain order.The colors I chose are "color-blind friendly":
LocationScale <- c("#440a48","#47da7a","#dba0ff")
#scales::show_col(LocationScale)

#now we want to merge the two. We are taking the order from the locations and telling it that it needs to match the order from the cbPalette.
names(LocationScale) <- levels(E1vsE2_LOCATION_df1$Differentially_Abundant_Location)
#Have you ever used the scale_colour_manual command? Now we can combine that command and all of the stuff we want in it into a new object, LocationColors. At the end of your ggplot code, just add a +LocationColors (or whatever you decide to name it), and it should pull the colors you made! 
LocationColors<- scale_colour_manual(name="Location", values=LocationScale)
```



#When we compared the synthetic vs the seminatural in each gut section we found:
E1_duod vs E2_duod: in the duodenum only ASV58 was found to be differentially abundant in the semi-natural diet compared to the synthetic diet.  
E1_ile vs E2_ile: in the ileum only ASV58 was found to be differentially abundant in the semi-natural diet compared to the synthetic diet. 
E1_cec vs E2_cec: In the cecum ASV13, ASV80 and ASV8 are differentially abundant in the synthetic diet treatment while ASV58 is differentially abundant in the semi-natural diet. 
```{r}
fig_trend = E1vsE2_LOCATION_df1 %>% 
  ggplot(aes(x = lfc_sign_change_if_needed, y = taxon, color = Differentially_Abundant_Location)) + 
  geom_point(size = 4, position = position_dodge()) +
  geom_errorbar(
    aes(xmin = lfc_sign_change_if_needed - se, xmax = lfc_sign_change_if_needed + se), 
    width = .2) +
  geom_vline(aes(xintercept = 0), linetype = "dashed")+
  #facet_wrap(~Differentially_Abundant_Location) +
  labs(x = "Log Fold Change", y = "ASV", title = "Differentially Abundant Taxa\n by Gut Section and Treatment ") +
  LocationColors +
  scale_x_continuous(limits=c(-7, 7)) +
  theme_bw() +
  theme(
    plot.title = element_text(hjust = 0.5),
    title =element_text(size=18, face='bold'),
    axis.text.x = element_text(color="black", size=12, angle = 90),
    axis.text.y = element_text(color="black", size=12),
    strip.text = element_text(size=6),
    strip.text.y = element_text(color="black", size=12, angle=90),
    legend.title = element_text(color="black", size=14),
    legend.text = element_text(color="black", size=12),
    legend.key.size = unit(1, 'cm'),
    legend.position = "bottom",
    axis.title.x = element_text(color="black", size=12),
    axis.title.y = element_text(color="black", size=12, angle=90),
    panel.grid.major = element_blank(), # switch off major gridlines
    panel.grid.minor = element_blank()# switch off minor gridlines
  )
#CREATE A PDF AND SAVES IT TO THE RIGHT DIRECTORY/FOLDER. 
pdf("/Users/connermertz/Dropbox (vlab)/Conner Mertz/UNM biology/Chapter 1 E1 vs E2/Methods/16S rRNA gene sequencing/Sequencing_workup/ANCOMoutput/Plots/E1vsE2_Location_LFC.pdf", height = 4, width = 8)
fig_trend
dev.off()
```
Results: 
#Differentially abundant ASVs associated with the Synthetic or Semi-natural diet identified by ANCOMBC2 pairwise comparisons. Each dot represents a differentially abundant ASV (determined via pair-wise comparisons) plotted by its taxonomic assignment on the y- axis and log-fold change on the x axis. Positive log fold change indicated taxa enriched in the synthetic diet, while negative log fold change taxa enriched in the semi-natural diet.Error bars were calculated using the standard error. (standard errors for all differentially abundant taxa identified are below 1.1, and all p-values < 0.002) 

#When we compared the synthetic vs the seminatural in each gut section we found:
E1_duod vs E2_duod: in the duodenum only ASV58 was found to be differentially abundant in the semi-natural diet compared to the synthetic diet.  
E1_ile vs E2_ile: in the ileum only ASV58 was found to be differentially abundant in the semi-natural diet compared to the synthetic diet. 
E1_cec vs E2_cec: In the cecum ASV13, ASV80 and ASV8 are differentially abundant in the synthetic diet treatment while ASV58 is differentially abundant in the semi-natural diet.






#####



#Now lets look at location pairwise comparisons WITHIN each treatment:
Starting with the Synthetic diet:
E1_duod vs E1_ile
E1_duod vs E1_cec
E1_ile vs E1_cec



```{r}
set.seed(123)
output = ancombc2(ps_raw_E1_forcomparison, assay_name = NULL, tax_level = NULL,
                  fix_formula = "Percent_Protein + Exp_Location", rand_formula = NULL,
                  p_adj_method = "holm", pseudo = 0, pseudo_sens = TRUE,
                  prv_cut = 0.10, lib_cut = 1000, s0_perc = 0.05,
                  group = "Exp_Location", struc_zero = TRUE, neg_lb = TRUE,
                  alpha = 0.05, n_cl = 2, verbose = TRUE,
                  global = TRUE, pairwise = TRUE, dunnet = TRUE, trend = TRUE,
                  iter_control = list(tol = 1e-2, max_iter = 20, 
                                      verbose = TRUE),
                  em_control = list(tol = 1e-5, max_iter = 100),
                  lme_control = lme4::lmerControl(),
                  mdfdr_control = list(fwer_ctrl_method = "holm", B = 100),
                  trend_control = list(contrast = list(matrix(c(1, 0, -1, 1),
                                                              nrow = 2, 
                                                              byrow = TRUE),
                                                       matrix(c(-1, 0, 1, -1),
                                                              nrow = 2, 
                                                              byrow = TRUE)),
                                       node = list(2, 2),
                                       solver = "ECOS",
                                       B = 100))
```
Save the output
```{r}
res_prim = output$res
saveRDS(res_prim, "/Users/connermertz/Dropbox (vlab)/Conner Mertz/UNM biology/Chapter 1 E1 vs E2/Methods/16S rRNA gene sequencing/Sequencing_workup/ANCOMoutput/codeexports/E1_forcomparison/E1_LOCATION_res_prim.rds")
#read in the ANCOMBC output if you want to see E1 vs E2 by Location
E1_LOCATION_res_prim <- readRDS("/Users/connermertz/Dropbox (vlab)/Conner Mertz/UNM biology/Chapter 1 E1 vs E2/Methods/16S rRNA gene sequencing/Sequencing_workup/ANCOMoutput/codeexports/E1_forcomparison/E1_LOCATION_res_prim.rds")

res_pair = output$res_pair
saveRDS(res_pair, "/Users/connermertz/Dropbox (vlab)/Conner Mertz/UNM biology/Chapter 1 E1 vs E2/Methods/16S rRNA gene sequencing/Sequencing_workup/ANCOMoutput/codeexports/E1_forcomparison/E1_LOCATION_res_pair.rds")
E1_LOCATION_res_pair <- readRDS("/Users/connermertz/Dropbox (vlab)/Conner Mertz/UNM biology/Chapter 1 E1 vs E2/Methods/16S rRNA gene sequencing/Sequencing_workup/ANCOMoutput/codeexports/E1_forcomparison/E1_LOCATION_res_pair.rds")

res_trend = output$res_trend
saveRDS(res_trend, "/Users/connermertz/Dropbox (vlab)/Conner Mertz/UNM biology/Chapter 1 E1 vs E2/Methods/16S rRNA gene sequencing/Sequencing_workup/ANCOMoutput/codeexports/E1_forcomparison/E1_LOCATION_res_trend.rds")
E1_LOCATION_res_trend <- readRDS("/Users/connermertz/Dropbox (vlab)/Conner Mertz/UNM biology/Chapter 1 E1 vs E2/Methods/16S rRNA gene sequencing/Sequencing_workup/ANCOMoutput/codeexports/E1_forcomparison/E1_LOCATION_res_trend.rds")
```


##ANCOM-BC2 pairwise directional test
ANCOM-BC2 pairwise directional test aims to determine taxa that are differentially abundant between any pair of two groups across three or more experimental groups, while controlling the mdFDR.

In this example, we want to identify taxa that are differentially abundant between any pair of two groups across “lean”, “overweight”, and “obese”. The result contains: 1) log fold changes, 2) standard errors, 3) test statistics, 4) p-values, 5) adjusted p-values, 6) indicators of whether the taxon is differentially abundant (TRUE) or not (FALSE).
```{r}
E1_LOCATION_res_pair
write.csv(E1_LOCATION_res_pair, "/Users/connermertz/Dropbox (vlab)/Conner Mertz/UNM biology/Chapter 1 E1 vs E2/Methods/16S rRNA gene sequencing/Sequencing_workup/ANCOMoutput/codeexports/E1_forcomparison/E1_LOCATION_res_pair_edit.csv")

#read in your res_pair csv: 
E1_LOCATION_res_pair2 <- read.csv("/Users/connermertz/Dropbox (vlab)/Conner Mertz/UNM biology/Chapter 1 E1 vs E2/Methods/16S rRNA gene sequencing/Sequencing_workup/ANCOMoutput/codeexports/E1_forcomparison/E1_LOCATION_res_pair_edit.csv")
#change name so target_ids match on both files
names(E1_LOCATION_res_pair2)[2] ="taxon" 

#read in your taxonomy classification file 
Taxonomy_classification <- read.csv("/Users/connermertz/Dropbox (vlab)/Conner Mertz/UNM biology/Chapter 1 E1 vs E2/Methods/16S rRNA gene sequencing/Sequencing_workup/DADA2_output/ALL_taxa.csv") #if this doesn't load be sure to download the file from the cloud, then it will work
#change name so target_ids match on both files
names(Taxonomy_classification)[1] ="taxon"

#now we need to get our ANCOMBC results and join it with our Taxonomy results from DADA2 (combining two data frames)
#we only want to keep the ASVs that showed up with ANCOMBC, we don't need the other ASVs that showed up
Part_1 <- Taxonomy_classification %>% 
  inner_join (E1_LOCATION_res_pair2, by = "taxon")

write.csv(Part_1,"/Users/connermertz/Dropbox (vlab)/Conner Mertz/UNM biology/Chapter 1 E1 vs E2/Methods/16S rRNA gene sequencing/Sequencing_workup/ANCOMoutput/codeexports/E1_forcomparison/E1_LOCATION_res_pair_AND_Taxonomy.csv") # went in here and edited file to my liking
```



#E1_Location
ANCOMBC2
pairwise comparisons by Location
```{r}
#create your data frame for log fold change of pairwise comparisons for dot plot! We will use this to graph dot plots of bargraphs
df_fig_trend = E1_LOCATION_res_pair %>%
  dplyr::filter(diff_Exp_LocationE1_DUOD == 1) %>%
  mutate(lfc_E1_DUOD = ifelse(diff_Exp_LocationE1_DUOD == 1, 
                             lfc_Exp_LocationE1_DUOD, 0)) %>%
              transmute(taxon,
              lfc = lfc_E1_DUOD,
              se = se_Exp_LocationE1_DUOD,
              p_val= p_Exp_LocationE1_DUOD,
              group = "E1_DUOD - E1_CEC") %>%
              
  bind_rows(
  E1_LOCATION_res_pair %>%
  dplyr::filter(diff_Exp_LocationE1_IL == 1) %>%
  mutate(lfc_E1_IL = ifelse(diff_Exp_LocationE1_IL == 1, 
                             lfc_Exp_LocationE1_IL, 0)) %>%
            transmute(taxon,
                      lfc = lfc_E1_IL,
                      se = se_Exp_LocationE1_IL,
                      p_val = p_Exp_LocationE1_IL,
                      group = "E1_IL - E1_CEC")
 
    ) %>%                 
    bind_rows(
    E1_LOCATION_res_pair %>%
    dplyr::filter(diff_Exp_LocationE1_IL_Exp_LocationE1_DUOD == 1) %>%
    mutate(lfc_E1_IL_E1_DUOD = ifelse(diff_Exp_LocationE1_IL_Exp_LocationE1_DUOD == 1, 
                                   lfc_Exp_LocationE1_IL_Exp_LocationE1_DUOD, 0)) %>%
                    transmute(taxon,
                      lfc = lfc_E1_IL_E1_DUOD,
                      se = se_Exp_LocationE1_IL_Exp_LocationE1_DUOD,
                      p_val = p_Exp_LocationE1_IL_Exp_LocationE1_DUOD,
                      group = "E1_IL - E1_DUOD")
    )
      df_fig_trend$group = factor(df_fig_trend$group, 
                            levels = c("E1_DUOD - E1_CEC",
                                      "E1_IL - E1_CEC",
                                      "E1_IL - E1_DUOD"))


```

Save the df you created
```{r}
write.csv(df_fig_trend, "/Users/connermertz/Dropbox (vlab)/Conner Mertz/UNM biology/Chapter 1 E1 vs E2/Methods/16S rRNA gene sequencing/Sequencing_workup/ANCOMoutput/codeexports/E1_forcomparison/E1_LOCATION_df_fig_trend.csv") #ADD IN OTHER CATEGORIES YOU WANT TO GRAPH 

#read in your EDITED DF csv: 
E1_LOCATION_df_fig_trend <- read.csv("/Users/connermertz/Dropbox (vlab)/Conner Mertz/UNM biology/Chapter 1 E1 vs E2/Methods/16S rRNA gene sequencing/Sequencing_workup/ANCOMoutput/codeexports/E1_forcomparison/E1_LOCATION_df_fig_trend.csv")
#change name so target_ids match on both files
names(E1_LOCATION_df_fig_trend)[2] ="taxon" 

#read in your TAXA classification file 
Taxonomy_classification <- read.csv("/Users/connermertz/Dropbox (vlab)/Conner Mertz/UNM biology/Chapter 1 E1 vs E2/Methods/16S rRNA gene sequencing/Sequencing_workup/DADA2_output/ALL_taxa.csv") #if this doesn't load be sure to dowload the file from the cloud, then it will work
#change name so target_ids match on both files
names(Taxonomy_classification)[1] ="taxon"

#now we need to get our DF results and join it with our Taxonomy results from CAT (combining two data frames)
#we only want to keep the genes that showed up with gapmind, we don't need the other genes that showed up
Part_DF <- Taxonomy_classification %>% 
  inner_join (E1_LOCATION_df_fig_trend, by = "taxon")

write.csv(Part_DF,"/Users/connermertz/Dropbox (vlab)/Conner Mertz/UNM biology/Chapter 1 E1 vs E2/Methods/16S rRNA gene sequencing/Sequencing_workup/ANCOMoutput/codeexports/E1_forcomparison/E1_LOCATION_DF_FIG_TREND_AND_Taxonomy.csv") # went in here and edited file to my liking. I saved the new version as edit_E1_LOCATION_DF_FIG_TREND_AND_Taxonomy.csv. That is what I will be using for the figures. I basically switched the signs for the differential abundant taxa to be positive for E1 and negative for E2 but consistent with whatever category it was being compared to. 

```

#Now lets look at location pairwise comparisons WITHIN each treatment:
Starting with the Synthetic diet:
E1_duod vs E1_ile
E1_duod vs E1_cec
E1_ile vs E1_cec
```{r}
E1_LOCATION_df <- read.csv("/Users/connermertz/Dropbox (vlab)/Conner Mertz/UNM biology/Chapter 1 E1 vs E2/Methods/16S rRNA gene sequencing/Sequencing_workup/ANCOMoutput/codeexports/E1_forcomparison/edit_E1_LOCATION_DF_FIG_TREND_AND_Taxonomy.csv") %>%
  filter(group ==  "E1_DUOD - E1_IL" |
           group == "E1_IL - E1_DUOD" |
           group == "E1_DUOD - E1_CEC" |
           group == "E1_CEC - E1_DUOD" |
           group == "E1_IL - E1_CEC" |
           group == "E1_CEC - E1_IL" )%>%
 mutate(Differentially_Abundant_Group = factor(Differentially_Abundant_Group, levels = c("E1_DUOD","E1_IL", "E1_CEC"))) %>%
 mutate(Differentially_Abundant_Location = factor(Differentially_Abundant_Location, levels = c("DUOD","IL", "CEC"))) %>%
 mutate(taxon = factor(taxon, levels = c("ASV8", "ASV31", "ASV6", "ASV78", "ASV1", "ASV10", "ASV21", "ASV72", "ASV15", "ASV56", "ASV44", "ASV16", "ASV75", "ASV37", "ASV82", "ASV105", "ASV133")))

```


change color
```{r}
#first you want to factor your items. These are the names of the Location in the order I want them to be every time. 
E1_LOCATION_df$Differentially_Abundant_Location <-factor(E1_LOCATION_df$Differentially_Abundant_Location, levels=c("DUOD","IL", "CEC"))

#now create an object that is a list of colors in a certain order.The colors I chose are "color-blind friendly":
LocationScale <- c("#440a48","#47da7a","#dba0ff")
#scales::show_col(LocationScale)

#now we want to merge the two. We are taking the order from the locations and telling it that it needs to match the order from the cbPalette.
names(LocationScale) <- levels(E1_LOCATION_df$Differentially_Abundant_Location)
#Have you ever used the scale_colour_manual command? Now we can combine that command and all of the stuff we want in it into a new object, LocationColors. At the end of your ggplot code, just add a +LocationColors (or whatever you decide to name it), and it should pull the colors you made! 
LocationColors.1 <- scale_colour_manual(name="Location", values=LocationScale)
```



E1 BY LOCATION
```{r}
fig_trend = E1_LOCATION_df %>% 
  ggplot(aes(x = lfc, y = taxon, color = Differentially_Abundant_Location)) + 
  geom_point(size = 3, position = position_dodge()) +
  geom_errorbar(
    aes(xmin = lfc - se, xmax = lfc + se), 
    width = .2) +
  geom_vline(aes(xintercept = 0), linetype = "dashed")+
  facet_wrap(~group, strip.position="bottom") +
  labs(x = "Log Fold Change", y = "ASV", title = "Synthetic Diet: Differentially Abundant Taxa  \nin each gut section") +
  LocationColors.1 +
  scale_x_continuous(limits=c(-7, 7)) +
  theme_bw() +
  theme(
    plot.title = element_text(hjust = 0.5),
    title =element_text(size=18, face='bold'),
    axis.text.x = element_text(color="black", size=12, angle = 90),
    axis.text.y = element_text(color="black", size=12),
    strip.text = element_text(size=6),
    strip.text.y = element_text(color="black", size=12, angle=90),
    legend.title = element_text(color="black", size=14),
    legend.text = element_text(color="black", size=12),
    legend.key.size = unit(1, 'cm'),
    legend.position = "right",
    axis.title.x = element_text(color="black", size=12),
    axis.title.y = element_text(color="black", size=12, angle=90),
    panel.grid.major = element_blank(), # switch off major gridlines
    panel.grid.minor = element_blank()# switch off minor gridlines
  )
#CREATE A PDF AND SAVES IT TO THE RIGHT DIRECTORY/FOLDER. 
pdf("/Users/connermertz/Dropbox (vlab)/Conner Mertz/UNM biology/Chapter 1 E1 vs E2/Methods/16S rRNA gene sequencing/Sequencing_workup/ANCOMoutput/E1_forcomparison_Plots/E1_Location_LFC.pdf", height = 6, width = 8)
fig_trend
dev.off()
```

#Differentially abundant ASVs associated with the Synthetic Diet by gut sections identified by ANCOMBC2 pairwise comparisons. Each dot represents a differentially abundant ASV plotted by its taxonomic assignment on the y- axis and log-fold change on the x axis. Positive log fold change indicated taxa enriched in the Duodenum (left panel) or Ileum (right pannel), while negative log fold change indicated taxa enriched in the Cecum. Error bars were calculated using the standard error. (standard errors for all differentially abundant taxa identified are below 1.1, and all p-values < 0.003) 

#When we compared gut sections in the synthetic treatment we found:
E1_duod vs E1_ile: No differentially abundant taxa identified in the duodenum vs the ileum, therefore the two gut sections that comprise the small intestine have no statistically significant differences in taxa. 
E1_duod vs E1_cec: in the duodenum ASV10 and ASV1 were found to be differentially abundant compared to the cecum, while ASV105, ASV75, ASV16, ASV44, ASV6, ASV31, ASV8. 
E1_ile vs E1_cec: in the ileum ASV15, ASV72, ASV21, ASV10, ASV1, ASV78 was found to be differentially abundant compared to the cecum, while ASV133, ASV105, ASV82, ASV37, ASV16, ASV44, ASV56, ASV6, ASV31 was found to be differentially abundant in the cecum compared to the ileum.
















#Now lets look at location pairwise comparisons WITHIN each treatment:
Now lets do the Semi-Natural diet:
E2_duod vs E2_ile
E2_duod vs E2_cec
E2_ile vs E2_cec


```{r}
set.seed(123)
output = ancombc2(E2_ps_raw, assay_name = NULL, tax_level = NULL,
                  fix_formula = "Percent_Protein + Exp_Location", rand_formula = NULL,
                  p_adj_method = "holm", pseudo = 0, pseudo_sens = TRUE,
                  prv_cut = 0.10, lib_cut = 1000, s0_perc = 0.05,
                  group = "Exp_Location", struc_zero = TRUE, neg_lb = TRUE,
                  alpha = 0.05, n_cl = 2, verbose = TRUE,
                  global = TRUE, pairwise = TRUE, dunnet = TRUE, trend = TRUE,
                  iter_control = list(tol = 1e-2, max_iter = 20, 
                                      verbose = TRUE),
                  em_control = list(tol = 1e-5, max_iter = 100),
                  lme_control = lme4::lmerControl(),
                  mdfdr_control = list(fwer_ctrl_method = "holm", B = 100),
                  trend_control = list(contrast = list(matrix(c(1, 0, -1, 1),
                                                              nrow = 2, 
                                                              byrow = TRUE),
                                                       matrix(c(-1, 0, 1, -1),
                                                              nrow = 2, 
                                                              byrow = TRUE)),
                                       node = list(2, 2),
                                       solver = "ECOS",
                                       B = 100))
```
Save the output
```{r}
res_prim = output$res
saveRDS(res_prim, "/Users/connermertz/Dropbox (vlab)/Conner Mertz/UNM biology/Chapter 1 E1 vs E2/Methods/16S rRNA gene sequencing/Sequencing_workup/ANCOMoutput/codeexports/E2_ps_raw/E2_Location_res_prim.rds")
#read in the ANCOMBC output if you want to see E2
E2_Location_res_prim <- readRDS("/Users/connermertz/Dropbox (vlab)/Conner Mertz/UNM biology/Chapter 1 E1 vs E2/Methods/16S rRNA gene sequencing/Sequencing_workup/ANCOMoutput/codeexports/E2_ps_raw/E2_Location_res_prim.rds")

res_pair = output$res_pair
saveRDS(res_pair, "/Users/connermertz/Dropbox (vlab)/Conner Mertz/UNM biology/Chapter 1 E1 vs E2/Methods/16S rRNA gene sequencing/Sequencing_workup/ANCOMoutput/codeexports/E2_ps_raw/E2_Location_res_pair.rds")
E2_Location_res_pair <- readRDS("/Users/connermertz/Dropbox (vlab)/Conner Mertz/UNM biology/Chapter 1 E1 vs E2/Methods/16S rRNA gene sequencing/Sequencing_workup/ANCOMoutput/codeexports/E2_ps_raw/E2_Location_res_pair.rds")

res_trend = output$res_trend
saveRDS(res_trend, "/Users/connermertz/Dropbox (vlab)/Conner Mertz/UNM biology/Chapter 1 E1 vs E2/Methods/16S rRNA gene sequencing/Sequencing_workup/ANCOMoutput/codeexports/E2_ps_raw/E2_Location_res_trend.rds")
E2_Location_res_trend <- readRDS("/Users/connermertz/Dropbox (vlab)/Conner Mertz/UNM biology/Chapter 1 E1 vs E2/Methods/16S rRNA gene sequencing/Sequencing_workup/ANCOMoutput/codeexports/E2_ps_raw/E2_Location_res_trend.rds")
```


##ANCOM-BC2 pairwise directional test
ANCOM-BC2 pairwise directional test aims to determine taxa that are differentially abundant between any pair of two groups across three or more experimental groups, while controlling the mdFDR.

In this example, we want to identify taxa that are differentially abundant between any pair of two groups across “lean”, “overweight”, and “obese”. The result contains: 1) log fold changes, 2) standard errors, 3) test statistics, 4) p-values, 5) adjusted p-values, 6) indicators of whether the taxon is differentially abundant (TRUE) or not (FALSE).
```{r}
E2_Location_res_pair
write.csv(E2_Location_res_pair, "/Users/connermertz/Dropbox (vlab)/Conner Mertz/UNM biology/Chapter 1 E1 vs E2/Methods/16S rRNA gene sequencing/Sequencing_workup/ANCOMoutput/codeexports/E2_ps_raw/E2_Location_res_pair_edit.csv")

#read in your res_pair csv: 
E2_Location_res_pair2 <- read.csv("/Users/connermertz/Dropbox (vlab)/Conner Mertz/UNM biology/Chapter 1 E1 vs E2/Methods/16S rRNA gene sequencing/Sequencing_workup/ANCOMoutput/codeexports/E2_ps_raw/E2_Location_res_pair_edit.csv")
#change name so target_ids match on both files
names(E2_Location_res_pair2)[2] ="taxon" 

#read in your taxonomy classification file 
Taxonomy_classification <- read.csv("/Users/connermertz/Dropbox (vlab)/Conner Mertz/UNM biology/Chapter 1 E1 vs E2/Methods/16S rRNA gene sequencing/Sequencing_workup/DADA2_output/ALL_taxa.csv") #if this doesn't load be sure to download the file from the cloud, then it will work
#change name so target_ids match on both files
names(Taxonomy_classification)[1] ="taxon"

#now we need to get our ANCOMBC results and join it with our Taxonomy results from DADA2 (combining two data frames)
#we only want to keep the ASVs that showed up with ANCOMBC, we don't need the other ASVs that showed up
Part_1 <- Taxonomy_classification %>% 
  inner_join (E2_Location_res_pair2, by = "taxon")

write.csv(Part_1,"/Users/connermertz/Dropbox (vlab)/Conner Mertz/UNM biology/Chapter 1 E1 vs E2/Methods/16S rRNA gene sequencing/Sequencing_workup/ANCOMoutput/codeexports/E2_ps_raw/E2_Location_res_pair_AND_Taxonomy.csv") # went in here and edited file to my liking
```



#E2_Location
ANCOMBC2
pairwise comparisons by Location
```{r}
#create your data frame for log fold change of pairwise comparisons for dot plot! We will use this to graph dot plots of bargraphs
df_fig_trend = E2_Location_res_pair %>%
  dplyr::filter(diff_Exp_LocationE2_DUOD == 1) %>%
  mutate(lfc_E2_DUOD = ifelse(diff_Exp_LocationE2_DUOD == 1, 
                             lfc_Exp_LocationE2_DUOD, 0)) %>%
              transmute(taxon,
              lfc = lfc_E2_DUOD,
              se = se_Exp_LocationE2_DUOD,
              p_val= p_Exp_LocationE2_DUOD,
              group = "E2_DUOD - E2_CEC") %>%
              
  bind_rows(
  E2_Location_res_pair %>%
  dplyr::filter(diff_Exp_LocationE2_IL == 1) %>%
  mutate(lfc_E2_IL = ifelse(diff_Exp_LocationE2_IL == 1, 
                             lfc_Exp_LocationE2_IL, 0)) %>%
            transmute(taxon,
                      lfc = lfc_E2_IL,
                      se = se_Exp_LocationE2_IL,
                      p_val = p_Exp_LocationE2_IL,
                      group = "E2_IL - E2_CEC")
 
    ) %>%                 
    bind_rows(
    E2_Location_res_pair %>%
    dplyr::filter(diff_Exp_LocationE2_IL_Exp_LocationE2_DUOD == 1) %>%
    mutate(lfc_E2_IL_E2_DUOD = ifelse(diff_Exp_LocationE2_IL_Exp_LocationE2_DUOD == 1, 
                                   lfc_Exp_LocationE2_IL_Exp_LocationE2_DUOD, 0)) %>%
                    transmute(taxon,
                      lfc = lfc_E2_IL_E2_DUOD,
                      se = se_Exp_LocationE2_IL_Exp_LocationE2_DUOD,
                      p_val = p_Exp_LocationE2_IL_Exp_LocationE2_DUOD,
                      group = "E2_IL - E2_DUOD")
    )
      df_fig_trend$group = factor(df_fig_trend$group, 
                            levels = c("E2_DUOD - E2_CEC",
                                      "E2_IL - E2_CEC",
                                      "E2_IL - E2_DUOD"))


```

Save the df you created
```{r}
write.csv(df_fig_trend, "/Users/connermertz/Dropbox (vlab)/Conner Mertz/UNM biology/Chapter 1 E1 vs E2/Methods/16S rRNA gene sequencing/Sequencing_workup/ANCOMoutput/codeexports/E2_ps_raw/E2_Location_df_fig_trend.csv") #ADD IN OTHER CATEGORIES YOU WANT TO GRAPH 

#read in your EDITED DF csv: 
E2_Location_df_fig_trend <- read.csv("/Users/connermertz/Dropbox (vlab)/Conner Mertz/UNM biology/Chapter 1 E1 vs E2/Methods/16S rRNA gene sequencing/Sequencing_workup/ANCOMoutput/codeexports/E2_ps_raw/E2_Location_df_fig_trend.csv")
#change name so target_ids match on both files
names(E2_Location_df_fig_trend)[2] ="taxon" 

#read in your TAXA classification file 
Taxonomy_classification <- read.csv("/Users/connermertz/Dropbox (vlab)/Conner Mertz/UNM biology/Chapter 1 E1 vs E2/Methods/16S rRNA gene sequencing/Sequencing_workup/DADA2_output/ALL_taxa.csv") #if this doesn't load be sure to dowload the file from the cloud, then it will work
#change name so target_ids match on both files
names(Taxonomy_classification)[1] ="taxon"

#now we need to get our DF results and join it with our Taxonomy results from CAT (combining two data frames)
#we only want to keep the genes that showed up with gapmind, we don't need the other genes that showed up
Part_DF <- Taxonomy_classification %>% 
  inner_join (E2_Location_df_fig_trend, by = "taxon")

write.csv(Part_DF,"/Users/connermertz/Dropbox (vlab)/Conner Mertz/UNM biology/Chapter 1 E1 vs E2/Methods/16S rRNA gene sequencing/Sequencing_workup/ANCOMoutput/codeexports/E2_ps_raw/E2_Location_DF_FIG_TREND_AND_Taxonomy.csv") # went in here and edited file to my liking. I saved the new version as edit_E2_Location_DF_FIG_TREND_AND_Taxonomy.csv. That is what I will be using for the figures. I basically switched the signs for the differential abundant taxa to be positive for E1 and negative for E2 but consistent with whatever category it was being compared to. 

```

#Now lets look at location pairwise comparisons WITHIN each treatment:
Starting with the Synthetic diet:
E2_duod vs E2_ile
E2_duod vs E2_cec
E2_ile vs E2_cec
```{r}
E2_Location_df <- read.csv("/Users/connermertz/Dropbox (vlab)/Conner Mertz/UNM biology/Chapter 1 E1 vs E2/Methods/16S rRNA gene sequencing/Sequencing_workup/ANCOMoutput/codeexports/E2_ps_raw/edit_E2_Location_DF_FIG_TREND_AND_Taxonomy.csv") %>%
  filter(group ==  "E2_DUOD - E2_IL" |
           group == "E2_IL - E2_DUOD" |
           group == "E2_DUOD - E2_CEC" |
           group == "E2_CEC - E2_DUOD" |
           group == "E2_IL - E2_CEC" |
           group == "E2_CEC - E2_IL" )%>%
 mutate(Differentially_Abundant_Group = factor(Differentially_Abundant_Group, levels = c("E2_DUOD","E2_IL", "E2_CEC"))) %>%
 mutate(Differentially_Abundant_Location = factor(Differentially_Abundant_Location, levels = c("DUOD","IL", "CEC"))) %>%
 mutate(taxon = factor(taxon, levels = c("ASV396", "ASV14", "ASV112", "ASV19", "ASV109", "ASV11", "ASV43", "ASV6", "ASV78", "ASV63", "ASV1", "ASV55", "ASV236", "ASV320", "ASV21", "ASV72","ASV159","ASV56","ASV9", "ASV44","ASV16","ASV37","ASV139","ASV168", "ASV118")))

```


change color
```{r}
#first you want to factor your items. These are the names of the Location in the order I want them to be every time. 
E2_Location_df$Differentially_Abundant_Location <-factor(E2_Location_df$Differentially_Abundant_Location, levels=c("DUOD","IL", "CEC"))

#now create an object that is a list of colors in a certain order.The colors I chose are "color-blind friendly":
LocationScale <- c("#440a48","#47da7a","#dba0ff")
#scales::show_col(LocationScale)

#now we want to merge the two. We are taking the order from the locations and telling it that it needs to match the order from the cbPalette.
names(LocationScale) <- levels(E2_Location_df$Differentially_Abundant_Location)
#Have you ever used the scale_colour_manual command? Now we can combine that command and all of the stuff we want in it into a new object, LocationColors. At the end of your ggplot code, just add a +LocationColors (or whatever you decide to name it), and it should pull the colors you made! 
LocationColors.1 <- scale_colour_manual(name="Location", values=LocationScale)
```



E2 BY LOCATION
```{r}
fig_trend = E2_Location_df %>% 
  ggplot(aes(x = lfc, y = taxon, color = Differentially_Abundant_Location)) + 
  geom_point(size = 3, position = position_dodge()) +
  geom_errorbar(
    aes(xmin = lfc - se, xmax = lfc + se), 
    width = .2) +
  geom_vline(aes(xintercept = 0), linetype = "dashed")+
  facet_wrap(~group, strip.position="bottom") +
  labs(x = "Log Fold Change", y = "ASV", title = "Semi-Natural Diet: Differentially Abundant Taxa  \nin each gut section") +
  LocationColors.1 +
  scale_x_continuous(limits=c(-7, 7)) +
  theme_bw() +
  theme(
    plot.title = element_text(hjust = 0.5),
    title =element_text(size=18, face='bold'),
    axis.text.x = element_text(color="black", size=12, angle = 90),
    axis.text.y = element_text(color="black", size=12),
    strip.text = element_text(size=6),
    strip.text.y = element_text(color="black", size=12, angle=90),
    legend.title = element_text(color="black", size=14),
    legend.text = element_text(color="black", size=12),
    legend.key.size = unit(1, 'cm'),
    legend.position = "right",
    axis.title.x = element_text(color="black", size=12),
    axis.title.y = element_text(color="black", size=12, angle=90),
    panel.grid.major = element_blank(), # switch off major gridlines
    panel.grid.minor = element_blank()# switch off minor gridlines
  )
#CREATE A PDF AND SAVES IT TO THE RIGHT DIRECTORY/FOLDER. 
pdf("/Users/connermertz/Dropbox (vlab)/Conner Mertz/UNM biology/Chapter 1 E1 vs E2/Methods/16S rRNA gene sequencing/Sequencing_workup/ANCOMoutput/E2_ps_raw_Plots/E2_Location_LFC.pdf", height = 6, width = 8)
fig_trend
dev.off()
```

#Differentially abundant ASVs associated with the Semi-Natural Diet by gut sections identified by ANCOMBC2 pairwise comparisons. Each dot represents a differentially abundant ASV plotted by its taxonomic assignment on the y- axis and log-fold change on the x axis. Positive log fold change indicated taxa enriched in the Duodenum (left panel) or Ileum (right pannel), while negative log fold change indicated taxa enriched in the Cecum. Error bars were calculated using the standard error. (standard errors for all differentially abundant taxa identified are below 0.9, and all p-values < 0.002) 

#When we compared gut sections in the synthetic treatment we found:
E2_duod vs E2_ile: No differentially abundant taxa identified in the duodenum vs the ileum, therefore the two gut sections that comprise the small intestine have no statistically significant differences in taxa. 
E2_duod vs E2_cec: in the duodenum ASV118, ASV72, ASV21, ASV55, ASV1, ASV78, ASV396 were found to be differentially abundant compared to the cecum, while ASV168, ASV139, ASV37, ASV16, ASV44, ASV9, ASV56, ASV6, ASV43, ASV11, ASV109, ASV19, ASV112, ASV14. 
E2_ile vs E2_cec: in the ileum ASV118, ASV159, ASV72, ASV21, ASV320, ASV236, ASV55, ASV1, ASV63, ASV78 was found to be differentially abundant compared to the cecum, while ASV139, ASV37, ASV16, ASV44, ASV56, ASV6, ASV43, ASV11, ASV19, ASV112 was found to be differentially abundant in the cecum compared to the ileum. 










lets see if we can't join these together!

```{r}
E1vsE2_LOCATION_df <- read.csv("/Users/connermertz/Dropbox (vlab)/Conner Mertz/UNM biology/Chapter 1 E1 vs E2/Methods/16S rRNA gene sequencing/Sequencing_workup/ANCOMoutput/codeexports/PS_ran_separate_combo/edit_E1_E2_LOCATION_DF.csv") %>%
  filter(group ==  "E1_DUOD - E1_IL" |
           group == "E1_IL - E1_DUOD" |
           group == "E1_DUOD - E1_CEC" |
           group == "E1_CEC - E1_DUOD" |
           group == "E1_IL - E1_CEC" |
           group == "E1_CEC - E1_IL" |
         group ==  "E2_DUOD - E2_IL" |
           group == "E2_IL - E2_DUOD" |
           group == "E2_DUOD - E2_CEC" |
           group == "E2_CEC - E2_DUOD" |
           group == "E2_IL - E2_CEC" |
           group == "E2_CEC - E2_IL")%>%
 mutate(Differentially_Abundant_Group = factor(Differentially_Abundant_Group, levels = c("E1_DUOD","E1_IL", "E1_CEC", "E2_DUOD","E2_IL", "E2_CEC"))) %>%
 mutate(Differentially_Abundant_Location = factor(Differentially_Abundant_Location, levels = c("DUOD","IL", "CEC"))) %>%
 mutate(taxon = factor(taxon, levels = c("ASV396", "ASV8",	"ASV14",	"ASV112", "ASV19", "ASV109", "ASV31", "ASV11", "ASV43", "ASV6", "ASV78", "ASV63", "ASV1", "ASV10", "ASV55", "ASV236", "ASV320", "ASV21", "ASV72", "ASV15", "ASV159", "ASV56", "ASV9", "ASV44", "ASV16", "ASV75", "ASV37", "ASV82", "ASV139", "ASV105", "ASV133", "ASV168", "ASV118")))

```

change color
```{r}
#first you want to factor your items. These are the names of the Location in the order I want them to be every time. 
E1vsE2_LOCATION_df$Differentially_Abundant_Location <-factor(E1vsE2_LOCATION_df$Differentially_Abundant_Location, levels=c("DUOD","IL", "CEC"))

#now create an object that is a list of colors in a certain order.The colors I chose are "color-blind friendly":
LocationScale <- c("#440a48","#47da7a","#dba0ff")
#scales::show_col(LocationScale)

#now we want to merge the two. We are taking the order from the locations and telling it that it needs to match the order from the cbPalette.
names(LocationScale) <- levels(E1vsE2_LOCATION_df$Differentially_Abundant_Location)
#Have you ever used the scale_colour_manual command? Now we can combine that command and all of the stuff we want in it into a new object, LocationColors. At the end of your ggplot code, just add a +LocationColors (or whatever you decide to name it), and it should pull the colors you made! 
LocationColors <- scale_colour_manual(name="Location", values=LocationScale)
```


```{r}
fig_trend = E1vsE2_LOCATION_df %>% 
  ggplot(aes(x = lfc, y = taxon, color = Differentially_Abundant_Location, shape = Differentially_Abundant_Experiment)) + 
  geom_point(size = 4, position = position_dodge()) +
  geom_errorbar(
    aes(xmin = lfc - se, xmax = lfc + se), 
    width = .2) +
  geom_vline(aes(xintercept = 0), linetype = "dashed")+
  facet_wrap(~LocationvsLocation, strip.position="top") +
  labs(x = "Log Fold Change", y = "ASV", title = "Differentially Abundant Taxa") +
  LocationColors +
  scale_x_continuous(limits=c(-7, 7)) +
  theme_bw() +
  theme(
    plot.title = element_text(hjust = 0.5),
    title =element_text(size=18, face='bold'),
    axis.text.x = element_text(color="black", size=12, angle = 90),
    axis.text.y = element_text(color="black", size=12),
    strip.text = element_text(size=6),
    strip.text.y = element_text(color="black", size=12, angle=90),
    legend.title = element_text(color="black", size=14),
    legend.text = element_text(color="black", size=12),
    legend.key.size = unit(1, 'cm'),
    legend.position = "bottom",
    strip.background = element_blank(), #to remove facet wrap title
    axis.title.x = element_text(color="black", size=12),
    axis.title.y = element_text(color="black", size=12, angle=90),
    panel.grid.major = element_blank(), # switch off major gridlines
    panel.grid.minor = element_blank()# switch off minor gridlines
  )
#CREATE A PDF AND SAVES IT TO THE RIGHT DIRECTORY/FOLDER. 
pdf("/Users/connermertz/Dropbox (vlab)/Conner Mertz/UNM biology/Chapter 1 E1 vs E2/Methods/16S rRNA gene sequencing/Sequencing_workup/ANCOMoutput/PlotsforComparison/E1_E2_LocationComps_LFC2.pdf", height = 9, width = 10)
fig_trend
dev.off()
```
#Differentially abundant ASVs associated with the Synthetic (circle) or Semi-Natural Diet (triangle) by gut section identified by ANCOMBC2 pairwise comparisons. Each dot represents a differentially abundant ASV plotted by its taxonomic assignment on the y- axis and log-fold change on the x axis. Positive log fold change indicated taxa enriched in the Duodenum (left pannel) or Ileum (right pannel), while negative log fold change indicates taxa enriched in the Cecum. Error bars were calculated using the standard error. (standard errors for all differentially abundant taxa identified are below 0.9, and all p-values < 0.002) 
#When we compared gut sections in the synthetic treatment we found:
E1_duod vs E1_ile: No differentially abundant taxa identified in the duodenum vs the ileum, therefore the two gut sections that comprise the small intestine have no statistically significant differences in taxa. 
E1_duod vs E1_cec: in the duodenum ASV10 and ASV1 were found to be differentially abundant compared to the cecum, while ASV105, ASV75, ASV16, ASV44, ASV6, ASV31, ASV8. 
E1_ile vs E1_cec: in the ileum ASV15, ASV72, ASV21, ASV10, ASV1, ASV78 was found to be differentially abundant compared to the cecum, while ASV133, ASV105, ASV82, ASV37, ASV16, ASV44, ASV56, ASV6, ASV31 was found to be differentially abundant in the cecum compared to the ileum.
Therefore: ASV1 is present in both small intestinal gut sections, and ASV37, ASV16, ASV444, ASV56, and ASV6, ASV8 are present in the cecum. 
#When we compared gut sections in the semi-natural treatment we found:
E2_duod vs E2_ile: No differentially abundant taxa identified in the duodenum vs the ileum, therefore the two gut sections that comprise the small intestine have no statistically significant differences in taxa. 
E2_duod vs E2_cec: in the duodenum ASV118, ASV72, ASV21, ASV55, ASV1, ASV78, ASV396 were found to be differentially abundant compared to the cecum, while ASV168, ASV139, ASV37, ASV16, ASV44, ASV9, ASV56, ASV6, ASV43, ASV11, ASV109, ASV19, ASV112, ASV14. 
E2_ile vs E2_cec: in the ileum ASV118, ASV159, ASV72, ASV21, ASV320, ASV236, ASV55, ASV1, ASV63, ASV78 was found to be differentially abundant compared to the cecum, while ASV139, ASV37, ASV16, ASV44, ASV56, ASV6, ASV43, ASV11, ASV19, ASV112 was found to be differentially abundant in the cecum compared to the ileum. 




#############DIET#############################


#ANCOMBC2 by PROTEIN- E1 vs E2 for all Diets samples
```{r}
set.seed(123)
output = ancombc2(ps_raw_E1_E2_forcomparison, assay_name = NULL, tax_level = NULL,
                  fix_formula = "Location + Exp_Diet", rand_formula = NULL,
                  p_adj_method = "holm", pseudo = 0, pseudo_sens = TRUE,
                  prv_cut = 0.10, lib_cut = 1000, s0_perc = 0.05,
                  group = "Exp_Diet", struc_zero = TRUE, neg_lb = TRUE,
                  alpha = 0.05, n_cl = 2, verbose = TRUE,
                  global = TRUE, pairwise = TRUE, dunnet = TRUE, trend = TRUE,
                  iter_control = list(tol = 1e-2, max_iter = 20, 
                                      verbose = TRUE),
                  em_control = list(tol = 1e-5, max_iter = 100),
                  lme_control = lme4::lmerControl(),
                  mdfdr_control = list(fwer_ctrl_method = "holm", B = 100),
                  trend_control = list(contrast = list(matrix(c(1, 0, 0, 0, 0,
                                                               -1, 1, 0, 0, 0,
                                                                0, -1, 1, 0, 0, 
                                                                0, 0, -1, 1, 0,
                                                                0, 0, 0, -1, 1),
                                                              nrow = 5, 
                                                              byrow = TRUE),
                                                       matrix(c(1, 0, 0, 0, 0,
                                                               -1, 1, 0, 0, 0,
                                                                0, -1, 1, 0, 0, 
                                                                0, 0, -1, 1, 0,
                                                                0, 0, 0, -1, 1),
                                                              nrow = 5, 
                                                              byrow = TRUE)),
                                       node = list(2, 2),
                                       solver = "ECOS",
                                       B = 100))

```
Save the output
```{r}
res_prim = output$res
saveRDS(res_prim, "/Users/connermertz/Dropbox (vlab)/Conner Mertz/UNM biology/Chapter 1 E1 vs E2/Methods/16S rRNA gene sequencing/Sequencing_workup/ANCOMoutput/codeexports/E1_E2_forcomparison/E1vsE2_PROTEIN_res_prim.rds")
#read in the ANCOMBC output if you want to see E1 vs E2 by PROTEIN
E1vsE2_PROTEIN_res_prim <- readRDS("/Users/connermertz/Dropbox (vlab)/Conner Mertz/UNM biology/Chapter 1 E1 vs E2/Methods/16S rRNA gene sequencing/Sequencing_workup/ANCOMoutput/codeexports/E1_E2_forcomparison/E1vsE2_PROTEIN_res_prim.rds")

res_pair = output$res_pair
saveRDS(res_pair, "/Users/connermertz/Dropbox (vlab)/Conner Mertz/UNM biology/Chapter 1 E1 vs E2/Methods/16S rRNA gene sequencing/Sequencing_workup/ANCOMoutput/codeexports/E1_E2_forcomparison/E1vsE2_PROTEIN_res_pair.rds")
E1vsE2_PROTEIN_res_pair <- readRDS("/Users/connermertz/Dropbox (vlab)/Conner Mertz/UNM biology/Chapter 1 E1 vs E2/Methods/16S rRNA gene sequencing/Sequencing_workup/ANCOMoutput/codeexports/E1_E2_forcomparison/E1vsE2_PROTEIN_res_pair.rds")

res_trend = output$res_trend
saveRDS(res_trend, "/Users/connermertz/Dropbox (vlab)/Conner Mertz/UNM biology/Chapter 1 E1 vs E2/Methods/16S rRNA gene sequencing/Sequencing_workup/ANCOMoutput/codeexports/E1_E2_forcomparison/E1vsE2_PROTEIN_res_trend.rds")
E1vsE2_PROTEIN_res_trend <- readRDS("/Users/connermertz/Dropbox (vlab)/Conner Mertz/UNM biology/Chapter 1 E1 vs E2/Methods/16S rRNA gene sequencing/Sequencing_workup/ANCOMoutput/codeexports/E1_E2_forcomparison/E1vsE2_PROTEIN_res_trend.rds")
```

ANCOM-BC2 primary analysis
```{r}
E1vsE2_PROTEIN_res_prim
```


##ANCOM-BC2 pairwise directional test
ANCOM-BC2 pairwise directional test aims to determine taxa that are differentially abundant between any pair of two groups across three or more experimental groups, while controlling the mdFDR.

In this example, we want to identify taxa that are differentially abundant between any pair of two groups across “lean”, “overweight”, and “obese”. The result contains: 1) log fold changes, 2) standard errors, 3) test statistics, 4) p-values, 5) adjusted p-values, 6) indicators of whether the taxon is differentially abundant (TRUE) or not (FALSE).
```{r}
E1vsE2_PROTEIN_res_pair
write.csv(E1vsE2_PROTEIN_res_pair, "/Users/connermertz/Dropbox (vlab)/Conner Mertz/UNM biology/Chapter 1 E1 vs E2/Methods/16S rRNA gene sequencing/Sequencing_workup/ANCOMoutput/codeexports/E1_E2_forcomparison/E1vsE2_PROTEIN_res_pair_edit.csv")

#read in your res_pair csv: 
E1vsE2_PROTEIN_res_pair2 <- read.csv("/Users/connermertz/Dropbox (vlab)/Conner Mertz/UNM biology/Chapter 1 E1 vs E2/Methods/16S rRNA gene sequencing/Sequencing_workup/ANCOMoutput/codeexports/E1_E2_forcomparison/E1vsE2_PROTEIN_res_pair_edit.csv")
#change name so target_ids match on both files
names(E1vsE2_PROTEIN_res_pair2)[2] ="taxon" 

#read in your RNA classification file 
Taxonomy_classification <- read.csv("/Users/connermertz/Dropbox (vlab)/Conner Mertz/UNM biology/Chapter 1 E1 vs E2/Methods/16S rRNA gene sequencing/Sequencing_workup/DADA2_output/ALL_taxa.csv") #if this doesn't load be sure to dowload the file from the cloud, then it will work
#change name so target_ids match on both files
names(Taxonomy_classification)[1] ="taxon"

#now we need to get our gapmind results and join it with our Taxonomy results from CAT (combining two data frames)
#we only want to keep the genes that showed up with gapmind, we don't need the other genes that showed up
Part_1 <- Taxonomy_classification %>% 
  inner_join (E1vsE2_PROTEIN_res_pair2, by = "taxon")

write.csv(Part_1,"/Users/connermertz/Dropbox (vlab)/Conner Mertz/UNM biology/Chapter 1 E1 vs E2/Methods/16S rRNA gene sequencing/Sequencing_workup/ANCOMoutput/codeexports/E1_E2_forcomparison/E1vsE2_PROTEIN_res_pair_AND_Taxonomy.csv") # went in here and edited file to my liking
```


```{r}
#Plot ANCOM-BC2 pairwise directional test
df_fig_pair = E1vsE2_PROTEIN_res_pair %>%
    filter(diff_Exp_DietE1_2.5P == 1 | 
             diff_Exp_DietE1_5P == 1 |
             diff_Exp_DietE2_10P == 1 |
             diff_Exp_DietE2_2.5P == 1 |
             diff_Exp_DietE2_5P == 1 |
             diff_Exp_DietE1_5P_Exp_DietE1_2.5P == 1 |
             diff_Exp_DietE2_10P_Exp_DietE1_2.5P == 1 |
             diff_Exp_DietE2_2.5P_Exp_DietE1_2.5P == 1 |
             diff_Exp_DietE2_5P_Exp_DietE1_2.5P == 1 |
             diff_Exp_DietE2_10P_Exp_DietE1_5P == 1 |
             diff_Exp_DietE2_2.5P_Exp_DietE1_5P == 1 |
             diff_Exp_DietE2_5P_Exp_DietE1_5P == 1 |
             diff_Exp_DietE2_2.5P_Exp_DietE2_10P == 1 |
             diff_Exp_DietE2_5P_Exp_DietE2_10P == 1 |
             diff_Exp_DietE2_5P_Exp_DietE2_2.5P == 1) %>%
    mutate(lfc_E1_2.5P = ifelse(diff_Exp_DietE1_2.5P == 1, 
                             lfc_Exp_DietE1_2.5P, 0),
           lfc_E1_5P = ifelse(diff_Exp_DietE1_5P == 1, 
                                   lfc_Exp_DietE1_5P, 0),
           lfc_E2_10P = ifelse(diff_Exp_DietE2_10P == 1, 
                                   lfc_Exp_DietE2_10P, 0),
           lfc_E2_2.5P = ifelse(diff_Exp_DietE2_2.5P == 1, 
                                   lfc_Exp_DietE2_2.5P, 0),
           lfc_E2_5P = ifelse(diff_Exp_DietE2_5P == 1, 
                                   lfc_Exp_DietE2_5P, 0),
           lfc_E1_5P_E1_2.5P = ifelse(diff_Exp_DietE1_5P_Exp_DietE1_2.5P == 1, 
                                   lfc_Exp_DietE1_5P_Exp_DietE1_2.5P, 0),
           lfc_E2_10P_E1_2.5P = ifelse(diff_Exp_DietE2_10P_Exp_DietE1_2.5P == 1, 
                                   lfc_Exp_DietE2_10P_Exp_DietE1_2.5P, 0),
           lfc_E2_2.5P_E1_2.5P = ifelse(diff_Exp_DietE2_2.5P_Exp_DietE1_2.5P == 1, 
                                   lfc_Exp_DietE2_2.5P_Exp_DietE1_2.5P, 0),
           lfc_E2_5P_E1_2.5P = ifelse(diff_Exp_DietE2_5P_Exp_DietE1_2.5P == 1, 
                                   lfc_Exp_DietE2_5P_Exp_DietE1_2.5P, 0),
           lfc_E2_10P_E1_5P = ifelse(diff_Exp_DietE2_10P_Exp_DietE1_5P == 1, 
                                   lfc_Exp_DietE2_10P_Exp_DietE1_5P, 0),
           lfc_E2_2.5P_E1_5P = ifelse(diff_Exp_DietE2_2.5P_Exp_DietE1_5P == 1, 
                                   lfc_Exp_DietE2_2.5P_Exp_DietE1_5P, 0),
           lfc_E2_5P_E1_5P = ifelse(diff_Exp_DietE2_5P_Exp_DietE1_5P == 1, 
                                   lfc_Exp_DietE2_5P_Exp_DietE1_5P, 0),
           lfc_E2_2.5P_E2_10P = ifelse(diff_Exp_DietE2_2.5P_Exp_DietE2_10P == 1, 
                                   lfc_Exp_DietE2_2.5P_Exp_DietE2_10P, 0),
           lfc_E2_5P_E2_10P = ifelse(diff_Exp_DietE2_5P_Exp_DietE2_10P == 1, 
                                   lfc_Exp_DietE2_5P_Exp_DietE2_10P, 0),
           lfc_E2_5P_E2_2.5P = ifelse(diff_Exp_DietE2_5P_Exp_DietE2_2.5P == 1, 
                                        lfc_Exp_DietE2_5P_Exp_DietE2_2.5P, 0)) %>%
    transmute(taxon, 
              `E1_2.5P vs. E1_10P` = round(lfc_E1_2.5P, 2), 
              `E1_5P vs. E1_10P` = round(lfc_E1_5P, 2),
              `E2_10P vs. E1_10P` = round(lfc_E2_10P, 2),
              `E2_2.5P vs. E1_10P` = round(lfc_E2_2.5P, 2),
              `E2_5P vs. E1_10P` = round(lfc_E2_5P, 2),
              `E1_5P vs. E1_2.5P` = round(lfc_E1_5P_E1_2.5P, 2),
              `E2_10P vs. E1_2.5P` = round(lfc_E2_10P_E1_2.5P, 2),
              `E2_2.5P vs. E1_2.5P` = round(lfc_E2_2.5P_E1_2.5P, 2),
              `E2_5P vs. E1_2.5P` = round(lfc_E2_5P_E1_2.5P, 2),
              `E2_10P vs. E1_5P` = round(lfc_E2_10P_E1_5P, 2),
              `E2_2.5P vs. E1_5P` = round(lfc_E2_2.5P_E1_5P, 2),
              `E2_5P vs. E1_5P` = round(lfc_E2_5P_E1_5P, 2),
              `E2_2.5P vs. E2_10P` = round(lfc_E2_2.5P_E2_10P, 2),
              `E2_5P vs. E2_10P` = round(lfc_E2_5P_E2_10P, 2),
              `E2_5P vs. E2_2.5P` = round(lfc_E2_5P_E2_2.5P, 2)
              ) %>%
    pivot_longer(cols = `E1_2.5P vs. E1_10P`:`E2_5P vs. E2_2.5P`, 
                 names_to = "group", values_to = "value") %>%
    arrange(taxon)
df_fig_pair$group = factor(df_fig_pair$group, 
                           levels = c("E1_2.5P vs. E1_10P",
                                      "E1_5P vs. E1_10P",
                                      "E2_10P vs. E1_10P",
                                      "E2_2.5P vs. E1_10P",
                                      "E2_5P vs. E1_10P",
                                      "E1_5P vs. E1_2.5P",
                                      "E2_10P vs. E1_2.5P",
                                      "E2_2.5P vs. E1_2.5P",
                                      "E2_5P vs. E1_2.5P",
                                      "E2_10P vs. E1_5P",
                                      "E2_2.5P vs. E1_5P",
                                      "E2_5P vs. E1_5P",
                                      "E2_2.5P vs. E2_10P",
                                      "E2_5P vs. E2_10P",
                                      "E2_5P vs. E2_2.5P"))
  
lo = floor(min(df_fig_pair$value))
up = ceiling(max(df_fig_pair$value))
mid = (lo + up)/2
fig_pair = df_fig_pair %>%
  ggplot(aes(x = group, y = taxon, fill = value)) + 
  geom_tile(color = "black") +
  scale_fill_gradient2(low = "blue", high = "red", mid = "white", 
                       na.value = "white", midpoint = mid, limit = c(lo, up),
                       name = NULL) +
  geom_text(aes(group, taxon, label = value), color = "black", size = 4) +
  labs(x = NULL, y = NULL, title = "Log fold change of pairwise comparisons in E1 vs E2 by % Protein") +
  theme_bw() +
  theme(
    plot.title = element_text(hjust = 1),
    title =element_text(size=22, face='bold'),
    axis.text.x = element_text(color="black", size=12, angle = 90, hjust = 1),
    axis.text.y = element_text(color="black", size=12),
    strip.text.y = element_text(color="black", size=12, angle=90),
    #legend.title = element_text(color="black", size=20),
    legend.text = element_text(color="black", size=14),
    legend.key.size = unit(1, 'cm'),
    #legend.position = "right",
    strip.text = element_text(size = 16),
    axis.title.x = element_text(color="black", size=14),
    axis.title.y = element_text(color="black", size=14, angle=90),
    panel.grid.major = element_blank(), # switch off major gridlines
    panel.grid.minor = element_blank(), # switch off minor gridlines
  )
#CREATE A PDF AND SAVES IT TO THE RIGHT DIRECTORY/FOLDER. 
pdf("/Users/connermertz/Dropbox (vlab)/Conner Mertz/UNM biology/Chapter 1 E1 vs E2/Methods/16S rRNA gene sequencing/Sequencing_workup/ANCOMoutput/Plots/E1vsE2_Protein_LFC_Pairwise.pdf", height = 12, width = 14)
fig_pair
dev.off()
```


#E1_E2_PROTEIN
ANCOMBC2
pairwise comparisons by experiment and PROTEIN
```{r}
#create your data frame for log fold change of pairwise comparisons for dot plot! We will use this to graph dot plots of bargraphs
df_res_pair = E1vsE2_PROTEIN_res_pair %>%
  dplyr::filter(diff_Exp_DietE1_2.5P == 1) %>%
  mutate(lfc_E1_2.5P = ifelse(diff_Exp_DietE1_2.5P == 1, 
                             lfc_Exp_DietE1_2.5P, 0)) %>%
              transmute(taxon,
              lfc = lfc_E1_2.5P,
              se = se_Exp_DietE1_2.5P,
              p_val= p_Exp_DietE1_2.5P,
              group = "E1_2.5P - E1_10P") %>%
              
  bind_rows(
  E1vsE2_PROTEIN_res_pair %>%
  dplyr::filter(diff_Exp_DietE1_5P == 1) %>%
  mutate(lfc_E1_5P = ifelse(diff_Exp_DietE1_5P == 1, 
                             lfc_Exp_DietE1_5P, 0)) %>%
            transmute(taxon,
                      lfc = lfc_E1_5P,
                      se = se_Exp_DietE1_5P,
                      p_val = p_Exp_DietE1_5P,
                      group = "E1_5P - E1_10P")
  ) %>%
    bind_rows(
    E1vsE2_PROTEIN_res_pair %>%
    dplyr::filter(diff_Exp_DietE2_10P == 1) %>%
    mutate(lfc_E2_10P = ifelse(diff_Exp_DietE2_10P == 1, 
                                   lfc_Exp_DietE2_10P, 0)) %>%
            transmute(taxon,
                      lfc = lfc_E2_10P,
                      se = se_Exp_DietE2_10P,
                      p_val= p_Exp_DietE2_10P,
                      group = "E2_10P - E1_10P")
    ) %>%
   bind_rows(
    E1vsE2_PROTEIN_res_pair %>%
    dplyr::filter(diff_Exp_DietE2_2.5P == 1) %>%
    mutate(lfc_E2_2.5P = ifelse(diff_Exp_DietE2_2.5P == 1, 
                                   lfc_Exp_DietE2_2.5P, 0)) %>%
                    transmute(taxon,
                      lfc = lfc_E2_2.5P,
                      se = se_Exp_DietE2_2.5P,
                      p_val = p_Exp_DietE2_2.5P,
                      group = "E2_2.5P - E1_10P")
    ) %>%  
   bind_rows(
    E1vsE2_PROTEIN_res_pair %>%
    dplyr::filter(diff_Exp_DietE2_5P == 1) %>%
    mutate(lfc_E2_5P = ifelse(diff_Exp_DietE2_5P == 1, 
                                   lfc_Exp_DietE2_5P, 0)) %>%  
                    transmute(taxon,
                      lfc = lfc_E2_5P,
                      se = se_Exp_DietE2_5P,
                      p_val = p_Exp_DietE2_5P,
                      group = "E2_5P - E1_10P") 
    ) %>%                 
    bind_rows(
    E1vsE2_PROTEIN_res_pair %>%
    dplyr::filter(diff_Exp_DietE1_5P_Exp_DietE1_2.5P == 1) %>%
    mutate(lfc_E1_5P_E1_2.5P = ifelse(diff_Exp_DietE1_5P_Exp_DietE1_2.5P == 1, 
                                   lfc_Exp_DietE1_5P_Exp_DietE1_2.5P, 0)) %>%
                    transmute(taxon,
                      lfc = lfc_E1_5P_E1_2.5P,
                      se = se_Exp_DietE1_5P_Exp_DietE1_2.5P,
                      p_val = p_Exp_DietE1_5P_Exp_DietE1_2.5P,
                      group = "E1_5P - E1_2.5P")
    ) %>%              
    bind_rows(
    E1vsE2_PROTEIN_res_pair %>%
    dplyr::filter(diff_Exp_DietE2_10P_Exp_DietE1_2.5P == 1) %>%
    mutate(lfc_E2_10P_E1_2.5P = ifelse(diff_Exp_DietE2_10P_Exp_DietE1_2.5P == 1, 
                                   lfc_Exp_DietE2_10P_Exp_DietE1_2.5P, 0)) %>%
                    transmute(taxon,
                      lfc = lfc_E2_10P_E1_2.5P,
                      se = se_Exp_DietE2_10P_Exp_DietE1_2.5P,
                      p_val = p_Exp_DietE2_10P_Exp_DietE1_2.5P,
                      group = "E2_10P - E1_2.5P") 
    ) %>%              
    bind_rows(
    E1vsE2_PROTEIN_res_pair %>%
    dplyr::filter(diff_Exp_DietE2_2.5P_Exp_DietE1_2.5P == 1) %>%
    mutate(lfc_E2_2.5P_E1_2.5P = ifelse(diff_Exp_DietE2_2.5P_Exp_DietE1_2.5P == 1, 
                                   lfc_Exp_DietE2_2.5P_Exp_DietE1_2.5P, 0)) %>%
                    transmute(taxon,
                      lfc = lfc_E2_2.5P_E1_2.5P,
                      se = se_Exp_DietE2_2.5P_Exp_DietE1_2.5P,
                      p_val = p_Exp_DietE2_2.5P_Exp_DietE1_2.5P,
                      group = "E2_2.5P - E1_2.5P")
    ) %>%
    bind_rows(
    E1vsE2_PROTEIN_res_pair %>%
    dplyr::filter(diff_Exp_DietE2_5P_Exp_DietE1_2.5P == 1) %>%
    mutate(lfc_E2_5P_E1_2.5P = ifelse(diff_Exp_DietE2_5P_Exp_DietE1_2.5P == 1, 
                                   lfc_Exp_DietE2_5P_Exp_DietE1_2.5P, 0)) %>%
                    transmute(taxon,
                      lfc = lfc_E2_5P_E1_2.5P,
                      se = se_Exp_DietE2_5P_Exp_DietE1_2.5P,
                      p_val = p_Exp_DietE2_5P_Exp_DietE1_2.5P,
                      group = "E2_5P - E1_2.5P")
    ) %>%
    bind_rows(
    E1vsE2_PROTEIN_res_pair %>%
    dplyr::filter(diff_Exp_DietE2_10P_Exp_DietE1_5P == 1) %>%
    mutate(lfc_E2_10P_E1_5P = ifelse(diff_Exp_DietE2_10P_Exp_DietE1_5P == 1, 
                                   lfc_Exp_DietE2_10P_Exp_DietE1_5P, 0)) %>%
                transmute(taxon,
                      lfc = lfc_E2_10P_E1_5P,
                      se = se_Exp_DietE2_10P_Exp_DietE1_5P,
                      p_val = p_Exp_DietE2_10P_Exp_DietE1_5P,
                      group = "E2_10P - E1_5P")
    ) %>%
 bind_rows(
    E1vsE2_PROTEIN_res_pair %>%
    dplyr::filter(diff_Exp_DietE2_2.5P_Exp_DietE1_5P == 1) %>%
    mutate(lfc_E2_2.5P_E1_5P = ifelse(diff_Exp_DietE2_2.5P_Exp_DietE1_5P == 1, 
                                   lfc_Exp_DietE2_2.5P_Exp_DietE1_5P, 0)) %>% 
                    transmute(taxon,
                      lfc = lfc_E2_2.5P_E1_5P,
                      se = se_Exp_DietE2_2.5P_Exp_DietE1_5P,
                      p_val = p_Exp_DietE2_2.5P_Exp_DietE1_5P,
                      group = "E2_2.5P - E1_5P")
    ) %>%
bind_rows(
    E1vsE2_PROTEIN_res_pair %>%
    dplyr::filter(diff_Exp_DietE2_5P_Exp_DietE1_5P == 1) %>%
    mutate(lfc_E2_5P_E1_5P = ifelse(diff_Exp_DietE2_5P_Exp_DietE1_5P == 1, 
                                   lfc_Exp_DietE2_5P_Exp_DietE1_5P, 0)) %>% 
                    transmute(taxon,
                      lfc = lfc_E2_5P_E1_5P,
                      se = se_Exp_DietE2_5P_Exp_DietE1_5P,
                      p_val = p_Exp_DietE2_5P_Exp_DietE1_5P,
                      group = "E2_5P - E1_5P")
    ) %>%    
bind_rows(
    E1vsE2_PROTEIN_res_pair %>%
    dplyr::filter(diff_Exp_DietE2_2.5P_Exp_DietE2_10P == 1) %>%
    mutate(lfc_E2_2.5P_E2_10P = ifelse(diff_Exp_DietE2_2.5P_Exp_DietE2_10P == 1, 
                                   lfc_Exp_DietE2_2.5P_Exp_DietE2_10P, 0)) %>% 
                    transmute(taxon,
                      lfc = lfc_E2_2.5P_E2_10P,
                      se = se_Exp_DietE2_2.5P_Exp_DietE2_10P,
                      p_val= p_Exp_DietE2_2.5P_Exp_DietE2_10P,
                      group = "E2_2.5P - E2_10P")
) %>%
bind_rows(
    E1vsE2_PROTEIN_res_pair %>%
    dplyr::filter(diff_Exp_DietE2_5P_Exp_DietE2_10P == 1) %>%
    mutate(lfc_E2_5P_E2_10P = ifelse(diff_Exp_DietE2_5P_Exp_DietE2_10P == 1, 
                                   lfc_Exp_DietE2_5P_Exp_DietE2_10P, 0)) %>%
                    transmute(taxon,
                      lfc = lfc_E2_5P_E2_10P,
                      se = se_Exp_DietE2_5P_Exp_DietE2_10P,
                      p_val = p_Exp_DietE2_5P_Exp_DietE2_10P,
                      group = "E2_5P - E2_10P")
    ) %>%  
bind_rows(
    E1vsE2_PROTEIN_res_pair %>%
    dplyr::filter(diff_Exp_DietE2_5P_Exp_DietE2_2.5P == 1) %>%
    mutate(lfc_E2_5P_E2_2.5P = ifelse(diff_Exp_DietE2_5P_Exp_DietE2_2.5P == 1, 
                                        lfc_Exp_DietE2_5P_Exp_DietE2_2.5P, 0)) %>%
                    transmute(taxon,
                      lfc = lfc_E2_5P_E2_2.5P,
                      se = se_Exp_DietE2_5P_Exp_DietE2_2.5P,
                      p_val = p_Exp_DietE2_5P_Exp_DietE2_2.5P,
                      group = "E2_5P - E2_2.5P")
    )
      df_fig_trend$group = factor(df_fig_trend$group, 
                            levels = c("E1_2.5P - E1_10P",
                                      "E1_5P - E1_10P",
                                      "E2_10P - E1_10P",
                                      "E2_2.5P - E1_10P",
                                      "E2_5P - E1_10P",
                                      "E1_5P - E1_2.5P",
                                      "E2_10P - E1_2.5P",
                                      "E2_2.5P - E1_2.5P",
                                      "E2_5P - E1_2.5P",
                                      "E2_10P - E1_5P",
                                      "E2_2.5P - E1_5P",
                                      "E2_5P - E1_5P",
                                      "E2_2.5P - E2_10P",
                                      "E2_5P - E2_10P",
                                      "E2_5P - E2_2.5P"))


```


Save the df you created
```{r}
write.csv(df_res_pair, "/Users/connermertz/Dropbox (vlab)/Conner Mertz/UNM biology/Chapter 1 E1 vs E2/Methods/16S rRNA gene sequencing/Sequencing_workup/ANCOMoutput/codeexports/E1_E2_forcomparison/E1vsE2_PROTEIN_df_res_pair.csv") #ADD IN OTHER CATEGORIES YOU WANT TO GRAPH 

#read in your EDITED DF csv: 
E1vsE2_PROTEIN_df_res_pair <- read.csv("/Users/connermertz/Dropbox (vlab)/Conner Mertz/UNM biology/Chapter 1 E1 vs E2/Methods/16S rRNA gene sequencing/Sequencing_workup/ANCOMoutput/codeexports/E1_E2_forcomparison/E1vsE2_PROTEIN_df_res_pair.csv")
#change name so target_ids match on both files
names(E1vsE2_PROTEIN_df_res_pair)[2] ="taxon" 

#read in your TAXA classification file 
Taxonomy_classification <- read.csv("/Users/connermertz/Dropbox (vlab)/Conner Mertz/UNM biology/Chapter 1 E1 vs E2/Methods/16S rRNA gene sequencing/Sequencing_workup/DADA2_output/ALL_taxa.csv") #if this doesn't load be sure to dowload the file from the cloud, then it will work
#change name so target_ids match on both files
names(Taxonomy_classification)[1] ="taxon"

#now we need to get our DF results and join it with our Taxonomy results from CAT (combining two data frames)
#we only want to keep the genes that showed up with gapmind, we don't need the other genes that showed up
Part_DF <- Taxonomy_classification %>% 
  inner_join (E1vsE2_PROTEIN_df_res_pair, by = "taxon")

write.csv(Part_DF,"/Users/connermertz/Dropbox (vlab)/Conner Mertz/UNM biology/Chapter 1 E1 vs E2/Methods/16S rRNA gene sequencing/Sequencing_workup/ANCOMoutput/codeexports/E1_E2_forcomparison/E1vsE2_PROTEIN_DF_RES_PAIR_AND_Taxonomy.csv") # went in here and edited file to my liking. I saved the new version as edit_E1vsE2_PROTEIN_DF_RES_PAIR_AND_Taxonomy.csv. That is what I will be using for the figures. I basically switched the signs for the differential abundant taxa to be positive for E1 and negative for E2 but consistent with whatever category it was being compared to. 

```


Just keep it simple 
#trying to compare just by experiment and location so:
#E1_2.5P vs E2_2.5P
#E1_5P vs E2_5P
#E1_10P vs E2_10P

#read in the edited df
```{r}
E1vsE2_Protein_df <- read.csv("/Users/connermertz/Dropbox (vlab)/Conner Mertz/UNM biology/Chapter 1 E1 vs E2/Methods/16S rRNA gene sequencing/Sequencing_workup/ANCOMoutput/codeexports/E1_E2_forcomparison/edit_E1vsE2_PROTEIN_DF_RES_PAIR_AND_Taxonomy.csv")
```

#filter out to just include: 
#E1_2.5P vs E2_2.5P
#E1_5P vs E2_5P
#E1_10P vs E2_10P

AND

#filter to keep comparisons of interest and mutate so that location and ASV's are in the correct order

```{r}
E1vsE2_PROTEIN_df1 <- read.csv("/Users/connermertz/Dropbox (vlab)/Conner Mertz/UNM biology/Chapter 1 E1 vs E2/Methods/16S rRNA gene sequencing/Sequencing_workup/ANCOMoutput/codeexports/E1_E2_forcomparison/edit_E1vsE2_PROTEIN_DF_RES_PAIR_AND_Taxonomy.csv") %>%
  filter(group_in_E1vsE2_order ==  "E1_2.5PvsE2_2.5P" |
           group_in_E1vsE2_order ==  "E1_5PvsE2_5P" |
           group_in_E1vsE2_order ==  "E1_10PvsE2_10P" )%>%
 mutate(Differentially_Abundant_Group = factor(Differentially_Abundant_Group, levels = c("E1_2.5P","E1_5P", "E1_10P", "E2_2.5P", "E2_5P", "E2_10P"))) %>%
 mutate(Differentially_Abundant_Protein = factor(Differentially_Abundant_Protein, levels = c("2.5P","5P", "10P"))) %>%
 mutate(taxon = factor(taxon, levels = c("ASV8", "ASV29", "ASV27", "ASV80", "ASV142", "ASV39", "ASV1", "ASV170")))

#save df
write.csv(E1vsE2_PROTEIN_df1,"/Users/connermertz/Dropbox (vlab)/Conner Mertz/UNM biology/Chapter 1 E1 vs E2/Methods/16S rRNA gene sequencing/Sequencing_workup/ANCOMoutput/codeexports/E1_E2_forcomparison/E1vsE2_Protein_only.csv")

```

Change for Low, Med, High Protein Diet
```{r}
#first you want to factor your items. These are the names of the Location in the order I want them to be every time. 
E1vsE2_PROTEIN_df1$Differentially_Abundant_Protein <-factor(E1vsE2_PROTEIN_df1$Differentially_Abundant_Protein, levels = c("2.5P","5P", "10P"))

#now create an object that is a list of colors in a certain order.The colors I chose are "color-blind friendly":
Protein_Scale <- c("#cb146e", "#f7ce45","#1983fc")
#scales::show_col(DietScale)

#now we want to merge the two. We are taking the order from the locations and telling it that it needs to match the order from the cbPalette.
names(Protein_Scale) <- levels(E1vsE2_PROTEIN_df1$Differentially_Protein)
#Have you ever used the scale_colour_manual command? Now we can combine that command and all of the stuff we want in it into a new object, LocationColors. At the end of your ggplot code, just add a +DietColor (or whatever you decide to name it), and it should pull the colors you made! 
ProteinColors<- scale_colour_manual(name="Protein", values= Protein_Scale)
```

Plot it!
```{r}
fig_trend = E1vsE2_PROTEIN_df1 %>% 
  ggplot(aes(x = lfc_E1vsE2, y = taxon, color = Differentially_Abundant_Protein)) + 
  geom_point(size = 4, position = position_dodge()) +
  geom_errorbar(
    aes(xmin = lfc_E1vsE2 - se, xmax = lfc_E1vsE2 + se), 
    width = .2) +
  geom_vline(aes(xintercept = 0), linetype = "dashed")+
  #facet_wrap(~Differentially_Abundant_Location) +
  labs(x = "Log Fold Change", y = "ASV", title = "Differentially Abundant Taxa\n by Diet Treatment ") +
  ProteinColors +
  scale_x_continuous(limits=c(-7, 7)) +
  theme_bw() +
  theme(
    plot.title = element_text(hjust = 0.5),
    title =element_text(size=18, face='bold'),
    axis.text.x = element_text(color="black", size=12, angle = 90),
    axis.text.y = element_text(color="black", size=12),
    strip.text = element_text(size=6),
    strip.text.y = element_text(color="black", size=12, angle=90),
    legend.title = element_text(color="black", size=14),
    legend.text = element_text(color="black", size=12),
    legend.key.size = unit(1, 'cm'),
    legend.position = "bottom",
    axis.title.x = element_text(color="black", size=12),
    axis.title.y = element_text(color="black", size=12, angle=90),
    panel.grid.major = element_blank(), # switch off major gridlines
    panel.grid.minor = element_blank()# switch off minor gridlines
  )
#CREATE A PDF AND SAVES IT TO THE RIGHT DIRECTORY/FOLDER. 
pdf("/Users/connermertz/Dropbox (vlab)/Conner Mertz/UNM biology/Chapter 1 E1 vs E2/Methods/16S rRNA gene sequencing/Sequencing_workup/ANCOMoutput/Plots/E1vsE2_Protein_LFC.pdf", height = 4, width = 8)
fig_trend
dev.off()
```
#results: 

#Differentially abundant ASVs associated with the Synthetic or Semi-natural diet identified by ANCOMBC2 pairwise comparisons. Each dot represents a differentially abundant ASV (determined via pair-wise comparisons) plotted by its taxonomic assignment on the y- axis and log-fold change on the x axis. Positive log fold change indicated taxa enriched in the synthetic diet, while negative log fold change taxa enriched in the semi-natural diet.Error bars were calculated using the standard error. (standard errors for all differentially abundant taxa identified are below 1, and all p-values < 0.0001) 

#When we compared the synthetic vs the seminatural by % Protein we found:
E1_2.5P vs E2_2.5P: in the low protein diet treatments (2.5P) only ASV8 was found to be differentially abundant in the synthetic diet compared to the semi-natural diet.  
E1_5P vs E2_5P: in the medium diet treatment (5P), only ASV39 was found to be differentially abundant in the semi-natural diet compared to the synthetic diet. 
E1_10P vs E2_10P: in the high protein diet (10P), ASV80 was differentially abundant in the synthetic diet while ASV1 and ASV29 were found to be differentially abundant in the semi-natural diet. 


#Now lets look at protein pairwise comparisons WITHIN each treatment:
Starting with the Synthetic diet:
E1_2.5P vs E1_5P
E1_2.5P vs E1_10P
E1_5P vs E1_10P

```{r}
set.seed(123)
output = ancombc2(ps_raw_E1_forcomparison, assay_name = NULL, tax_level = NULL,
                  fix_formula = "Location + Exp_Diet", rand_formula = NULL,
                  p_adj_method = "holm", pseudo = 0, pseudo_sens = TRUE,
                  prv_cut = 0.10, lib_cut = 1000, s0_perc = 0.05,
                  group = "Exp_Diet", struc_zero = TRUE, neg_lb = TRUE,
                  alpha = 0.05, n_cl = 2, verbose = TRUE,
                  global = TRUE, pairwise = TRUE, dunnet = TRUE, trend = TRUE,
                  iter_control = list(tol = 1e-2, max_iter = 20, 
                                      verbose = TRUE),
                  em_control = list(tol = 1e-5, max_iter = 100),
                  lme_control = lme4::lmerControl(),
                  mdfdr_control = list(fwer_ctrl_method = "holm", B = 100),
                  trend_control = list(contrast = list(matrix(c(1, 0, -1, 1),
                                                              nrow = 2, 
                                                              byrow = TRUE),
                                                       matrix(c(-1, 0, 1, -1),
                                                              nrow = 2, 
                                                              byrow = TRUE)),
                                       node = list(2, 2),
                                       solver = "ECOS",
                                       B = 100))
```
Save the output
```{r}
res_prim = output$res
saveRDS(res_prim, "/Users/connermertz/Dropbox (vlab)/Conner Mertz/UNM biology/Chapter 1 E1 vs E2/Methods/16S rRNA gene sequencing/Sequencing_workup/ANCOMoutput/codeexports/E1_forcomparison/E1_PROTEIN_res_prim.rds")
#read in the ANCOMBC output if you want to see E1 vs E2 by PROTEIN
E1_PROTEIN_res_prim <- readRDS("/Users/connermertz/Dropbox (vlab)/Conner Mertz/UNM biology/Chapter 1 E1 vs E2/Methods/16S rRNA gene sequencing/Sequencing_workup/ANCOMoutput/codeexports/E1_forcomparison/E1_PROTEIN_res_prim.rds")

res_pair = output$res_pair
saveRDS(res_pair, "/Users/connermertz/Dropbox (vlab)/Conner Mertz/UNM biology/Chapter 1 E1 vs E2/Methods/16S rRNA gene sequencing/Sequencing_workup/ANCOMoutput/codeexports/E1_forcomparison/E1_PROTEIN_res_pair.rds")
E1_PROTEIN_res_pair <- readRDS("/Users/connermertz/Dropbox (vlab)/Conner Mertz/UNM biology/Chapter 1 E1 vs E2/Methods/16S rRNA gene sequencing/Sequencing_workup/ANCOMoutput/codeexports/E1_forcomparison/E1_PROTEIN_res_pair.rds")

res_trend = output$res_trend
saveRDS(res_trend, "/Users/connermertz/Dropbox (vlab)/Conner Mertz/UNM biology/Chapter 1 E1 vs E2/Methods/16S rRNA gene sequencing/Sequencing_workup/ANCOMoutput/codeexports/E1_forcomparison/E1_PROTEIN_res_trend.rds")
E1_PROTEIN_res_trend <- readRDS("/Users/connermertz/Dropbox (vlab)/Conner Mertz/UNM biology/Chapter 1 E1 vs E2/Methods/16S rRNA gene sequencing/Sequencing_workup/ANCOMoutput/codeexports/E1_forcomparison/E1_PROTEIN_res_trend.rds")
```

##ANCOM-BC2 pairwise directional test
ANCOM-BC2 pairwise directional test aims to determine taxa that are differentially abundant between any pair of two groups across three or more experimental groups, while controlling the mdFDR.

In this example, we want to identify taxa that are differentially abundant between any pair of two groups across “lean”, “overweight”, and “obese”. The result contains: 1) log fold changes, 2) standard errors, 3) test statistics, 4) p-values, 5) adjusted p-values, 6) indicators of whether the taxon is differentially abundant (TRUE) or not (FALSE).
```{r}
E1_PROTEIN_res_pair
write.csv(E1_PROTEIN_res_pair, "/Users/connermertz/Dropbox (vlab)/Conner Mertz/UNM biology/Chapter 1 E1 vs E2/Methods/16S rRNA gene sequencing/Sequencing_workup/ANCOMoutput/codeexports/E1_forcomparison/E1_PROTEIN_res_pair_edit.csv")

#read in your res_pair csv: 
E1_PROTEIN_res_pair2 <- read.csv("/Users/connermertz/Dropbox (vlab)/Conner Mertz/UNM biology/Chapter 1 E1 vs E2/Methods/16S rRNA gene sequencing/Sequencing_workup/ANCOMoutput/codeexports/E1_forcomparison/E1_PROTEIN_res_pair_edit.csv")
#change name so target_ids match on both files
names(E1_PROTEIN_res_pair2)[2] ="taxon" 

#read in your RNA classification file 
Taxonomy_classification <- read.csv("/Users/connermertz/Dropbox (vlab)/Conner Mertz/UNM biology/Chapter 1 E1 vs E2/Methods/16S rRNA gene sequencing/Sequencing_workup/DADA2_output/ALL_taxa.csv") #if this doesn't load be sure to dowload the file from the cloud, then it will work
#change name so target_ids match on both files
names(Taxonomy_classification)[1] ="taxon"

#now we need to get our gapmind results and join it with our Taxonomy results from CAT (combining two data frames)
#we only want to keep the genes that showed up with gapmind, we don't need the other genes that showed up
Part_1 <- Taxonomy_classification %>% 
  inner_join (E1_PROTEIN_res_pair2, by = "taxon")

write.csv(Part_1,"/Users/connermertz/Dropbox (vlab)/Conner Mertz/UNM biology/Chapter 1 E1 vs E2/Methods/16S rRNA gene sequencing/Sequencing_workup/ANCOMoutput/codeexports/E1_forcomparison/E1_PROTEIN_res_pair_AND_Taxonomy.csv") # went in here and edited file to my liking
```

#E1_PROTEIN
ANCOMBC2
pairwise comparisons by experiment and PROTEIN
```{r}
#create your data frame for log fold change of pairwise comparisons for dot plot! We will use this to graph dot plots of bargraphs
df_res_pair = E1_PROTEIN_res_pair %>%
  dplyr::filter(diff_Exp_DietE1_2.5P == 1) %>%
  mutate(lfc_E1_2.5P = ifelse(diff_Exp_DietE1_2.5P == 1, 
                             lfc_Exp_DietE1_2.5P, 0)) %>%
              transmute(taxon,
              lfc = lfc_E1_2.5P,
              se = se_Exp_DietE1_2.5P,
              p_val= p_Exp_DietE1_2.5P,
              group = "E1_2.5P - E1_10P") %>%
              
  bind_rows(
  E1_PROTEIN_res_pair %>%
  dplyr::filter(diff_Exp_DietE1_5P == 1) %>%
  mutate(lfc_E1_5P = ifelse(diff_Exp_DietE1_5P == 1, 
                             lfc_Exp_DietE1_5P, 0)) %>%
            transmute(taxon,
                      lfc = lfc_E1_5P,
                      se = se_Exp_DietE1_5P,
                      p_val = p_Exp_DietE1_5P,
                      group = "E1_5P - E1_10P")
      ) %>%                 
    bind_rows(
    E1_PROTEIN_res_pair %>%
    dplyr::filter(diff_Exp_DietE1_5P_Exp_DietE1_2.5P == 1) %>%
    mutate(lfc_E1_5P_E1_2.5P = ifelse(diff_Exp_DietE1_5P_Exp_DietE1_2.5P == 1, 
                                   lfc_Exp_DietE1_5P_Exp_DietE1_2.5P, 0)) %>%
                    transmute(taxon,
                      lfc = lfc_E1_5P_E1_2.5P,
                      se = se_Exp_DietE1_5P_Exp_DietE1_2.5P,
                      p_val = p_Exp_DietE1_5P_Exp_DietE1_2.5P,
                      group = "E1_5P - E1_2.5P")
    )
      df_fig_trend$group = factor(df_fig_trend$group, 
                            levels = c("E1_2.5P - E1_10P",
                                      "E1_5P - E1_10P",
                                      "E1_5P - E1_2.5P"))


```


Save the df you created
```{r}
write.csv(df_res_pair, "/Users/connermertz/Dropbox (vlab)/Conner Mertz/UNM biology/Chapter 1 E1 vs E2/Methods/16S rRNA gene sequencing/Sequencing_workup/ANCOMoutput/codeexports/E1_forcomparison/E1_PROTEIN_df_res_pair.csv") #ADD IN OTHER CATEGORIES YOU WANT TO GRAPH 

#read in your EDITED DF csv: 
E1_PROTEIN_df_res_pair <- read.csv("/Users/connermertz/Dropbox (vlab)/Conner Mertz/UNM biology/Chapter 1 E1 vs E2/Methods/16S rRNA gene sequencing/Sequencing_workup/ANCOMoutput/codeexports/E1_forcomparison/E1_PROTEIN_df_res_pair.csv")
#change name so target_ids match on both files
names(E1_PROTEIN_df_res_pair)[2] ="taxon" 

#read in your TAXA classification file 
Taxonomy_classification <- read.csv("/Users/connermertz/Dropbox (vlab)/Conner Mertz/UNM biology/Chapter 1 E1 vs E2/Methods/16S rRNA gene sequencing/Sequencing_workup/DADA2_output/ALL_taxa.csv") #if this doesn't load be sure to dowload the file from the cloud, then it will work
#change name so target_ids match on both files
names(Taxonomy_classification)[1] ="taxon"

#now we need to get our DF results and join it with our Taxonomy results from CAT (combining two data frames)
#we only want to keep the genes that showed up with gapmind, we don't need the other genes that showed up
Part_DF <- Taxonomy_classification %>% 
  inner_join (E1_PROTEIN_df_res_pair, by = "taxon")

write.csv(Part_DF,"/Users/connermertz/Dropbox (vlab)/Conner Mertz/UNM biology/Chapter 1 E1 vs E2/Methods/16S rRNA gene sequencing/Sequencing_workup/ANCOMoutput/codeexports/E1_forcomparison/E1_PROTEIN_DF_RES_PAIR_AND_Taxonomy.csv") # went in here and edited file to my liking. I saved the new version as edit_E1_PROTEIN_DF_RES_PAIR_AND_Taxonomy.csv. That is what I will be using for the figures. 

```

#Now lets look at protein pairwise comparisons WITHIN each treatment:
Starting with the Synthetic diet:
E1_2.5P vs E1_5P
E1_2.5P vs E1_10P
E1_5P vs E1_10P
```{r}
E1_PROTEIN_df <- read.csv("/Users/connermertz/Dropbox (vlab)/Conner Mertz/UNM biology/Chapter 1 E1 vs E2/Methods/16S rRNA gene sequencing/Sequencing_workup/ANCOMoutput/codeexports/E1_forcomparison/edit_E1_PROTEIN_DF_RES_PAIR_AND_Taxonomy.csv") %>%
  filter(group ==  "E1_2.5P - E1_5P" |
           group == "E1_5P - E1_2.5P" |
           group == "E1_2.5P - E1_10P" |
           group == "E1_10P - E1_2.5P" |
           group == "E1_5P - E1_10P" |
           group == "E1_10P - E1_5P" )%>%
 mutate(Differentially_Abundant_Group = factor(Differentially_Abundant_Group, levels = c("E1_2.5P","E1_5P", "E1_10P"))) %>%
 mutate(Differentially_Abundant_Protein = factor(Differentially_Abundant_Protein, levels = c("2.5P","5P", "10P"))) %>%
 mutate(taxon = factor(taxon, levels = c("ASV80", "ASV1", "ASV62")))

```


change color
```{r}
#first you want to factor your items. These are the names of the Location in the order I want them to be every time. 
E1_PROTEIN_df$Differentially_Abundant_Protein <-factor(E1_PROTEIN_df$Differentially_Abundant_Protein, levels = c("2.5P","5P", "10P"))

#now create an object that is a list of colors in a certain order.The colors I chose are "color-blind friendly":
Protein_Scale <- c("#cb146e", "#f7ce45","#1983fc")
#scales::show_col(DietScale)

#now we want to merge the two. We are taking the order from the locations and telling it that it needs to match the order from the cbPalette.
names(Protein_Scale) <- levels(E1_PROTEIN_df$Differentially_Protein)
#Have you ever used the scale_colour_manual command? Now we can combine that command and all of the stuff we want in it into a new object, LocationColors. At the end of your ggplot code, just add a +DietColor (or whatever you decide to name it), and it should pull the colors you made! 
ProteinColors.1<- scale_colour_manual(name="Protein", values= Protein_Scale)
```



E1 by Protein
```{r}
fig_trend = E1_PROTEIN_df %>% 
  ggplot(aes(x = lfc, y = taxon, color = Differentially_Abundant_Protein, Shape = Differentially_Abundant_Experiment)) + 
  geom_point(size = 3, position = position_dodge()) +
  geom_errorbar(
    aes(xmin = lfc - se, xmax = lfc + se), 
    width = .2) +
  geom_vline(aes(xintercept = 0), linetype = "dashed")+
  facet_wrap(~group, strip.position="bottom") +
  labs(x = "Log Fold Change", y = "ASV", title = "Synthetic Diet: Differentially Abundant Taxa  \n by Percent Protein") +
  ProteinColors.1 +
  scale_x_continuous(limits=c(-7, 7)) +
  theme_bw() +
  theme(
    plot.title = element_text(hjust = 0.5),
    title =element_text(size=18, face='bold'),
    axis.text.x = element_text(color="black", size=12, angle = 90),
    axis.text.y = element_text(color="black", size=12),
    strip.text = element_text(size=6),
    strip.text.y = element_text(color="black", size=12, angle=90),
    legend.title = element_text(color="black", size=14),
    legend.text = element_text(color="black", size=12),
    legend.key.size = unit(1, 'cm'),
    legend.position = "right",
    axis.title.x = element_text(color="black", size=12),
    axis.title.y = element_text(color="black", size=12, angle=90),
    panel.grid.major = element_blank(), # switch off major gridlines
    panel.grid.minor = element_blank()# switch off minor gridlines
  )
#CREATE A PDF AND SAVES IT TO THE RIGHT DIRECTORY/FOLDER. 
pdf("/Users/connermertz/Dropbox (vlab)/Conner Mertz/UNM biology/Chapter 1 E1 vs E2/Methods/16S rRNA gene sequencing/Sequencing_workup/ANCOMoutput/E1_forcomparison_Plots/E1_PROTEIN_LFC.pdf", height = 6, width = 8)
fig_trend
dev.off()
```
#Differentially abundant ASVs associated with the Synthetic Diet by Percent Protein identified by ANCOMBC2 pairwise comparisons. Each dot represents a differentially abundant ASV plotted by its taxonomic assignment on the y- axis and log-fold change on the x axis. Positive log fold change indicated taxa enriched in the low protein diet (left panel) or Medium protein diet (right pannel), while negative log fold change indicated taxa enriched in the high protein diet. Error bars were calculated using the standard error. (standard errors for all differentially abundant taxa identified are below 1, and all p-values < 0.0001) 

#When we compared differentially Abundant taxa by percent protein in the synthetic treatment we found:
E1_2.5P vs E1_5P: ASV62 differentially abundant in the low protein diet compared to the medium protein diet in the synthetic treatment. 
E1_2.5P vs E1_10P: ASV1 and ASV 62 are differnentially abundant in the low protein diet and ASV80 is differentially abundant in the high protein diet when comparing the high to low protein diets in the synthetic treatment. 
E1_5P vs E1_10P: only ASV1 is differentially abundant in the medium protein diet compared to the high protein diet in the synthetic treatment.




#Now lets look at protein pairwise comparisons WITHIN each treatment:
Now the Semi-natural diet:
E2_2.5P vs E2_5P
E2_2.5P vs E2_10P
E2_5P vs E2_10P
```{r}
set.seed(123)
output = ancombc2(E2_ps_raw, assay_name = NULL, tax_level = NULL,
                  fix_formula = "Location + Exp_Diet", rand_formula = NULL,
                  p_adj_method = "holm", pseudo = 0, pseudo_sens = TRUE,
                  prv_cut = 0.10, lib_cut = 1000, s0_perc = 0.05,
                  group = "Exp_Diet", struc_zero = TRUE, neg_lb = TRUE,
                  alpha = 0.05, n_cl = 2, verbose = TRUE,
                  global = TRUE, pairwise = TRUE, dunnet = TRUE, trend = TRUE,
                  iter_control = list(tol = 1e-2, max_iter = 20, 
                                      verbose = TRUE),
                  em_control = list(tol = 1e-5, max_iter = 100),
                  lme_control = lme4::lmerControl(),
                  mdfdr_control = list(fwer_ctrl_method = "holm", B = 100),
                  trend_control = list(contrast = list(matrix(c(1, 0, -1, 1),
                                                              nrow = 2, 
                                                              byrow = TRUE),
                                                       matrix(c(-1, 0, 1, -1),
                                                              nrow = 2, 
                                                              byrow = TRUE)),
                                       node = list(2, 2),
                                       solver = "ECOS",
                                       B = 100))
```
Save the output
```{r}
res_prim = output$res
saveRDS(res_prim, "/Users/connermertz/Dropbox (vlab)/Conner Mertz/UNM biology/Chapter 1 E1 vs E2/Methods/16S rRNA gene sequencing/Sequencing_workup/ANCOMoutput/codeexports/E2_ps_raw/E2_PROTEIN_res_prim.rds")
#read in the ANCOMBC output if you want to see E1 vs E2 by PROTEIN
E2_PROTEIN_res_prim <- readRDS("/Users/connermertz/Dropbox (vlab)/Conner Mertz/UNM biology/Chapter 1 E1 vs E2/Methods/16S rRNA gene sequencing/Sequencing_workup/ANCOMoutput/codeexports/E2_ps_raw/E2_PROTEIN_res_prim.rds")

res_pair = output$res_pair
saveRDS(res_pair, "/Users/connermertz/Dropbox (vlab)/Conner Mertz/UNM biology/Chapter 1 E1 vs E2/Methods/16S rRNA gene sequencing/Sequencing_workup/ANCOMoutput/codeexports/E2_ps_raw/E2_PROTEIN_res_pair.rds")
E2_PROTEIN_res_pair <- readRDS("/Users/connermertz/Dropbox (vlab)/Conner Mertz/UNM biology/Chapter 1 E1 vs E2/Methods/16S rRNA gene sequencing/Sequencing_workup/ANCOMoutput/codeexports/E2_ps_raw/E2_PROTEIN_res_pair.rds")

res_trend = output$res_trend
saveRDS(res_trend, "/Users/connermertz/Dropbox (vlab)/Conner Mertz/UNM biology/Chapter 1 E1 vs E2/Methods/16S rRNA gene sequencing/Sequencing_workup/ANCOMoutput/codeexports/E2_ps_raw/E2_PROTEIN_res_trend.rds")
E2_PROTEIN_res_trend <- readRDS("/Users/connermertz/Dropbox (vlab)/Conner Mertz/UNM biology/Chapter 1 E1 vs E2/Methods/16S rRNA gene sequencing/Sequencing_workup/ANCOMoutput/codeexports/E2_ps_raw/E2_PROTEIN_res_trend.rds")
```

##ANCOM-BC2 pairwise directional test
ANCOM-BC2 pairwise directional test aims to determine taxa that are differentially abundant between any pair of two groups across three or more experimental groups, while controlling the mdFDR.

In this example, we want to identify taxa that are differentially abundant between any pair of two groups across “lean”, “overweight”, and “obese”. The result contains: 1) log fold changes, 2) standard errors, 3) test statistics, 4) p-values, 5) adjusted p-values, 6) indicators of whether the taxon is differentially abundant (TRUE) or not (FALSE).
```{r}
E2_PROTEIN_res_pair
write.csv(E2_PROTEIN_res_pair, "/Users/connermertz/Dropbox (vlab)/Conner Mertz/UNM biology/Chapter 1 E1 vs E2/Methods/16S rRNA gene sequencing/Sequencing_workup/ANCOMoutput/codeexports/E2_ps_raw/E2_PROTEIN_res_pair_edit.csv")

#read in your res_pair csv: 
E2_PROTEIN_res_pair2 <- read.csv("/Users/connermertz/Dropbox (vlab)/Conner Mertz/UNM biology/Chapter 1 E1 vs E2/Methods/16S rRNA gene sequencing/Sequencing_workup/ANCOMoutput/codeexports/E2_ps_raw/E2_PROTEIN_res_pair_edit.csv")
#change name so target_ids match on both files
names(E2_PROTEIN_res_pair2)[2] ="taxon" 

#read in your RNA classification file 
Taxonomy_classification <- read.csv("/Users/connermertz/Dropbox (vlab)/Conner Mertz/UNM biology/Chapter 1 E1 vs E2/Methods/16S rRNA gene sequencing/Sequencing_workup/DADA2_output/ALL_taxa.csv") #if this doesn't load be sure to dowload the file from the cloud, then it will work
#change name so target_ids match on both files
names(Taxonomy_classification)[1] ="taxon"

#now we need to get our gapmind results and join it with our Taxonomy results from CAT (combining two data frames)
#we only want to keep the genes that showed up with gapmind, we don't need the other genes that showed up
Part_1 <- Taxonomy_classification %>% 
  inner_join (E2_PROTEIN_res_pair2, by = "taxon")

write.csv(Part_1,"/Users/connermertz/Dropbox (vlab)/Conner Mertz/UNM biology/Chapter 1 E1 vs E2/Methods/16S rRNA gene sequencing/Sequencing_workup/ANCOMoutput/codeexports/E2_ps_raw/E2_PROTEIN_res_pair_AND_Taxonomy.csv") # went in here and edited file to my liking
```


#E2_PROTEIN
ANCOMBC2
pairwise comparisons by experiment and PROTEIN
```{r}
#create your data frame for log fold change of pairwise comparisons for dot plot! We will use this to graph dot plots of bargraphs
df_fig_trend = E2_PROTEIN_res_pair %>%
  dplyr::filter(diff_Exp_DietE2_2.5P == 1) %>%
  mutate(lfc_E2_2.5P = ifelse(diff_Exp_DietE2_2.5P == 1, 
                             lfc_Exp_DietE2_2.5P, 0)) %>%
              transmute(taxon,
              lfc = lfc_E2_2.5P,
              se = se_Exp_DietE2_2.5P,
              p_val= p_Exp_DietE2_2.5P,
              group = "E2_2.5P - E2_10P") %>%
              
  bind_rows(
  E2_PROTEIN_res_pair %>%
  dplyr::filter(diff_Exp_DietE2_5P == 1) %>%
  mutate(lfc_E2_5P = ifelse(diff_Exp_DietE2_5P == 1, 
                             lfc_Exp_DietE2_5P, 0)) %>%
            transmute(taxon,
                      lfc = lfc_E2_5P,
                      se = se_Exp_DietE2_5P,
                      p_val = p_Exp_DietE2_5P,
                      group = "E2_5P - E2_10P")
      ) %>%                 
    bind_rows(
    E2_PROTEIN_res_pair %>%
    dplyr::filter(diff_Exp_DietE2_5P_Exp_DietE2_2.5P == 1) %>%
    mutate(lfc_E2_5P_E2_2.5P = ifelse(diff_Exp_DietE2_5P_Exp_DietE2_2.5P == 1, 
                                   lfc_Exp_DietE2_5P_Exp_DietE2_2.5P, 0)) %>%
                    transmute(taxon,
                      lfc = lfc_E2_5P_E2_2.5P,
                      se = se_Exp_DietE2_5P_Exp_DietE2_2.5P,
                      p_val = p_Exp_DietE2_5P_Exp_DietE2_2.5P,
                      group = "E2_5P - E2_2.5P")
    )
      df_fig_trend$group = factor(df_fig_trend$group, 
                            levels = c("E2_2.5P - E2_10P",
                                      "E2_5P - E2_10P",
                                      "E2_5P - E2_2.5P"))


```


Save the df you created
```{r}
write.csv(df_fig_trend, "/Users/connermertz/Dropbox (vlab)/Conner Mertz/UNM biology/Chapter 1 E1 vs E2/Methods/16S rRNA gene sequencing/Sequencing_workup/ANCOMoutput/codeexports/E2_ps_raw/E2_PROTEIN_df_res_pair.csv") #ADD IN OTHER CATEGORIES YOU WANT TO GRAPH 

#read in your EDITED DF csv: 
E2_PROTEIN_df_res_pair <- read.csv("/Users/connermertz/Dropbox (vlab)/Conner Mertz/UNM biology/Chapter 1 E1 vs E2/Methods/16S rRNA gene sequencing/Sequencing_workup/ANCOMoutput/codeexports/E2_ps_raw/E2_PROTEIN_df_res_pair.csv")
#change name so target_ids match on both files
names(E2_PROTEIN_df_res_pair)[2] ="taxon" 

#read in your TAXA classification file 
Taxonomy_classification <- read.csv("/Users/connermertz/Dropbox (vlab)/Conner Mertz/UNM biology/Chapter 1 E1 vs E2/Methods/16S rRNA gene sequencing/Sequencing_workup/DADA2_output/ALL_taxa.csv") #if this doesn't load be sure to dowload the file from the cloud, then it will work
#change name so target_ids match on both files
names(Taxonomy_classification)[1] ="taxon"

#now we need to get our DF results and join it with our Taxonomy results from CAT (combining two data frames)
#we only want to keep the genes that showed up with gapmind, we don't need the other genes that showed up
Part_DF <- Taxonomy_classification %>% 
  inner_join (E2_PROTEIN_df_res_pair, by = "taxon")

write.csv(Part_DF,"/Users/connermertz/Dropbox (vlab)/Conner Mertz/UNM biology/Chapter 1 E1 vs E2/Methods/16S rRNA gene sequencing/Sequencing_workup/ANCOMoutput/codeexports/E2_ps_raw/E2_PROTEIN_DF_RES_PAIR_AND_Taxonomy.csv") # went in here and edited file to my liking. I saved the new version as edit_E2_PROTEIN_DF_RES_PAIR_AND_Taxonomy.csv. That is what I will be using for the figures. 

```

#Now lets look at protein pairwise comparisons WITHIN each treatment:
Starting with the Semi-Natural diet:
E2_2.5P vs E2_5P
E2_2.5P vs E2_10P
E2_5P vs E2_10P
```{r}
E2_PROTEIN_df <- read.csv("/Users/connermertz/Dropbox (vlab)/Conner Mertz/UNM biology/Chapter 1 E1 vs E2/Methods/16S rRNA gene sequencing/Sequencing_workup/ANCOMoutput/codeexports/E2_ps_raw/edit_E2_PROTEIN_DF_RES_PAIR_AND_Taxonomy.csv") %>%
  filter(group ==  "E2_2.5P - E2_5P" |
           group == "E2_5P - E2_2.5P" |
           group == "E2_2.5P - E2_10P" |
           group == "E2_10P - E2_2.5P" |
           group == "E2_5P - E2_10P" |
           group == "E2_10P - E2_5P" )%>%
 mutate(Differentially_Abundant_Group = factor(Differentially_Abundant_Group, levels = c("E2_2.5P","E2_5P", "E2_10P"))) %>%
 mutate(Differentially_Abundant_Protein = factor(Differentially_Abundant_Protein, levels = c("2.5P","5P", "10P"))) %>%
 mutate(taxon = factor(taxon, levels = c("ASV17", "ASV5")))

```


change color
```{r}
#first you want to factor your items. These are the names of the Location in the order I want them to be every time. 
E2_PROTEIN_df$Differentially_Abundant_Protein <-factor(E2_PROTEIN_df$Differentially_Abundant_Protein, levels = c("2.5P","5P", "10P"))

#now create an object that is a list of colors in a certain order.The colors I chose are "color-blind friendly":
Protein_Scale <- c("#cb146e", "#f7ce45","#1983fc")
#scales::show_col(DietScale)

#now we want to merge the two. We are taking the order from the locations and telling it that it needs to match the order from the cbPalette.
names(Protein_Scale) <- levels(E2_PROTEIN_df$Differentially_Protein)
#Have you ever used the scale_colour_manual command? Now we can combine that command and all of the stuff we want in it into a new object, LocationColors. At the end of your ggplot code, just add a +DietColor (or whatever you decide to name it), and it should pull the colors you made! 
ProteinColors.1<- scale_colour_manual(name="Protein", values= Protein_Scale)
```



E2 by Protein
```{r}
fig_trend = E2_PROTEIN_df %>% 
  ggplot(aes(x = lfc, y = taxon, color = Differentially_Abundant_Protein, Shape = Differentially_Abundant_Experiment)) + 
  geom_point(size = 3, position = position_dodge()) +
  geom_errorbar(
    aes(xmin = lfc - se, xmax = lfc + se), 
    width = .2) +
  geom_vline(aes(xintercept = 0), linetype = "dashed")+
  facet_wrap(~group, strip.position="bottom") +
  labs(x = "Log Fold Change", y = "ASV", title = "Semi-Natural Diet: Differentially Abundant Taxa  \n by Percent Protein") +
  ProteinColors.1 +
  scale_x_continuous(limits=c(-7, 7)) +
  theme_bw() +
  theme(
    plot.title = element_text(hjust = 0.5),
    title =element_text(size=18, face='bold'),
    axis.text.x = element_text(color="black", size=12, angle = 90),
    axis.text.y = element_text(color="black", size=12),
    strip.text = element_text(size=6),
    strip.text.y = element_text(color="black", size=12, angle=90),
    legend.title = element_text(color="black", size=14),
    legend.text = element_text(color="black", size=12),
    legend.key.size = unit(1, 'cm'),
    legend.position = "right",
    axis.title.x = element_text(color="black", size=12),
    axis.title.y = element_text(color="black", size=12, angle=90),
    panel.grid.major = element_blank(), # switch off major gridlines
    panel.grid.minor = element_blank()# switch off minor gridlines
  )
#CREATE A PDF AND SAVES IT TO THE RIGHT DIRECTORY/FOLDER. 
pdf("/Users/connermertz/Dropbox (vlab)/Conner Mertz/UNM biology/Chapter 1 E1 vs E2/Methods/16S rRNA gene sequencing/Sequencing_workup/ANCOMoutput/E2_ps_raw_Plots/E2_PROTEIN_LFC.pdf", height = 6, width = 8)
fig_trend
dev.off()
```
#Differentially abundant ASVs associated with the Semi-Natural Diet by Percent Protein identified by ANCOMBC2 pairwise comparisons. Each dot represents a differentially abundant ASV plotted by its taxonomic assignment on the y- axis and log-fold change on the x axis. Positive log fold change indicated taxa enriched in the low protein diet (left panel) or Medium protein diet (right pannel), while negative log fold change indicated taxa enriched in the high protein diet. Error bars were calculated using the standard error. (standard errors for all differentially abundant taxa identified are below 1, and all p-values < 0.000001) 

#When we compared differentially Abundant taxa by percent protein in the Semi-Natural treatment we found:
E2_2.5P vs E2_5P: ASV5 is differentially abundant in the low protein diet compared to the medium protein diet in the Semi-Natural treatment. 
E2_2.5P vs E2_10P: ASV5 and ASV17 are differnentially abundant in the low protein diet compared to the high protein diet in the Semi-Natural treatment. 
E2_5P vs E2_10P: no ASVs were found to be differentially abundant when comparing the medium and high protein diets in the Semi-Natural treatment.




lets see if we can't join E1 and E2 Protein results together!
```{r}
E1vsE2_Protein_df <- read.csv("/Users/connermertz/Dropbox (vlab)/Conner Mertz/UNM biology/Chapter 1 E1 vs E2/Methods/16S rRNA gene sequencing/Sequencing_workup/ANCOMoutput/codeexports/PS_ran_separate_combo/edit_E1_E2_PROTEIN_DF.csv") %>%
  filter(group ==  "E1_2.5P - E1_5P" |
           group == "E1_5P - E1_2.5P" |
           group == "E1_2.5P - E1_10P" |
           group == "E1_10P - E1_2.5P" |
           group == "E1_5P - E1_10P" |
           group == "E1_10P - E1_5P" |
         group ==  "E2_2.5P - E2_5P" |
           group == "E2_5P - E2_2.5P" |
           group == "E2_2.5P - E2_10P" |
           group == "E2_10P - E2_2.5P" |
           group == "E2_5P - E2_10P" |
           group == "E2_10P - E2_5P")%>%
 mutate(Differentially_Abundant_Group = factor(Differentially_Abundant_Group, levels = c("E1_2.5P","E1_5P", "E1_10P", "E2_2.5P","E2_5P", "E2_10P"))) %>%
 mutate(Differentially_Abundant_Protein = factor(Differentially_Abundant_Protein, levels = c("2.5P","5P", "10P"))) %>%
 mutate(taxon = factor(taxon, levels = c("ASV80", "ASV17", "ASV5", "ASV1", "ASV62")))

```

Set up colors
```{r}
#first you want to factor your items. These are the names of the Location in the order I want them to be every time. 
E1vsE2_Protein_df$Differentially_Abundant_Protein <-factor(E1vsE2_Protein_df$Differentially_Abundant_Protein, levels = c("2.5P","5P", "10P"))

#now create an object that is a list of colors in a certain order.The colors I chose are "color-blind friendly":
Protein_Scale <- c("#cb146e", "#f7ce45","#1983fc")
#scales::show_col(DietScale)

#now we want to merge the two. We are taking the order from the locations and telling it that it needs to match the order from the cbPalette.
names(Protein_Scale) <- levels(E1vsE2_Protein_df$Differentially_Protein)
#Have you ever used the scale_colour_manual command? Now we can combine that command and all of the stuff we want in it into a new object, LocationColors. At the end of your ggplot code, just add a +DietColor (or whatever you decide to name it), and it should pull the colors you made! 
ProteinColors.3<- scale_colour_manual(name="Protein", values= Protein_Scale)
```



```{r}
fig_trend = E1vsE2_Protein_df %>% 
  ggplot(aes(x = lfc, y = taxon, color = Differentially_Abundant_Protein, shape = Differentially_Abundant_Experiment)) + 
  geom_point(size = 4, position = position_dodge()) +
  geom_errorbar(
    aes(xmin = lfc - se, xmax = lfc + se), 
    width = .2) +
  geom_vline(aes(xintercept = 0), linetype = "dashed")+
  facet_wrap(~PercentProteinvsPercentProtein, strip.position="top") +
  labs(x = "Log Fold Change", y = "ASV", title = "Differentially Abundant Taxa") +
  ProteinColors.3 +
  scale_x_continuous(limits=c(-7, 7)) +
  theme_bw() +
  theme(
    plot.title = element_text(hjust = 0.5),
    title =element_text(size=18, face='bold'),
    axis.text.x = element_text(color="black", size=12, angle = 90),
    axis.text.y = element_text(color="black", size=12),
    strip.text = element_text(size=6),
    strip.text.y = element_text(color="black", size=12, angle=90),
    legend.title = element_text(color="black", size=14),
    legend.text = element_text(color="black", size=12),
    legend.key.size = unit(1, 'cm'),
    legend.position = "bottom",
     strip.background = element_blank(), #to remove facet wrap title
    axis.title.x = element_text(color="black", size=12),
    axis.title.y = element_text(color="black", size=12, angle=90),
    panel.grid.major = element_blank(), # switch off major gridlines
    panel.grid.minor = element_blank()# switch off minor gridlines
  )
#CREATE A PDF AND SAVES IT TO THE RIGHT DIRECTORY/FOLDER. 
pdf("/Users/connermertz/Dropbox (vlab)/Conner Mertz/UNM biology/Chapter 1 E1 vs E2/Methods/16S rRNA gene sequencing/Sequencing_workup/ANCOMoutput/PlotsforComparison/E1_E2_ProteinComps_LFC.pdf", height = 3, width = 10)
fig_trend
dev.off()
```


#Differentially abundant ASVs associated with the Synthetic and Semi-Natural Diet by Percent Protein identified by ANCOMBC2 pairwise comparisons. Each dot represents a differentially abundant ASV plotted by its taxonomic assignment on the y- axis and log-fold change on the x axis. Each point represents an ASV differentially abundant in each pairwaise diet comparison. Error bars were calculated using the standard error. (standard errors for all differentially abundant taxa identified are below 1, and all p-values < 0.0001) 

#When we compared differentially Abundant taxa by percent protein in the synthetic treatment we found:
E1_2.5P vs E1_5P: ASV62 is differentially abundant in the low protein diet compared to the medium protein diet in the synthetic treatment. 
E1_2.5P vs E1_10P: ASV1 and ASV 62 are differnentially abundant in the low protein diet and ASV80 is differentially abundant in the high protein diet when comparing the high to low protein diets in the synthetic treatment. 
E1_5P vs E1_10P: only ASV1 is differentially abundant in the medium protein diet compared to the high protein diet in the synthetic treatment. 
#When we compared differentially Abundant taxa by percent protein in the semi-natural treatment we found:
E2_2.5P vs E2_5P: ASV5 is differentially abundant in the low protein diet compared to the medium protein diet in the Semi-Natural treatment. 
E2_2.5P vs E2_10P: ASV5 and ASV17 are differnentially abundant in the low protein diet compared to the high protein diet in the Semi-Natural treatment. 
E2_5P vs E2_10P: no ASVs were found to be differentially abundant when comparing the medium and high protein diets in the Semi-Natural treatment.


























































#########messing around code

#bar graphs
```{r}
E1vsE2_LOCATION_df_fig <- read.csv("/Users/connermertz/Documents/Sequence Analysis/E1_E2_All_Samples/codeexports/ANCOMBC/E1vsE2_LOCATION_DF_FIG_TREND_AND_Taxonomy.csv")

fig_trend = E1vsE2_LOCATION_df_fig %>% 
  ggplot(aes(x = group, y = lfc, fill = group)) + 
  geom_bar(stat = "identity", position = position_dodge(), color = "black") +
  geom_errorbar(aes(ymin = lfc - se, ymax = lfc + se), width = .2,
                position = position_dodge(.9)) +
  facet_wrap(vars(taxon), nrow = 2, scales = "free") +
  labs(x = NULL, y = NULL, title = "Log fold change Pairwise") +
  scale_fill_manual(values= c("#6d83da",
"#9bb240",
"#5a3a8e",
"#63c471",
"#c872c5",
"#5f8d3d",
"#8d2c66",
"#45c097",
"#dc5359",
"#d49c3b",
"#d76092",
"#a1843c",
"#b34656",
"#c96234",
"#9d472c")) +
  theme_bw() +
  theme(
    plot.title = element_text(hjust = 1),
    title =element_text(size=22, face='bold'),
    axis.text.x = element_text(color="black", size=12, angle = 90, hjust = 1),
    axis.text.y = element_text(color="black", size=12),
    strip.text.y = element_text(color="black", size=12, angle=90),
    #legend.title = element_text(color="black", size=20),
    legend.text = element_text(color="black", size=14),
    legend.key.size = unit(1, 'cm'),
    #legend.position = "right",
    strip.text = element_text(size = 16),
    axis.title.x = element_text(color="black", size=14),
    axis.title.y = element_text(color="black", size=14, angle=90),
    panel.grid.major = element_blank(), # switch off major gridlines
    panel.grid.minor = element_blank(), # switch off minor gridlines
  )
#CREATE A PDF AND SAVES IT TO THE RIGHT DIRECTORY/FOLDER. 
pdf("/Users/connermertz/Documents/Conner Documents/UNM biology/Chapter 1 E1 vs E2/ANCOMoutput/E1vsE2_Location_Pairwise_bar.fig.WITHMYCSV.pdf", height = 12, width = 22)
fig_trend
dev.off()
```

E1vsE2 by Location
#Differentially abundant ASVs associated with the Synthetic or Semi-natural diet identified by ANCOMBC2 pairwise comparisons. Each dot represents a differentially abundant ASV (determined via pair-wise comparisons) plotted by its taxonomic assignment on the y- axis and log-fold change on the x axis. Positive log fold change indicated taxa enriched in the synthetic diet, while negative log fold change taxa enriched in the semi-natural diet.  
```{r}
fig_trend = E1vsE2_LOCATION_df_fig %>% 
  ggplot(aes(x = lfc_sign_changed_if.needed, y = taxon, color = Differentially_Abundant_Group)) + 
  geom_point(size = 3, position=position_dodge(0.9)) +
  #geom_errorbar(
  #  aes(xmin = lfc_sign_changed_if.needed - se, xmax = lfc_sign_changed_if.needed + se), 
  #  width = .2,
  #  position=position_jitterdodge()) +
  geom_vline(aes(xintercept = 0), linetype = "dashed")+
  facet_wrap(~Differentially_Abundant_Location) +
  labs(x = "Log Fold Change", y = "ASV", title = "Differentially Abundant Taxa of \n Pairwise Comparisons across Locations") +
  #scale_color_manual(values = phyColor) +
  theme_bw() +
  theme(
    plot.title = element_text(hjust = 0.5),
    title =element_text(size=18, face='bold'),
    axis.text.x = element_text(color="black", size=12, angle = 90, hjust = 1),
    axis.text.y = element_text(color="black", size=12),
    strip.text = element_text(size=6),
    strip.text.y = element_text(color="black", size=12, angle=90),
    legend.title = element_text(color="black", size=14),
    legend.text = element_text(color="black", size=12),
    legend.key.size = unit(1, 'cm'),
    legend.position = "right",
    axis.title.x = element_text(color="black", size=12),
    axis.title.y = element_text(color="black", size=12, angle=90),
    panel.grid.major = element_blank(), # switch off major gridlines
    panel.grid.minor = element_blank(), # switch off minor gridlines
  )
fig_trend
#CREATE A PDF AND SAVES IT TO THE RIGHT DIRECTORY/FOLDER. 
#pdf("/Users/connermertz/Documents/Conner Documents/UNM biology/Chapter 1 E1 vs E2/ANCOMoutput/E1vsE2_Location_Pairwise_dot.withcomparisons_fig.pdf", height = 6, width = 20)
#fig_trend
#dev.off()
```


#try same but scatter
```{r}
fig_trend = E1vsE2_LOCATION_df_fig %>% 
  ggplot(aes(x = lfc, y = taxon, color = Phylum)) + 
  geom_point(stat = "identity", position = position_dodge()) +
  geom_errorbar(aes(xmin = lfc - se, xmax = lfc + se), width = .2,
                position = position_dodge(.9)) +
  facet_wrap(Differentially_Abundant_Experiment~ Differentially_Abundant_Location+Differentially_Abundant_Group+group, ncol=8) +
  labs(x = "Log Fold Change", y = "ASV", title = "Differentially Abundant Taxa of \n Pairwise Comparisons across Locations") +
  scale_color_manual(values = phyColor) +
  theme_bw() +
  theme(
    plot.title = element_text(hjust = 0.5),
    title =element_text(size=18, face='bold'),
    axis.text.x = element_text(color="black", size=12, angle = 90, hjust = 1),
    axis.text.y = element_text(color="black", size=12),
    strip.text = element_text(size=6),
    strip.text.y = element_text(color="black", size=12, angle=90),
    legend.title = element_text(color="black", size=14),
    legend.text = element_text(color="black", size=12),
    legend.key.size = unit(1, 'cm'),
    legend.position = "right",
    axis.title.x = element_text(color="black", size=12),
    axis.title.y = element_text(color="black", size=12, angle=90),
    panel.grid.major = element_blank(), # switch off major gridlines
    panel.grid.minor = element_blank(), # switch off minor gridlines
  )
fig_trend
#CREATE A PDF AND SAVES IT TO THE RIGHT DIRECTORY/FOLDER. 
pdf("/Users/connermertz/Documents/Conner Documents/UNM biology/Chapter 1 E1 vs E2/ANCOMoutput/E1vsE2_Location_Pairwise_dot.withcomparisons_fig.pdf", height = 6, width = 20)
fig_trend
dev.off()
```

```{r}
fig_trend = E1vsE2_LOCATION_df_fig %>% 
  ggplot(aes(x = lfc, y = taxon, fill = Phylum)) + 
  geom_bar(stat = "identity", position = position_dodge(), color = "black") +
  geom_errorbar(aes(xmin = lfc - se, xmax = lfc + se), width = .2,
                position = position_dodge(.9)) +
  facet_wrap(Differentially_Abundant_Experiment~ Differentially_Abundant_Location+Differentially_Abundant_Group+ group, nrow =1) +
  labs(x = "Log Fold Change", y = "ASV", title = "Differentially Abundant Taxa of \n Pairwise Comparisons across Locations") +
  scale_fill_manual(values = phyColor) +
  theme_bw() +
  theme(
    plot.title = element_text(hjust = 0.5),
    title =element_text(size=16, face='bold'),
    axis.text.x = element_text(color="black", size=12, angle = 90, hjust = 1),
    axis.text.y = element_text(color="black", size=12),
    strip.text.y = element_text(color="black", size=12, angle=90),
    legend.title = element_text(color="black", size=14),
    legend.text = element_text(color="black", size=12),
    legend.key.size = unit(1, 'cm'),
    legend.position = "bottom",
    strip.text = element_text(size = 12),
    axis.title.x = element_text(color="black", size=12),
    axis.title.y = element_text(color="black", size=12, angle=90),
    panel.grid.major = element_blank(), # switch off major gridlines
    panel.grid.minor = element_blank(), # switch off minor gridlines
  )
fig_trend
#CREATE A PDF AND SAVES IT TO THE RIGHT DIRECTORY/FOLDER. 
pdf("/Users/connermertz/Documents/Conner Documents/UNM biology/Chapter 1 E1 vs E2/ANCOMoutput/E1vsE2_Location_Pairwise_bar.fig_withcompar.pdf", height = 6, width = 20)
fig_trend
dev.off()
```



```{r}
fig_trend = df_fig_trend %>% 
  ggplot(aes(x = lfc, y = group, color = taxon)) + 
  geom_point(stat = "identity", position = position_dodge()) +
  geom_errorbar(aes(xmin = lfc - se, xmax = lfc + se), width = .2,
                position = position_dodge(.9)) +
  #facet_wrap(vars(group), nrow = 2, scales = "free") +
  labs(x = NULL, y = NULL, title = "Log fold change Pairwise") +
  scale_color_manual(values= c("#6d83da",
"#9bb240",
"#5a3a8e",
"#63c471",
"#c872c5",
"#5f8d3d",
"#8d2c66",
"#45c097",
"#dc5359",
"#d49c3b",
"#d76092",
"#a1843c",
"#b34656",
"#c96234",
"#9d472c",
"black")) +
  theme_bw() +
  theme(
    plot.title = element_text(hjust = 1),
    title =element_text(size=22, face='bold'),
    axis.text.x = element_text(color="black", size=12, angle = 90, hjust = 1),
    axis.text.y = element_text(color="black", size=12),
    strip.text.y = element_text(color="black", size=12, angle=90),
    #legend.title = element_text(color="black", size=20),
    legend.text = element_text(color="black", size=14),
    legend.key.size = unit(1, 'cm'),
    #legend.position = "right",
    strip.text = element_text(size = 16),
    axis.title.x = element_text(color="black", size=14),
    axis.title.y = element_text(color="black", size=14, angle=90),
    panel.grid.major = element_blank(), # switch off major gridlines
    panel.grid.minor = element_blank(), # switch off minor gridlines
  )
#CREATE A PDF AND SAVES IT TO THE RIGHT DIRECTORY/FOLDER. 
pdf("/Users/connermertz/Documents/Conner Documents/UNM biology/Chapter 1 E1 vs E2/ANCOMoutput/E1vsE2_Location_Pairwise_dot.test.pdf", height = 12, width = 22)
fig_trend
dev.off()
```




###############
#ANCOMBC2 by PROTEIN- E1 vs E2 for all locations samples
```{r}
set.seed(123)
output = ancombc2(ps_raw_E1_E2_forcomparison, assay_name = NULL, tax_level = NULL,
                  fix_formula = "Location + Exp_Diet", rand_formula = NULL,
                  p_adj_method = "holm", pseudo = 0, pseudo_sens = TRUE,
                  prv_cut = 0.10, lib_cut = 1000, s0_perc = 0.05,
                  group = "Exp_Diet", struc_zero = TRUE, neg_lb = TRUE,
                  alpha = 0.05, n_cl = 2, verbose = TRUE,
                  global = TRUE, pairwise = TRUE, dunnet = TRUE, trend = TRUE,
                  iter_control = list(tol = 1e-2, max_iter = 20, 
                                      verbose = TRUE),
                  em_control = list(tol = 1e-5, max_iter = 100),
                  lme_control = lme4::lmerControl(),
                  mdfdr_control = list(fwer_ctrl_method = "holm", B = 100),
                  trend_control = list(contrast = list(matrix(c(1, 0, 0, 0, 0,
                                                               -1, 1, 0, 0, 0,
                                                                0, -1, 1, 0, 0, 
                                                                0, 0, -1, 1, 0,
                                                                0, 0, 0, -1, 1),
                                                              nrow = 5, 
                                                              byrow = TRUE),
                                                       matrix(c(1, 0, 0, 0, 0,
                                                               -1, 1, 0, 0, 0,
                                                                0, -1, 1, 0, 0, 
                                                                0, 0, -1, 1, 0,
                                                                0, 0, 0, -1, 1),
                                                              nrow = 5, 
                                                              byrow = TRUE)),
                                       node = list(2, 2),
                                       solver = "ECOS",
                                       B = 100))

```
Save the output
```{r}
res_prim = output$res
saveRDS(res_prim, "/Users/connermertz/Documents/Sequence Analysis/E1_E2_All_Samples/codeexports/ANCOMBC/E1vsE2_PROTEIN_res_prim.rds")
#read in the ANCOMBC output if you want to see E1 vs E2 by PROTEIN
E1vsE2_PROTEIN_res_prim <- readRDS("/Users/connermertz/Documents/Sequence Analysis/E1_E2_All_Samples/codeexports/ANCOMBC/E1vsE2_PROTEIN_res_prim.rds")

res_pair = output$res_pair
saveRDS(res_pair, "/Users/connermertz/Documents/Sequence Analysis/E1_E2_All_Samples/codeexports/ANCOMBC/E1vsE2_PROTEIN_res_pair.rds")
E1vsE2_PROTEIN_res_pair <- readRDS("/Users/connermertz/Documents/Sequence Analysis/E1_E2_All_Samples/codeexports/ANCOMBC/E1vsE2_PROTEIN_res_pair.rds")

res_trend = output$res_trend
saveRDS(res_trend, "/Users/connermertz/Documents/Sequence Analysis/E1_E2_All_Samples/codeexports/ANCOMBC/E1vsE2_PROTEIN_res_trend.rds")
E1vsE2_PROTEIN_res_trend <- readRDS("/Users/connermertz/Documents/Sequence Analysis/E1_E2_All_Samples/codeexports/ANCOMBC/E1vsE2_PROTEIN_res_trend.rds")
```

ANCOM-BC2 primary analysis
```{r}
E1vsE2_PROTEIN_res_prim
```


##ANCOM-BC2 pairwise directional test
ANCOM-BC2 pairwise directional test aims to determine taxa that are differentially abundant between any pair of two groups across three or more experimental groups, while controlling the mdFDR.

In this example, we want to identify taxa that are differentially abundant between any pair of two groups across “lean”, “overweight”, and “obese”. The result contains: 1) log fold changes, 2) standard errors, 3) test statistics, 4) p-values, 5) adjusted p-values, 6) indicators of whether the taxon is differentially abundant (TRUE) or not (FALSE).
```{r}
E1vsE2_PROTEIN_res_pair
write.csv(E1vsE2_PROTEIN_res_pair, "/Users/connermertz/Documents/Sequence Analysis/E1_E2_All_Samples/codeexports/ANCOMBC/E1vsE2_PROTEIN_res_pair_edit.csv")

#read in your res_pair csv: 
E1vsE2_PROTEIN_res_pair2 <- read.csv("/Users/connermertz/Documents/Sequence Analysis/E1_E2_All_Samples/codeexports/ANCOMBC/E1vsE2_PROTEIN_res_pair_edit.csv")
#change name so target_ids match on both files
names(E1vsE2_PROTEIN_res_pair2)[2] ="taxon" 

#read in your RNA classification file 
Taxonomy_classification <- read.csv("/Users/connermertz/Documents/Sequence Analysis/E1_E2_All_Samples/ALL_taxa.csv") #if this doesn't load be sure to dowload the file from the cloud, then it will work
#change name so target_ids match on both files
names(Taxonomy_classification)[1] ="taxon"

#now we need to get our gapmind results and join it with our Taxonomy results from CAT (combining two data frames)
#we only want to keep the genes that showed up with gapmind, we don't need the other genes that showed up
Part_1 <- Taxonomy_classification %>% 
  inner_join (E1vsE2_PROTEIN_res_pair2, by = "taxon")

write.csv(Part_1,"/Users/connermertz/Documents/Sequence Analysis/E1_E2_All_Samples/codeexports/ANCOMBC/E1vsE2_PROTEIN_res_pair_AND_Taxonomy.csv") # went in here and edited file to my liking
```


```{r}
#Plot ANCOM-BC2 pairwise directional test
df_fig_pair = E1vsE2_PROTEIN_res_pair %>%
    filter(diff_Exp_DietE1_2.5P == 1 | 
             diff_Exp_DietE1_5P == 1 |
             diff_Exp_DietE2_10P == 1 |
             diff_Exp_DietE2_2.5P == 1 |
             diff_Exp_DietE2_5P == 1 |
             diff_Exp_DietE1_5P_Exp_DietE1_2.5P == 1 |
             diff_Exp_DietE2_10P_Exp_DietE1_2.5P == 1 |
             diff_Exp_DietE2_2.5P_Exp_DietE1_2.5P == 1 |
             diff_Exp_DietE2_5P_Exp_DietE1_2.5P == 1 |
             diff_Exp_DietE2_10P_Exp_DietE1_5P == 1 |
             diff_Exp_DietE2_2.5P_Exp_DietE1_5P == 1 |
             diff_Exp_DietE2_5P_Exp_DietE1_5P == 1 |
             diff_Exp_DietE2_2.5P_Exp_DietE2_10P == 1 |
             diff_Exp_DietE2_5P_Exp_DietE2_10P == 1 |
             diff_Exp_DietE2_5P_Exp_DietE2_2.5P == 1) %>%
    mutate(lfc_E1_2.5P = ifelse(diff_Exp_DietE1_2.5P == 1, 
                             lfc_Exp_DietE1_2.5P, 0),
           lfc_E1_5P = ifelse(diff_Exp_DietE1_5P == 1, 
                                   lfc_Exp_DietE1_5P, 0),
           lfc_E2_10P = ifelse(diff_Exp_DietE2_10P == 1, 
                                   lfc_Exp_DietE2_10P, 0),
           lfc_E2_2.5P = ifelse(diff_Exp_DietE2_2.5P == 1, 
                                   lfc_Exp_DietE2_2.5P, 0),
           lfc_E2_5P = ifelse(diff_Exp_DietE2_5P == 1, 
                                   lfc_Exp_DietE2_5P, 0),
           lfc_E1_5P_E1_2.5P = ifelse(diff_Exp_DietE1_5P_Exp_DietE1_2.5P == 1, 
                                   lfc_Exp_DietE1_5P_Exp_DietE1_2.5P, 0),
           lfc_E2_10P_E1_2.5P = ifelse(diff_Exp_DietE2_10P_Exp_DietE1_2.5P == 1, 
                                   lfc_Exp_DietE2_10P_Exp_DietE1_2.5P, 0),
           lfc_E2_2.5P_E1_2.5P = ifelse(diff_Exp_DietE2_2.5P_Exp_DietE1_2.5P == 1, 
                                   lfc_Exp_DietE2_2.5P_Exp_DietE1_2.5P, 0),
           lfc_E2_5P_E1_2.5P = ifelse(diff_Exp_DietE2_5P_Exp_DietE1_2.5P == 1, 
                                   lfc_Exp_DietE2_5P_Exp_DietE1_2.5P, 0),
           lfc_E2_10P_E1_5P = ifelse(diff_Exp_DietE2_10P_Exp_DietE1_5P == 1, 
                                   lfc_Exp_DietE2_10P_Exp_DietE1_5P, 0),
           lfc_E2_2.5P_E1_5P = ifelse(diff_Exp_DietE2_2.5P_Exp_DietE1_5P == 1, 
                                   lfc_Exp_DietE2_2.5P_Exp_DietE1_5P, 0),
           lfc_E2_5P_E1_5P = ifelse(diff_Exp_DietE2_5P_Exp_DietE1_5P == 1, 
                                   lfc_Exp_DietE2_5P_Exp_DietE1_5P, 0),
           lfc_E2_2.5P_E2_10P = ifelse(diff_Exp_DietE2_2.5P_Exp_DietE2_10P == 1, 
                                   lfc_Exp_DietE2_2.5P_Exp_DietE2_10P, 0),
           lfc_E2_5P_E2_10P = ifelse(diff_Exp_DietE2_5P_Exp_DietE2_10P == 1, 
                                   lfc_Exp_DietE2_5P_Exp_DietE2_10P, 0),
           lfc_E2_5P_E2_2.5P = ifelse(diff_Exp_DietE2_5P_Exp_DietE2_2.5P == 1, 
                                        lfc_Exp_DietE2_5P_Exp_DietE2_2.5P, 0)) %>%
    transmute(taxon, 
              `E1_2.5P vs. E1_10P` = round(lfc_E1_2.5P, 2), 
              `E1_5P vs. E1_10P` = round(lfc_E1_5P, 2),
              `E2_10P vs. E1_10P` = round(lfc_E2_10P, 2),
              `E2_2.5P vs. E1_10P` = round(lfc_E2_2.5P, 2),
              `E2_5P vs. E1_10P` = round(lfc_E2_5P, 2),
              `E1_5P vs. E1_2.5P` = round(lfc_E1_5P_E1_2.5P, 2),
              `E2_10P vs. E1_2.5P` = round(lfc_E2_10P_E1_2.5P, 2),
              `E2_2.5P vs. E1_2.5P` = round(lfc_E2_2.5P_E1_2.5P, 2),
              `E2_5P vs. E1_2.5P` = round(lfc_E2_5P_E1_2.5P, 2),
              `E2_10P vs. E1_5P` = round(lfc_E2_10P_E1_5P, 2),
              `E2_2.5P vs. E1_5P` = round(lfc_E2_2.5P_E1_5P, 2),
              `E2_5P vs. E1_5P` = round(lfc_E2_5P_E1_5P, 2),
              `E2_2.5P vs. E2_10P` = round(lfc_E2_2.5P_E2_10P, 2),
              `E2_5P vs. E2_10P` = round(lfc_E2_5P_E2_10P, 2),
              `E2_5P vs. E2_2.5P` = round(lfc_E2_5P_E2_2.5P, 2)
              ) %>%
    pivot_longer(cols = `E1_2.5P vs. E1_10P`:`E2_5P vs. E2_2.5P`, 
                 names_to = "group", values_to = "value") %>%
    arrange(taxon)
df_fig_pair$group = factor(df_fig_pair$group, 
                           levels = c("E1_2.5P vs. E1_10P",
                                      "E1_5P vs. E1_10P",
                                      "E2_10P vs. E1_10P",
                                      "E2_2.5P vs. E1_10P",
                                      "E2_5P vs. E1_10P",
                                      "E1_5P vs. E1_2.5P",
                                      "E2_10P vs. E1_2.5P",
                                      "E2_2.5P vs. E1_2.5P",
                                      "E2_5P vs. E1_2.5P",
                                      "E2_10P vs. E1_5P",
                                      "E2_2.5P vs. E1_5P",
                                      "E2_5P vs. E1_5P",
                                      "E2_2.5P vs. E2_10P",
                                      "E2_5P vs. E2_10P",
                                      "E2_5P vs. E2_2.5P"))
  
lo = floor(min(df_fig_pair$value))
up = ceiling(max(df_fig_pair$value))
mid = (lo + up)/2
fig_pair = df_fig_pair %>%
  ggplot(aes(x = group, y = taxon, fill = value)) + 
  geom_tile(color = "black") +
  scale_fill_gradient2(low = "blue", high = "red", mid = "white", 
                       na.value = "white", midpoint = mid, limit = c(lo, up),
                       name = NULL) +
  geom_text(aes(group, taxon, label = value), color = "black", size = 4) +
  labs(x = NULL, y = NULL, title = "Log fold change of pairwise comparisons in E1 vs E2 by % Protein") +
  theme_bw() +
  theme(
    plot.title = element_text(hjust = 1),
    title =element_text(size=22, face='bold'),
    axis.text.x = element_text(color="black", size=12, angle = 90, hjust = 1),
    axis.text.y = element_text(color="black", size=12),
    strip.text.y = element_text(color="black", size=12, angle=90),
    #legend.title = element_text(color="black", size=20),
    legend.text = element_text(color="black", size=14),
    legend.key.size = unit(1, 'cm'),
    #legend.position = "right",
    strip.text = element_text(size = 16),
    axis.title.x = element_text(color="black", size=14),
    axis.title.y = element_text(color="black", size=14, angle=90),
    panel.grid.major = element_blank(), # switch off major gridlines
    panel.grid.minor = element_blank(), # switch off minor gridlines
  )
#CREATE A PDF AND SAVES IT TO THE RIGHT DIRECTORY/FOLDER. 
pdf("/Users/connermertz/Documents/Conner Documents/UNM biology/Chapter 1 E1 vs E2/ANCOMoutput/E1vsE2_Protein_LFC_Pairwise.pdf", height = 12, width = 14)
fig_pair
dev.off()
```

































#ANCOMBC by percent protein in each location- SHOWED NO DIFFERENTIALLY ABUNDANT TAXA! 
#Duod
#############################

#AMCOMBC E1vsE2 by Percent Protein in the DUODENUM:

```{r}
set.seed(123)
output = ancombc2(ps_raw_E1_E2_duod, assay_name = NULL, tax_level = NULL,
                  fix_formula = "Exp_Diet_Location", rand_formula = NULL,
                  p_adj_method = "holm", pseudo = 0, pseudo_sens = TRUE,
                  prv_cut = 0.10, lib_cut = 1000, s0_perc = 0.05,
                  group = "Exp_Diet_Location", struc_zero = TRUE, neg_lb = TRUE,
                  alpha = 0.05, n_cl = 2, verbose = TRUE,
                  global = TRUE, pairwise = TRUE, dunnet = TRUE, trend = TRUE,
                  iter_control = list(tol = 1e-2, max_iter = 20, 
                                      verbose = TRUE),
                  em_control = list(tol = 1e-5, max_iter = 100),
                  lme_control = lme4::lmerControl(),
                  mdfdr_control = list(fwer_ctrl_method = "holm", B = 100),
                  trend_control = list(contrast = list(matrix(c(1, 0, 0, 0, 0,
                                                               -1, 1, 0, 0, 0,
                                                                0, -1, 1, 0, 0, 
                                                                0, 0, -1, 1, 0,
                                                                0, 0, 0, -1, 1),
                                                              nrow = 5, 
                                                              byrow = TRUE),
                                                       matrix(c(1, 0, 0, 0, 0,
                                                               -1, 1, 0, 0, 0,
                                                                0, -1, 1, 0, 0, 
                                                                0, 0, -1, 1, 0,
                                                                0, 0, 0, -1, 1),
                                                              nrow = 5, 
                                                              byrow = TRUE)),
                                       node = list(2, 2),
                                       solver = "ECOS",
                                       B = 100))

```
#save the output
```{r}
res_prim = output$res
saveRDS(res_prim, "/Users/connermertz/Documents/Sequence Analysis/E1_E2_All_Samples/codeexports/ANCOMBC/E1vsE2_Protein_DUOD_res_prim.rds")
#read in the ANCOMBC output if you want to see E1 vs E2 by Location
E1vsE2_Protein_DUOD_res_prim <- readRDS("/Users/connermertz/Documents/Sequence Analysis/E1_E2_All_Samples/codeexports/ANCOMBC/E1vsE2_Protein_DUOD_res_prim.rds")

res_pair = output$res_pair
saveRDS(res_pair, "/Users/connermertz/Documents/Sequence Analysis/E1_E2_All_Samples/codeexports/ANCOMBC/E1vsE2_Protein_DUOD_res_pair.rds")
E1vsE2_Protein_DUOD_res_pair <- readRDS("/Users/connermertz/Documents/Sequence Analysis/E1_E2_All_Samples/codeexports/ANCOMBC/E1vsE2_Protein_DUOD_res_pair.rds")

res_global = output$res_global
saveRDS(res_global, "/Users/connermertz/Documents/Sequence Analysis/E1_E2_All_Samples/codeexports/ANCOMBC/E1vsE2_Protein_DUOD_res_global.rds")
E1vsE2_Protein_DUOD_res_global <- readRDS("/Users/connermertz/Documents/Sequence Analysis/E1_E2_All_Samples/codeexports/ANCOMBC/E1vsE2_Protein_DUOD_res_global.rds")

res_trend = output$res_trend
saveRDS(res_trend, "/Users/connermertz/Documents/Sequence Analysis/E1_E2_All_Samples/codeexports/ANCOMBC/E1vsE2_Protein_DUOD_res_trend.rds")
E1vsE2_Protein_DUOD_res_trend <- readRDS("/Users/connermertz/Documents/Sequence Analysis/E1_E2_All_Samples/codeexports/ANCOMBC/E1vsE2_Protein_DUOD_res_trend.rds")
```

ANCOM-BC2 global test aims to determine taxa that are differentially abundant between at least two groups across three or more experimental groups.

In this example, we want to identify taxa that are differentially abundant between at least two groups across “lean”, “overweight”, and “obese”. The result contains: 1) test statistics, 2) p-values, 3) adjusted p-values, 4) indicators of whether the taxon is differentially abundant (TRUE) or not (FALSE).
#look at res_prim
```{r}
E1vsE2_Protein_DUOD_res_prim
```
#look at global table
```{r}
E1vsE2_Protein_DUOD_res_global
```

#Plot it!
```{r}
df_Exp_Diet_Location = E1vsE2_Protein_DUOD_res_prim %>%
    dplyr::select(taxon, contains("Exp_Diet_Location")) 
df_fig_global = df_Exp_Diet_Location %>%
    dplyr::left_join(E1vsE2_Protein_DUOD_res_global %>%
                  transmute(taxon, diff_Exp_Diet_Location = diff_abn)) %>%
    dplyr::filter(diff_Exp_Diet_Location == 1) %>%
    mutate(lfc_E1_2.5P_DUOD = ifelse(diff_Exp_Diet_LocationE1_2.5P_DUOD == 1, 
                             lfc_Exp_Diet_LocationE1_2.5P_DUOD, 0),
           lfc_E1_5P_DUOD = ifelse(diff_Exp_Diet_LocationE1_5P_DUOD == 1, 
                                   lfc_Exp_Diet_LocationE1_5P_DUOD, 0),
           lfc_E2_10P_DUOD = ifelse(diff_Exp_Diet_LocationE2_10P_DUOD == 1, 
                                   lfc_Exp_Diet_LocationE2_10P_DUOD, 0),
           lfc_E2_2.5P_DUOD = ifelse(diff_Exp_Diet_LocationE2_2.5P_DUOD == 1, 
                                   lfc_Exp_Diet_LocationE2_2.5P_DUOD, 0),
           lfc_E2_5P_DUOD = ifelse(diff_Exp_Diet_LocationE2_5P_DUOD == 1, 
                                   lfc_Exp_Diet_LocationE2_5P_DUOD, 0)) %>%
    transmute(taxon, 
              `E1_2.5P_DUOD vs. E1_10P_DUOD` = round(lfc_E1_2.5P_DUOD, 2), 
              `E1_5P_DUOD vs. E1_10P_DUOD` = round(lfc_E1_5P_DUOD, 2),
              `E2_10P_DUOD vs. E1_10P_DUOD` = round(lfc_E2_10P_DUOD, 2),
              `E2_5P_DUOD vs. E1_10P_DUOD` = round(lfc_E2_5P_DUOD, 2)) %>%
    pivot_longer(cols = `E1_2.5P_DUOD vs. E1_10P_DUOD`:`E2_5P_DUOD vs. E1_10P_DUOD`, 
                 names_to = "group", values_to = "value") %>%
    arrange(taxon)
  
lo = floor(min(df_fig_global$value))
up = ceiling(max(df_fig_global$value))
mid = (lo + up)/2
fig_global = df_fig_global %>%
  ggplot(aes(x = group, y = taxon, fill = value)) + 
  geom_tile(color = "black") +
  scale_fill_gradient2(low = "blue", high = "red", mid = "white", 
                       na.value = "white", midpoint = mid, limit = c(lo, up),
                       name = NULL) +
  geom_text(aes(group, taxon, label = value), color = "black", size = 4) +
  labs(x = NULL, y = NULL, title = "Log fold changes for globally significant taxa in DUODENUM") +
  theme(plot.title = element_text(hjust = 0.5)) +
theme_bw() +
  theme(
    plot.title = element_text(hjust = 1),
    title =element_text(size=22, face='bold'),
    axis.text.x = element_text(color="black", size=12, angle = 60, hjust = 1),
    axis.text.y = element_text(color="black", size=12),
    strip.text.y = element_text(color="black", size=12, angle=90),
    #legend.title = element_text(color="black", size=20),
    legend.text = element_text(color="black", size=14),
    legend.key.size = unit(1, 'cm'),
    #legend.position = "right",
    strip.text = element_text(size = 16),
    axis.title.x = element_text(color="black", size=14),
    axis.title.y = element_text(color="black", size=14, angle=90),
    panel.grid.major = element_blank(), # switch off major gridlines
    panel.grid.minor = element_blank(), # switch off minor gridlines
  )
#CREATE A PDF AND SAVES IT TO THE RIGHT DIRECTORY/FOLDER. 
pdf("/Users/connermertz/Documents/Conner Documents/UNM biology/Chapter 1 E1 vs E2/ANCOMoutput/E1vsE2_Protein_DUOD_res_global_figure.pdf", height = 12, width = 14)
fig_global
dev.off()

#results: No taxa are shown to be differentially abundant in E1vsE2_Protein_DUOD_res_global
```

##ANCOM-BC2 pairwise directional test
ANCOM-BC2 pairwise directional test aims to determine taxa that are differentially abundant between any pair of two groups across three or more experimental groups, while controlling the mdFDR.

In this example, we want to identify taxa that are differentially abundant between any pair of two groups across “lean”, “overweight”, and “obese”. The result contains: 1) log fold changes, 2) standard errors, 3) test statistics, 4) p-values, 5) adjusted p-values, 6) indicators of whether the taxon is differentially abundant (TRUE) or not (FALSE).
```{r}
E1vsE2_Protein_DUOD_res_pair
```

#plot Pairwise tests
```{r}
df_fig_pair = E1vsE2_Protein_DUOD_res_pair %>%
    filter(diff_Exp_Diet_LocationE1_2.5P_DUOD == 1 | 
             diff_Exp_Diet_LocationE1_5P_DUOD == 1 |
             diff_Exp_Diet_LocationE2_10P_DUOD == 1 |
             diff_Exp_Diet_LocationE2_2.5P_DUOD == 1 |
             diff_Exp_Diet_LocationE2_5P_DUOD == 1 |
             diff_Exp_Diet_LocationE1_5P_DUOD_Exp_Diet_LocationE1_2.5P_DUOD == 1 |
             diff_Exp_Diet_LocationE2_10P_DUOD_Exp_Diet_LocationE1_2.5P_DUOD == 1 |
             diff_Exp_Diet_LocationE2_2.5P_DUOD_Exp_Diet_LocationE1_2.5P_DUOD == 1 |
             diff_Exp_Diet_LocationE2_5P_DUOD_Exp_Diet_LocationE1_2.5P_DUOD == 1 |
             diff_Exp_Diet_LocationE2_10P_DUOD_Exp_Diet_LocationE1_5P_DUOD == 1 |
             diff_Exp_Diet_LocationE2_2.5P_DUOD_Exp_Diet_LocationE1_5P_DUOD == 1 |
             diff_Exp_Diet_LocationE2_5P_DUOD_Exp_Diet_LocationE1_5P_DUOD == 1 |
             diff_Exp_Diet_LocationE2_2.5P_DUOD_Exp_Diet_LocationE2_10P_DUOD == 1 |
             diff_Exp_Diet_LocationE2_5P_DUOD_Exp_Diet_LocationE2_10P_DUOD == 1 |
             diff_Exp_Diet_LocationE2_5P_DUOD_Exp_Diet_LocationE2_2.5P_DUOD == 1) %>%
    mutate(lfc_E1_2.5P_DUOD = ifelse(diff_Exp_Diet_LocationE1_2.5P_DUOD == 1, 
                             lfc_Exp_Diet_LocationE1_2.5P_DUOD, 0),
           lfc_E1_5P_DUOD = ifelse(diff_Exp_Diet_LocationE1_5P_DUOD == 1, 
                                   lfc_Exp_Diet_LocationE1_5P_DUOD, 0),
           lfc_E2_10P_DUOD = ifelse(diff_Exp_Diet_LocationE2_10P_DUOD == 1, 
                                   lfc_Exp_Diet_LocationE2_10P_DUOD, 0),
           lfc_E2_2.5P_DUOD = ifelse(diff_Exp_Diet_LocationE2_2.5P_DUOD == 1, 
                                   lfc_Exp_Diet_LocationE2_2.5P_DUOD, 0),
           lfc_E2_5P_DUOD = ifelse(diff_Exp_Diet_LocationE2_5P_DUOD == 1, 
                                   lfc_Exp_Diet_LocationE2_5P_DUOD, 0),
           lfc_E1_5P_DUOD_E1_2.5P_DUOD = ifelse(diff_Exp_Diet_LocationE1_5P_DUOD_Exp_Diet_LocationE1_2.5P_DUOD == 1, 
                                   lfc_Exp_Diet_LocationE1_5P_DUOD_Exp_Diet_LocationE1_2.5P_DUOD, 0),
           lfc_E2_10P_DUOD_E1_2.5P_DUOD = ifelse(diff_Exp_Diet_LocationE2_10P_DUOD_Exp_Diet_LocationE1_2.5P_DUOD == 1, 
                                   lfc_Exp_Diet_LocationE2_10P_DUOD_Exp_Diet_LocationE1_2.5P_DUOD, 0),
           lfc_E2_2.5P_DUOD_E1_2.5P_DUOD = ifelse(diff_Exp_Diet_LocationE2_2.5P_DUOD_Exp_Diet_LocationE1_2.5P_DUOD == 1, 
                                   lfc_Exp_Diet_LocationE2_2.5P_DUOD_Exp_Diet_LocationE1_2.5P_DUOD, 0),
           lfc_E2_5P_DUOD_E1_2.5P_DUOD = ifelse(diff_Exp_Diet_LocationE2_5P_DUOD_Exp_Diet_LocationE1_2.5P_DUOD == 1, 
                                   lfc_Exp_Diet_LocationE2_5P_DUOD_Exp_Diet_LocationE1_2.5P_DUOD, 0),
           lfc_E2_10P_DUOD_E1_5P_DUOD = ifelse(diff_Exp_Diet_LocationE2_10P_DUOD_Exp_Diet_LocationE1_5P_DUOD == 1, 
                                   lfc_Exp_Diet_LocationE2_10P_DUOD_Exp_Diet_LocationE1_5P_DUOD, 0),
           lfc_E2_2.5P_DUOD_E1_5P_DUOD = ifelse(diff_Exp_Diet_LocationE2_2.5P_DUOD_Exp_Diet_LocationE1_5P_DUOD == 1, 
                                   lfc_Exp_Diet_LocationE2_2.5P_DUOD_Exp_Diet_LocationE1_5P_DUOD, 0),
           lfc_E2_5P_DUOD_E1_5P_DUOD = ifelse(diff_Exp_Diet_LocationE2_5P_DUOD_Exp_Diet_LocationE1_5P_DUOD == 1, 
                                   lfc_Exp_Diet_LocationE2_5P_DUOD_Exp_Diet_LocationE1_5P_DUOD, 0),
           lfc_E2_2.5P_DUOD_E2_10P_DUOD = ifelse(diff_Exp_Diet_LocationE2_2.5P_DUOD_Exp_Diet_LocationE2_10P_DUOD == 1, 
                                   lfc_Exp_Diet_LocationE2_2.5P_DUOD_Exp_Diet_LocationE2_10P_DUOD, 0),
           lfc_E2_5P_DUOD_E2_10P_DUOD = ifelse(diff_Exp_Diet_LocationE2_5P_DUOD_Exp_Diet_LocationE2_10P_DUOD == 1, 
                                   lfc_Exp_Diet_LocationE2_5P_DUOD_Exp_Diet_LocationE2_10P_DUOD, 0),
           lfc_E2_5P_DUOD_E2_2.5P_DUOD = ifelse(diff_Exp_Diet_LocationE2_5P_DUOD_Exp_Diet_LocationE2_2.5P_DUOD == 1, 
                                        lfc_Exp_Diet_LocationE2_5P_DUOD_Exp_Diet_LocationE2_2.5P_DUOD, 0)) %>%
    transmute(taxon, 
              `E1_2.5P_DUOD vs. E1_10P_DUOD` = round(lfc_E1_2.5P_DUOD, 2), 
              `E1_5P_DUOD vs. E1_10P_DUOD` = round(lfc_E1_5P_DUOD, 2),
              `E2_10P_DUOD vs. E1_10P_DUOD` = round(lfc_E2_10P_DUOD, 2),
              `E2_5P_DUOD vs. E1_10P_DUOD` = round(lfc_E2_5P_DUOD, 2),
              `E1_5P_DUOD vs. E1_2.5P_DUOD` = round(lfc_E1_5P_DUOD_E1_2.5P_DUOD, 2),
              `E2_10P_DUOD vs. E1_2.5P_DUOD` = round(lfc_E2_10P_DUOD_E1_2.5P_DUOD, 2),
              `E2_2.5P_DUOD vs. E1_2.5P_DUOD` = round(lfc_E2_2.5P_DUOD_E1_2.5P_DUOD, 2),
              `E2_5P_DUOD vs. E1_2.5P_DUOD` = round(lfc_E2_5P_DUOD_E1_2.5P_DUOD, 2),
              `E2_10P_DUOD vs. E1_5P_DUOD` = round(lfc_E2_10P_DUOD_E1_5P_DUOD, 2),
              `E2_2.5P_DUOD vs. E1_5P_DUOD` = round(lfc_E2_2.5P_DUOD_E1_5P_DUOD, 2),
              `E2_5P_DUOD vs. E1_5P_DUOD` = round(lfc_E2_5P_DUOD_E1_5P_DUOD, 2),
              `E2_2.5P_DUOD vs. E2_10P_DUOD` = round(lfc_E2_2.5P_DUOD_E2_10P_DUOD, 2),
              `E2_5P_DUOD vs. E2_10P_DUOD` = round(lfc_E2_5P_DUOD_E2_10P_DUOD, 2),
              `E2_5P_DUOD vs. E2_2.5P_DUOD` = round(lfc_E2_5P_DUOD_E2_2.5P_DUOD, 2)
              ) %>%
    pivot_longer(cols = `E1_2.5P_DUOD vs. E1_10P_DUOD`:`E2_5P_DUOD vs. E2_2.5P_DUOD`, 
                 names_to = "group", values_to = "value") %>%
    arrange(taxon)
df_fig_pair$group = factor(df_fig_pair$group, 
                           levels = c("E1_2.5P_DUOD vs. E1_10P_DUOD",
                                      "E1_5P_DUOD vs. E1_10P_DUOD",
                                      "E2_10P_DUOD vs. E1_10P_DUOD",
                                      "E2_5P_DUOD vs. E1_10P_DUOD",
                                      "E1_5P_DUOD vs. E1_2.5P_DUOD",
                                      "E2_10P_DUOD vs. E1_2.5P_DUOD",
                                      "E2_2.5P_DUOD vs. E1_2.5P_DUOD",
                                      "E2_5P_DUOD vs. E1_2.5P_DUOD",
                                      "E2_10P_DUOD vs. E1_5P_DUOD",
                                      "E2_2.5P_DUOD vs. E1_5P_DUOD",
                                      "E2_5P_DUOD vs. E1_5P_DUOD",
                                      "E2_2.5P_DUOD vs. E2_10P_DUOD",
                                      "E2_5P_DUOD vs. E2_10P_DUOD",
                                      "E2_5P_DUOD vs. E2_2.5P_DUOD"))
  
lo = floor(min(df_fig_pair$value))
up = ceiling(max(df_fig_pair$value))
mid = (lo + up)/2
fig_pair = df_fig_pair %>%
  ggplot(aes(x = group, y = taxon, fill = value)) + 
  geom_tile(color = "black") +
  scale_fill_gradient2(low = "blue", high = "red", mid = "white", 
                       na.value = "white", midpoint = mid, limit = c(lo, up),
                       name = NULL) +
  geom_text(aes(group, taxon, label = value), color = "black", size = 4) +
  labs(x = NULL, y = NULL, title = "Log fold change of pairwise comparisons in the DUODENUM E1 vs E2 by % Protein") +
  theme_minimal() +
  theme(plot.title = element_text(hjust = 0.5))
fig_pair

#results- no differentially abundant taxa identified E1vsE2_Protein_DUOD pairwise 
```


Res Trend:
Sometimes the experimental groups are intrinsically ordered (e.g., in a dose-response study), and we would like to see if the microbial abundances show some patterns accordingly. Examples of patterns include: monotonically increasing, monotonically decreasing, and umbrella shape.
We will test both the monotonically increasing and decreasing trends in this example. The result contains: 1) log fold changes, 2) standard errors, 3) test statistics, 4) p-values, 5) adjusted p-values, 6) indicators of whether the taxon is differentially abundant (TRUE) or not (FALSE).
```{r}
E1vsE2_Protein_DUOD_res_trend
```



```{r}
df_fig_trend = E1vsE2_Protein_DUOD_res_trend %>%
    dplyr::filter(diff_abn) %>%
    transmute(taxon,
              lfc = lfc_Exp_Diet_LocationE1_2.5P_DUOD,
              se = se_Exp_Diet_LocationE1_2.5P_DUOD,
              p_val,
              group = "E1_2.5P_DUOD - E1_10P_DUOD") %>%
    bind_rows(
        res_trend %>%
            dplyr::filter(diff_abn) %>%
            transmute(taxon,
                      lfc = lfc_Exp_Diet_LocationE1_5P_DUOD,
                      se = se_Exp_Diet_LocationE1_5P_DUOD,
                      p_val,
                      group = "E1_5P_DUOD - E1_10P_DUOD")
    ) %>%
    bind_rows(
        res_trend %>%
            dplyr::filter(diff_abn) %>%
            transmute(taxon,
                      lfc = lfc_Exp_Diet_LocationE2_10P_DUOD,
                      se = se_Exp_Diet_LocationE2_10P_DUOD,
                      p_val,
                      group = "E2_10P_DUOD - E1_10P_DUOD")
          
    ) %>%
  bind_rows(
        res_trend %>%
            dplyr::filter(diff_abn) %>%
            transmute(taxon,
                      lfc = lfc_Exp_Diet_LocationE2_2.5P_DUOD,
                      se = se_Exp_Diet_LocationE2_2.5P_DUOD,
                      p_val,
                      group = "E2_2.5P_DUOD - E1_10P_DUOD")
          
    ) %>%
  bind_rows(
        res_trend %>%
            dplyr::filter(diff_abn) %>%
            transmute(taxon,
                      lfc = lfc_Exp_Diet_LocationE2_5P_DUOD,
                      se = se_Exp_Diet_LocationE2_5P_DUOD,
                      p_val,
                      group = "E2_5P_DUOD - E1_10P_DUOD")
          
    )
df_fig_trend$group = factor(df_fig_trend$group, 
                           levels = c("E1_2.5P_DUOD - E1_10P_DUOD", "E1_5P_DUOD - E1_10P_DUOD","E2_10P_DUOD - E1_10P_DUOD","E2_2.5P_DUOD - E1_10P_DUOD", "E2_5P_DUOD - E1_10P_DUOD"))
  
fig_trend = df_fig_trend %>% 
  ggplot(aes(x = group, y = lfc, fill = group)) + 
  geom_bar(stat = "identity", position = position_dodge(), color = "black") +
  geom_errorbar(aes(ymin = lfc - se, ymax = lfc + se), width = .2,
                position = position_dodge(.9)) +
  facet_wrap(vars(taxon), nrow = 2, scales = "free") +
  labs(x = NULL, y = NULL, title = "Log fold change as compared to obese subjects") +
  scale_fill_brewer(palette = "Set2", name = NULL) +
  theme_bw() +
  theme(plot.title = element_text(hjust = 0.5),
        axis.text.x = element_blank(),
        axis.ticks.x = element_blank(),
        legend.position = c(0.92, 0.2))
fig_trend
```

###################################################
#ANCOMBC E1vsE2 by Percent Protein in the ILEUM:

```{r}
output = ancombc2(ps_raw_E1_E2_ile, assay_name = NULL, tax_level = NULL,
                  fix_formula = "Exp_Diet_Location", rand_formula = NULL,
                  p_adj_method = "holm", pseudo = 0, pseudo_sens = TRUE,
                  prv_cut = 0.10, lib_cut = 1000, s0_perc = 0.05,
                  group = "Exp_Diet_Location", struc_zero = TRUE, neg_lb = TRUE,
                  alpha = 0.05, n_cl = 2, verbose = TRUE,
                  global = TRUE, pairwise = TRUE, dunnet = TRUE, trend = TRUE,
                  iter_control = list(tol = 1e-2, max_iter = 20, 
                                      verbose = TRUE),
                  em_control = list(tol = 1e-5, max_iter = 100),
                  lme_control = lme4::lmerControl(),
                  mdfdr_control = list(fwer_ctrl_method = "holm", B = 100),
                  trend_control = list(contrast = list(matrix(c(1, 0, 0, 0, 0,
                                                               -1, 1, 0, 0, 0,
                                                                0, -1, 1, 0, 0, 
                                                                0, 0, -1, 1, 0,
                                                                0, 0, 0, -1, 1),
                                                              nrow = 5, 
                                                              byrow = TRUE),
                                                       matrix(c(1, 0, 0, 0, 0,
                                                               -1, 1, 0, 0, 0,
                                                                0, -1, 1, 0, 0, 
                                                                0, 0, -1, 1, 0,
                                                                0, 0, 0, -1, 1),
                                                              nrow = 5, 
                                                              byrow = TRUE)),
                                       node = list(2, 2),
                                       solver = "ECOS",
                                       B = 100))
```

#save the output
```{r}
res_prim = output$res
saveRDS(res_prim, "/Users/connermertz/Documents/Sequence Analysis/E1_E2_All_Samples/codeexports/ANCOMBC/E1vsE2_Protein_ILE_res_prim.rds")
#read in the ANCOMBC output if you want to see E1 vs E2 by Location
E1vsE2_Protein_ILE_res_prim <- readRDS("/Users/connermertz/Documents/Sequence Analysis/E1_E2_All_Samples/codeexports/ANCOMBC/E1vsE2_Protein_ILE_res_prim.rds")

res_pair = output$res_pair
saveRDS(res_pair, "/Users/connermertz/Documents/Sequence Analysis/E1_E2_All_Samples/codeexports/ANCOMBC/E1vsE2_Protein_ILE_res_pair.rds")
E1vsE2_Protein_ILE_res_pair <- readRDS("/Users/connermertz/Documents/Sequence Analysis/E1_E2_All_Samples/codeexports/ANCOMBC/E1vsE2_Protein_ILE_res_pair.rds")

res_global = output$res_global
saveRDS(res_global, "/Users/connermertz/Documents/Sequence Analysis/E1_E2_All_Samples/codeexports/ANCOMBC/E1vsE2_Protein_ILE_res_global.rds")
E1vsE2_Protein_ILE_res_global <- readRDS("/Users/connermertz/Documents/Sequence Analysis/E1_E2_All_Samples/codeexports/ANCOMBC/E1vsE2_Protein_ILE_res_global.rds")

res_trend = output$res_trend
saveRDS(res_trend, "/Users/connermertz/Documents/Sequence Analysis/E1_E2_All_Samples/codeexports/ANCOMBC/E1vsE2_Protein_ILE_res_trend.rds")
E1vsE2_Protein_ILE_res_trend <- readRDS("/Users/connermertz/Documents/Sequence Analysis/E1_E2_All_Samples/codeexports/ANCOMBC/E1vsE2_Protein_ILE_res_trend.rds")
```

##ANCOM-BC2 pairwise directional test
ANCOM-BC2 pairwise directional test aims to determine taxa that are differentially abundant between any pair of two groups across three or more experimental groups, while controlling the mdFDR.
```{r}
E1vsE2_Protein_ILE_res_pair
```
Look at res_pair results for E1_E2_%P_ILE
```{r}
df_fig_pair = E1vsE2_Protein_ILE_res_pair %>%
    filter(diff_Exp_Diet_LocationE1_2.5P_IL == 1 | 
             diff_Exp_Diet_LocationE1_5P_IL == 1 |
             diff_Exp_Diet_LocationE2_10P_IL == 1 |
             diff_Exp_Diet_LocationE2_2.5P_IL == 1 |
             diff_Exp_Diet_LocationE2_5P_IL == 1 |
             diff_Exp_Diet_LocationE1_5P_IL_Exp_Diet_LocationE1_2.5P_IL == 1 |
             diff_Exp_Diet_LocationE2_10P_IL_Exp_Diet_LocationE1_2.5P_IL == 1 |
             diff_Exp_Diet_LocationE2_2.5P_IL_Exp_Diet_LocationE1_2.5P_IL == 1 |
             diff_Exp_Diet_LocationE2_5P_IL_Exp_Diet_LocationE1_2.5P_IL == 1 |
             diff_Exp_Diet_LocationE2_10P_IL_Exp_Diet_LocationE1_5P_IL == 1 |
             diff_Exp_Diet_LocationE2_2.5P_IL_Exp_Diet_LocationE1_5P_IL == 1 |
             diff_Exp_Diet_LocationE2_5P_IL_Exp_Diet_LocationE1_5P_IL == 1 |
             diff_Exp_Diet_LocationE2_2.5P_IL_Exp_Diet_LocationE2_10P_IL == 1 |
             diff_Exp_Diet_LocationE2_5P_IL_Exp_Diet_LocationE2_10P_IL == 1 |
             diff_Exp_Diet_LocationE2_5P_IL_Exp_Diet_LocationE2_2.5P_IL == 1) %>%
    mutate(lfc_E1_2.5P_IL = ifelse(diff_Exp_Diet_LocationE1_2.5P_IL == 1, 
                             lfc_Exp_Diet_LocationE1_2.5P_IL, 0),
           lfc_E1_5P_IL = ifelse(diff_Exp_Diet_LocationE1_5P_IL == 1, 
                                   lfc_Exp_Diet_LocationE1_5P_IL, 0),
           lfc_E2_10P_IL = ifelse(diff_Exp_Diet_LocationE2_10P_IL == 1, 
                                   lfc_Exp_Diet_LocationE2_10P_IL, 0),
           lfc_E2_2.5P_IL = ifelse(diff_Exp_Diet_LocationE2_2.5P_IL == 1, 
                                   lfc_Exp_Diet_LocationE2_2.5P_IL, 0),
           lfc_E2_5P_IL = ifelse(diff_Exp_Diet_LocationE2_5P_IL == 1, 
                                   lfc_Exp_Diet_LocationE2_5P_IL, 0),
           lfc_E1_5P_IL_E1_2.5P_IL = ifelse(diff_Exp_Diet_LocationE1_5P_IL_Exp_Diet_LocationE1_2.5P_IL == 1, 
                                   lfc_Exp_Diet_LocationE1_5P_IL_Exp_Diet_LocationE1_2.5P_IL, 0),
           lfc_E2_10P_IL_E1_2.5P_IL = ifelse(diff_Exp_Diet_LocationE2_10P_IL_Exp_Diet_LocationE1_2.5P_IL == 1, 
                                   lfc_Exp_Diet_LocationE2_10P_IL_Exp_Diet_LocationE1_2.5P_IL, 0),
           lfc_E2_2.5P_IL_E1_2.5P_IL = ifelse(diff_Exp_Diet_LocationE2_2.5P_IL_Exp_Diet_LocationE1_2.5P_IL == 1, 
                                   lfc_Exp_Diet_LocationE2_2.5P_IL_Exp_Diet_LocationE1_2.5P_IL, 0),
           lfc_E2_5P_IL_E1_2.5P_IL = ifelse(diff_Exp_Diet_LocationE2_5P_IL_Exp_Diet_LocationE1_2.5P_IL == 1, 
                                   lfc_Exp_Diet_LocationE2_5P_IL_Exp_Diet_LocationE1_2.5P_IL, 0),
           lfc_E2_10P_IL_E1_5P_IL = ifelse(diff_Exp_Diet_LocationE2_10P_IL_Exp_Diet_LocationE1_5P_IL == 1, 
                                   lfc_Exp_Diet_LocationE2_10P_IL_Exp_Diet_LocationE1_5P_IL, 0),
           lfc_E2_2.5P_IL_E1_5P_IL = ifelse(diff_Exp_Diet_LocationE2_2.5P_IL_Exp_Diet_LocationE1_5P_IL == 1, 
                                   lfc_Exp_Diet_LocationE2_2.5P_IL_Exp_Diet_LocationE1_5P_IL, 0),
           lfc_E2_5P_IL_E1_5P_IL = ifelse(diff_Exp_Diet_LocationE2_5P_IL_Exp_Diet_LocationE1_5P_IL == 1, 
                                   lfc_Exp_Diet_LocationE2_5P_IL_Exp_Diet_LocationE1_5P_IL, 0),
           lfc_E2_2.5P_IL_E2_10P_IL = ifelse(diff_Exp_Diet_LocationE2_2.5P_IL_Exp_Diet_LocationE2_10P_IL == 1, 
                                   lfc_Exp_Diet_LocationE2_2.5P_IL_Exp_Diet_LocationE2_10P_IL, 0),
           lfc_E2_5P_IL_E2_10P_IL = ifelse(diff_Exp_Diet_LocationE2_5P_IL_Exp_Diet_LocationE2_10P_IL == 1, 
                                   lfc_Exp_Diet_LocationE2_5P_IL_Exp_Diet_LocationE2_10P_IL, 0),
           lfc_E2_5P_IL_E2_2.5P_IL = ifelse(diff_Exp_Diet_LocationE2_5P_IL_Exp_Diet_LocationE2_2.5P_IL == 1, 
                                        lfc_Exp_Diet_LocationE2_5P_IL_Exp_Diet_LocationE2_2.5P_IL, 0)) %>%
    transmute(taxon, 
              `E1_2.5P_IL vs. E1_10P_IL` = round(lfc_E1_2.5P_IL, 2), 
              `E1_5P_IL vs. E1_10P_IL` = round(lfc_E1_5P_IL, 2),
              `E2_10P_IL vs. E1_10P_IL` = round(lfc_E2_10P_IL, 2),
              `E2_5P_IL vs. E1_10P_IL` = round(lfc_E2_5P_IL, 2),
              `E1_5P_IL vs. E1_2.5P_IL` = round(lfc_E1_5P_IL_E1_2.5P_IL, 2),
              `E2_10P_IL vs. E1_2.5P_IL` = round(lfc_E2_10P_IL_E1_2.5P_IL, 2),
              `E2_2.5P_IL vs. E1_2.5P_IL` = round(lfc_E2_2.5P_IL_E1_2.5P_IL, 2),
              `E2_5P_IL vs. E1_2.5P_IL` = round(lfc_E2_5P_IL_E1_2.5P_IL, 2),
              `E2_10P_IL vs. E1_5P_IL` = round(lfc_E2_10P_IL_E1_5P_IL, 2),
              `E2_2.5P_IL vs. E1_5P_IL` = round(lfc_E2_2.5P_IL_E1_5P_IL, 2),
              `E2_5P_IL vs. E1_5P_IL` = round(lfc_E2_5P_IL_E1_5P_IL, 2),
              `E2_2.5P_IL vs. E2_10P_IL` = round(lfc_E2_2.5P_IL_E2_10P_IL, 2),
              `E2_5P_IL vs. E2_10P_IL` = round(lfc_E2_5P_IL_E2_10P_IL, 2),
              `E2_5P_IL vs. E2_2.5P_IL` = round(lfc_E2_5P_IL_E2_2.5P_IL, 2)
              ) %>%
    pivot_longer(cols = `E1_2.5P_IL vs. E1_10P_IL`:`E2_5P_IL vs. E2_2.5P_IL`, 
                 names_to = "group", values_to = "value") %>%
    arrange(taxon)
df_fig_pair$group = factor(df_fig_pair$group, 
                           levels = c("E1_2.5P_IL vs. E1_10P_IL",
                                      "E1_5P_IL vs. E1_10P_IL",
                                      "E2_10P_IL vs. E1_10P_IL",
                                      "E2_5P_IL vs. E1_10P_IL",
                                      "E1_5P_IL vs. E1_2.5P_IL",
                                      "E2_10P_IL vs. E1_2.5P_IL",
                                      "E2_2.5P_IL vs. E1_2.5P_IL",
                                      "E2_5P_IL vs. E1_2.5P_IL",
                                      "E2_10P_IL vs. E1_5P_IL",
                                      "E2_2.5P_IL vs. E1_5P_IL",
                                      "E2_5P_IL vs. E1_5P_IL",
                                      "E2_2.5P_IL vs. E2_10P_IL",
                                      "E2_5P_IL vs. E2_10P_IL",
                                      "E2_5P_IL vs. E2_2.5P_IL"))
  
lo = floor(min(df_fig_pair$value))
up = ceiling(max(df_fig_pair$value))
mid = (lo + up)/2
fig_pair = df_fig_pair %>%
  ggplot(aes(x = group, y = taxon, fill = value)) + 
  geom_tile(color = "black") +
  scale_fill_gradient2(low = "blue", high = "red", mid = "white", 
                       na.value = "white", midpoint = mid, limit = c(lo, up),
                       name = NULL) +
  geom_text(aes(group, taxon, label = value), color = "black", size = 4) +
  labs(x = NULL, y = NULL, title = "Log fold change of pairwise comparisons in the ILenum E1 vs E2 by % Protein") +
  theme_minimal() +
  theme(plot.title = element_text(hjust = 0.5))
fig_pair

#results- no differentially abundant taxa identified
```
Since theres no differentially abundant taxa using pairwise comparisons, lets look at global comparisons
ANCOM-BC2 global test aims to determine taxa that are differentially abundant between at least two groups across three or more experimental groups.

In this example, we want to identify taxa that are differentially abundant between at least two groups across “lean”, “overweight”, and “obese”. The result contains: 1) test statistics, 2) p-values, 3) adjusted p-values, 4) indicators of whether the taxon is differentially abundant (TRUE) or not (FALSE).
#load res_prim
```{r}
E1vsE2_Protein_ILE_res_prim 
```
#load res_blogal
```{r}
E1vsE2_Protein_ILE_res_global
```
#Plot global for ILE! 
```{r}
df_Exp_Diet_Location = E1vsE2_Protein_ILE_res_prim %>%
    dplyr::select(taxon, contains("Exp_Diet_Location")) 
df_fig_global = df_Exp_Diet_Location %>%
    dplyr::left_join(E1vsE2_Protein_ILE_res_global %>%
                  transmute(taxon, diff_Exp_Diet_Location = diff_abn)) %>%
    dplyr::filter(diff_Exp_Diet_Location == 1) %>%
    mutate(lfc_E1_2.5P_IL = ifelse(diff_Exp_Diet_LocationE1_2.5P_IL == 1, 
                             lfc_Exp_Diet_LocationE1_2.5P_IL, 0),
           lfc_E1_5P_IL = ifelse(diff_Exp_Diet_LocationE1_5P_IL == 1, 
                                   lfc_Exp_Diet_LocationE1_5P_IL, 0),
           lfc_E2_10P_IL = ifelse(diff_Exp_Diet_LocationE2_10P_IL == 1, 
                                   lfc_Exp_Diet_LocationE2_10P_IL, 0),
           lfc_E2_2.5P_IL = ifelse(diff_Exp_Diet_LocationE2_2.5P_IL == 1, 
                                   lfc_Exp_Diet_LocationE2_2.5P_IL, 0),
           lfc_E2_5P_IL = ifelse(diff_Exp_Diet_LocationE2_5P_IL == 1, 
                                   lfc_Exp_Diet_LocationE2_5P_IL, 0)) %>%
    transmute(taxon, 
              `E1_2.5P_IL vs. E1_10P_IL` = round(lfc_E1_2.5P_IL, 2), 
              `E1_5P_IL vs. E1_10P_IL` = round(lfc_E1_5P_IL, 2),
              `E2_10P_IL vs. E1_10P_IL` = round(lfc_E2_10P_IL, 2),
              `E2_5P_IL vs. E1_10P_IL` = round(lfc_E2_5P_IL, 2)) %>%
    pivot_longer(cols = `E1_2.5P_IL vs. E1_10P_IL`:`E2_5P_IL vs. E1_10P_IL`, 
                 names_to = "group", values_to = "value") %>%
    arrange(taxon)
  
lo = floor(min(df_fig_global$value))
up = ceiling(max(df_fig_global$value))
mid = (lo + up)/2
fig_global = df_fig_global %>%
  ggplot(aes(x = group, y = taxon, fill = value)) + 
  geom_tile(color = "black") +
  scale_fill_gradient2(low = "blue", high = "red", mid = "white", 
                       na.value = "white", midpoint = mid, limit = c(lo, up),
                       name = NULL) +
  geom_text(aes(group, taxon, label = value), color = "black", size = 4) +
  labs(x = NULL, y = NULL, title = "Log fold changes for globally significant taxa in Ileum") +
  theme(plot.title = element_text(hjust = 0.5)) +
theme_bw() +
  theme(
    plot.title = element_text(hjust = 1),
    title =element_text(size=22, face='bold'),
    axis.text.x = element_text(color="black", size=12, angle = 60, hjust = 1),
    axis.text.y = element_text(color="black", size=12),
    strip.text.y = element_text(color="black", size=12, angle=90),
    #legend.title = element_text(color="black", size=20),
    legend.text = element_text(color="black", size=14),
    legend.key.size = unit(1, 'cm'),
    #legend.position = "right",
    strip.text = element_text(size = 16),
    axis.title.x = element_text(color="black", size=14),
    axis.title.y = element_text(color="black", size=14, angle=90),
    panel.grid.major = element_blank(), # switch off major gridlines
    panel.grid.minor = element_blank(), # switch off minor gridlines
  )
fig_global
#results- no differentially abundant taxa by percent protein and experiment in Ileum
```

#############################

#AMCOMBC E1vsE2 by Percent Protein in the CECUM:

```{r}
set.seed(123)
output = ancombc2(ps_raw_E1_E2_cec, assay_name = NULL, tax_level = NULL,
                  fix_formula = "Exp_Diet_Location", rand_formula = NULL,
                  p_adj_method = "holm", pseudo = 0, pseudo_sens = TRUE,
                  prv_cut = 0.10, lib_cut = 1000, s0_perc = 0.05,
                  group = "Exp_Diet_Location", struc_zero = TRUE, neg_lb = TRUE,
                  alpha = 0.05, n_cl = 2, verbose = TRUE,
                  global = TRUE, pairwise = TRUE, dunnet = TRUE, trend = TRUE,
                  iter_control = list(tol = 1e-2, max_iter = 20, 
                                      verbose = TRUE),
                  em_control = list(tol = 1e-5, max_iter = 100),
                  lme_control = lme4::lmerControl(),
                  mdfdr_control = list(fwer_ctrl_method = "holm", B = 100),
                  trend_control = list(contrast = list(matrix(c(1, 0, 0, 0, 0,
                                                               -1, 1, 0, 0, 0,
                                                                0, -1, 1, 0, 0, 
                                                                0, 0, -1, 1, 0,
                                                                0, 0, 0, -1, 1),
                                                              nrow = 5, 
                                                              byrow = TRUE),
                                                       matrix(c(1, 0, 0, 0, 0,
                                                               -1, 1, 0, 0, 0,
                                                                0, -1, 1, 0, 0, 
                                                                0, 0, -1, 1, 0,
                                                                0, 0, 0, -1, 1),
                                                              nrow = 5, 
                                                              byrow = TRUE)),
                                       node = list(2, 2),
                                       solver = "ECOS",
                                       B = 100))

```
#save the output
```{r}
res_prim = output$res
saveRDS(res_prim, "/Users/connermertz/Documents/Sequence Analysis/E1_E2_All_Samples/codeexports/ANCOMBC/E1vsE2_Protein_CEC_res_prim.rds")
#read in the ANCOMBC output if you want to see E1 vs E2 by Location
E1vsE2_Protein_CEC_res_prim <- readRDS("/Users/connermertz/Documents/Sequence Analysis/E1_E2_All_Samples/codeexports/ANCOMBC/E1vsE2_Protein_CEC_res_prim.rds")

res_pair = output$res_pair
saveRDS(res_pair, "/Users/connermertz/Documents/Sequence Analysis/E1_E2_All_Samples/codeexports/ANCOMBC/E1vsE2_Protein_CEC_res_pair.rds")
E1vsE2_Protein_CEC_res_pair <- readRDS("/Users/connermertz/Documents/Sequence Analysis/E1_E2_All_Samples/codeexports/ANCOMBC/E1vsE2_Protein_CEC_res_pair.rds")

res_global = output$res_global
saveRDS(res_global, "/Users/connermertz/Documents/Sequence Analysis/E1_E2_All_Samples/codeexports/ANCOMBC/E1vsE2_Protein_CEC_res_global.rds")
E1vsE2_Protein_CEC_res_global <- readRDS("/Users/connermertz/Documents/Sequence Analysis/E1_E2_All_Samples/codeexports/ANCOMBC/E1vsE2_Protein_CEC_res_global.rds")

res_trend = output$res_trend
saveRDS(res_trend, "/Users/connermertz/Documents/Sequence Analysis/E1_E2_All_Samples/codeexports/ANCOMBC/E1vsE2_Protein_CEC_res_trend.rds")
E1vsE2_Protein_CEC_res_trend <- readRDS("/Users/connermertz/Documents/Sequence Analysis/E1_E2_All_Samples/codeexports/ANCOMBC/E1vsE2_Protein_CEC_res_trend.rds")
```

ANCOM-BC2 global test aims to determine taxa that are differentially abundant between at least two groups across three or more experimental groups.

In this example, we want to identify taxa that are differentially abundant between at least two groups across “lean”, “overweight”, and “obese”. The result contains: 1) test statistics, 2) p-values, 3) adjusted p-values, 4) indicators of whether the taxon is differentially abundant (TRUE) or not (FALSE).
#look at res_prim
```{r}
E1vsE2_Protein_CEC_res_prim
```
#look at global table
```{r}
E1vsE2_Protein_CEC_res_global
```

#Plot it!
```{r}
df_Exp_Diet_Location = E1vsE2_Protein_CEC_res_prim %>%
    dplyr::select(taxon, contains("Exp_Diet_Location")) 
df_fig_global = df_Exp_Diet_Location %>%
    dplyr::left_join(E1vsE2_Protein_CEC_res_global %>%
                  transmute(taxon, diff_Exp_Diet_Location = diff_abn)) %>%
    dplyr::filter(diff_Exp_Diet_Location == 1) %>%
    mutate(lfc_E1_2.5P_CEC = ifelse(diff_Exp_Diet_LocationE1_2.5P_CEC == 1, 
                             lfc_Exp_Diet_LocationE1_2.5P_CEC, 0),
           lfc_E1_5P_CEC = ifelse(diff_Exp_Diet_LocationE1_5P_CEC == 1, 
                                   lfc_Exp_Diet_LocationE1_5P_CEC, 0),
           lfc_E2_10P_CEC = ifelse(diff_Exp_Diet_LocationE2_10P_CEC == 1, 
                                   lfc_Exp_Diet_LocationE2_10P_CEC, 0),
           lfc_E2_2.5P_CEC = ifelse(diff_Exp_Diet_LocationE2_2.5P_CEC == 1, 
                                   lfc_Exp_Diet_LocationE2_2.5P_CEC, 0),
           lfc_E2_5P_CEC = ifelse(diff_Exp_Diet_LocationE2_5P_CEC == 1, 
                                   lfc_Exp_Diet_LocationE2_5P_CEC, 0)) %>%
    transmute(taxon, 
              `E1_2.5P_CEC vs. E1_10P_CEC` = round(lfc_E1_2.5P_CEC, 2), 
              `E1_5P_CEC vs. E1_10P_CEC` = round(lfc_E1_5P_CEC, 2),
              `E2_10P_CEC vs. E1_10P_CEC` = round(lfc_E2_10P_CEC, 2),
              `E2_5P_CEC vs. E1_10P_CEC` = round(lfc_E2_5P_CEC, 2)) %>%
    pivot_longer(cols = `E1_2.5P_CEC vs. E1_10P_CEC`:`E2_5P_CEC vs. E1_10P_CEC`, 
                 names_to = "group", values_to = "value") %>%
    arrange(taxon)
  
lo = floor(min(df_fig_global$value))
up = ceiling(max(df_fig_global$value))
mid = (lo + up)/2
fig_global = df_fig_global %>%
  ggplot(aes(x = group, y = taxon, fill = value)) + 
  geom_tile(color = "black") +
  scale_fill_gradient2(low = "blue", high = "red", mid = "white", 
                       na.value = "white", midpoint = mid, limit = c(lo, up),
                       name = NULL) +
  geom_text(aes(group, taxon, label = value), color = "black", size = 4) +
  labs(x = NULL, y = NULL, title = "Log fold changes for globally significant taxa in Cecum") +
  theme(plot.title = element_text(hjust = 0.5)) +
theme_bw() +
  theme(
    plot.title = element_text(hjust = 1),
    title =element_text(size=22, face='bold'),
    axis.text.x = element_text(color="black", size=12, angle = 60, hjust = 1),
    axis.text.y = element_text(color="black", size=12),
    strip.text.y = element_text(color="black", size=12, angle=90),
    #legend.title = element_text(color="black", size=20),
    legend.text = element_text(color="black", size=14),
    legend.key.size = unit(1, 'cm'),
    #legend.position = "right",
    strip.text = element_text(size = 16),
    axis.title.x = element_text(color="black", size=14),
    axis.title.y = element_text(color="black", size=14, angle=90),
    panel.grid.major = element_blank(), # switch off major gridlines
    panel.grid.minor = element_blank(), # switch off minor gridlines
  )
#CREATE A PDF AND SAVES IT TO THE RIGHT DIRECTORY/FOLDER. 
pdf("/Users/connermertz/Documents/Conner Documents/UNM biology/Chapter 1 E1 vs E2/ANCOMoutput/E1vsE2_Protein_CEC_res_global_figure.pdf", height = 12, width = 14)
fig_global
dev.off()

#results: ASV7 is the only differentially abundant ASV and shows up when looking at E2_10P_CEC vs E1_10P_CEC (differentially abundant in E1_10P_CEC by -5.59 lfc)
```

##ANCOM-BC2 pairwise directional test
ANCOM-BC2 pairwise directional test aims to determine taxa that are differentially abundant between any pair of two groups across three or more experimental groups, while controlling the mdFDR.

In this example, we want to identify taxa that are differentially abundant between any pair of two groups across “lean”, “overweight”, and “obese”. The result contains: 1) log fold changes, 2) standard errors, 3) test statistics, 4) p-values, 5) adjusted p-values, 6) indicators of whether the taxon is differentially abundant (TRUE) or not (FALSE).
```{r}
E1vsE2_Protein_CEC_res_pair
```

#plot Pairwise tests
```{r}
df_fig_pair = E1vsE2_Protein_CEC_res_pair %>%
    filter(diff_Exp_Diet_LocationE1_2.5P_CEC == 1 | 
             diff_Exp_Diet_LocationE1_5P_CEC == 1 |
             diff_Exp_Diet_LocationE2_10P_CEC == 1 |
             diff_Exp_Diet_LocationE2_2.5P_CEC == 1 |
             diff_Exp_Diet_LocationE2_5P_CEC == 1 |
             diff_Exp_Diet_LocationE1_5P_CEC_Exp_Diet_LocationE1_2.5P_CEC == 1 |
             diff_Exp_Diet_LocationE2_10P_CEC_Exp_Diet_LocationE1_2.5P_CEC == 1 |
             diff_Exp_Diet_LocationE2_2.5P_CEC_Exp_Diet_LocationE1_2.5P_CEC == 1 |
             diff_Exp_Diet_LocationE2_5P_CEC_Exp_Diet_LocationE1_2.5P_CEC == 1 |
             diff_Exp_Diet_LocationE2_10P_CEC_Exp_Diet_LocationE1_5P_CEC == 1 |
             diff_Exp_Diet_LocationE2_2.5P_CEC_Exp_Diet_LocationE1_5P_CEC == 1 |
             diff_Exp_Diet_LocationE2_5P_CEC_Exp_Diet_LocationE1_5P_CEC == 1 |
             diff_Exp_Diet_LocationE2_2.5P_CEC_Exp_Diet_LocationE2_10P_CEC == 1 |
             diff_Exp_Diet_LocationE2_5P_CEC_Exp_Diet_LocationE2_10P_CEC == 1 |
             diff_Exp_Diet_LocationE2_5P_CEC_Exp_Diet_LocationE2_2.5P_CEC == 1) %>%
    mutate(lfc_E1_2.5P_CEC = ifelse(diff_Exp_Diet_LocationE1_2.5P_CEC == 1, 
                             lfc_Exp_Diet_LocationE1_2.5P_CEC, 0),
           lfc_E1_5P_CEC = ifelse(diff_Exp_Diet_LocationE1_5P_CEC == 1, 
                                   lfc_Exp_Diet_LocationE1_5P_CEC, 0),
           lfc_E2_10P_CEC = ifelse(diff_Exp_Diet_LocationE2_10P_CEC == 1, 
                                   lfc_Exp_Diet_LocationE2_10P_CEC, 0),
           lfc_E2_2.5P_CEC = ifelse(diff_Exp_Diet_LocationE2_2.5P_CEC == 1, 
                                   lfc_Exp_Diet_LocationE2_2.5P_CEC, 0),
           lfc_E2_5P_CEC = ifelse(diff_Exp_Diet_LocationE2_5P_CEC == 1, 
                                   lfc_Exp_Diet_LocationE2_5P_CEC, 0),
           lfc_E1_5P_CEC_E1_2.5P_CEC = ifelse(diff_Exp_Diet_LocationE1_5P_CEC_Exp_Diet_LocationE1_2.5P_CEC == 1, 
                                   lfc_Exp_Diet_LocationE1_5P_CEC_Exp_Diet_LocationE1_2.5P_CEC, 0),
           lfc_E2_10P_CEC_E1_2.5P_CEC = ifelse(diff_Exp_Diet_LocationE2_10P_CEC_Exp_Diet_LocationE1_2.5P_CEC == 1, 
                                   lfc_Exp_Diet_LocationE2_10P_CEC_Exp_Diet_LocationE1_2.5P_CEC, 0),
           lfc_E2_2.5P_CEC_E1_2.5P_CEC = ifelse(diff_Exp_Diet_LocationE2_2.5P_CEC_Exp_Diet_LocationE1_2.5P_CEC == 1, 
                                   lfc_Exp_Diet_LocationE2_2.5P_CEC_Exp_Diet_LocationE1_2.5P_CEC, 0),
           lfc_E2_5P_CEC_E1_2.5P_CEC = ifelse(diff_Exp_Diet_LocationE2_5P_CEC_Exp_Diet_LocationE1_2.5P_CEC == 1, 
                                   lfc_Exp_Diet_LocationE2_5P_CEC_Exp_Diet_LocationE1_2.5P_CEC, 0),
           lfc_E2_10P_CEC_E1_5P_CEC = ifelse(diff_Exp_Diet_LocationE2_10P_CEC_Exp_Diet_LocationE1_5P_CEC == 1, 
                                   lfc_Exp_Diet_LocationE2_10P_CEC_Exp_Diet_LocationE1_5P_CEC, 0),
           lfc_E2_2.5P_CEC_E1_5P_CEC = ifelse(diff_Exp_Diet_LocationE2_2.5P_CEC_Exp_Diet_LocationE1_5P_CEC == 1, 
                                   lfc_Exp_Diet_LocationE2_2.5P_CEC_Exp_Diet_LocationE1_5P_CEC, 0),
           lfc_E2_5P_CEC_E1_5P_CEC = ifelse(diff_Exp_Diet_LocationE2_5P_CEC_Exp_Diet_LocationE1_5P_CEC == 1, 
                                   lfc_Exp_Diet_LocationE2_5P_CEC_Exp_Diet_LocationE1_5P_CEC, 0),
           lfc_E2_2.5P_CEC_E2_10P_CEC = ifelse(diff_Exp_Diet_LocationE2_2.5P_CEC_Exp_Diet_LocationE2_10P_CEC == 1, 
                                   lfc_Exp_Diet_LocationE2_2.5P_CEC_Exp_Diet_LocationE2_10P_CEC, 0),
           lfc_E2_5P_CEC_E2_10P_CEC = ifelse(diff_Exp_Diet_LocationE2_5P_CEC_Exp_Diet_LocationE2_10P_CEC == 1, 
                                   lfc_Exp_Diet_LocationE2_5P_CEC_Exp_Diet_LocationE2_10P_CEC, 0),
           lfc_E2_5P_CEC_E2_2.5P_CEC = ifelse(diff_Exp_Diet_LocationE2_5P_CEC_Exp_Diet_LocationE2_2.5P_CEC == 1, 
                                        lfc_Exp_Diet_LocationE2_5P_CEC_Exp_Diet_LocationE2_2.5P_CEC, 0)) %>%
    transmute(taxon, 
              `E1_2.5P_CEC vs. E1_10P_CEC` = round(lfc_E1_2.5P_CEC, 2), 
              `E1_5P_CEC vs. E1_10P_CEC` = round(lfc_E1_5P_CEC, 2),
              `E2_10P_CEC vs. E1_10P_CEC` = round(lfc_E2_10P_CEC, 2),
              `E2_5P_CEC vs. E1_10P_CEC` = round(lfc_E2_5P_CEC, 2),
              `E1_5P_CEC vs. E1_2.5P_CEC` = round(lfc_E1_5P_CEC_E1_2.5P_CEC, 2),
              `E2_10P_CEC vs. E1_2.5P_CEC` = round(lfc_E2_10P_CEC_E1_2.5P_CEC, 2),
              `E2_2.5P_CEC vs. E1_2.5P_CEC` = round(lfc_E2_2.5P_CEC_E1_2.5P_CEC, 2),
              `E2_5P_CEC vs. E1_2.5P_CEC` = round(lfc_E2_5P_CEC_E1_2.5P_CEC, 2),
              `E2_10P_CEC vs. E1_5P_CEC` = round(lfc_E2_10P_CEC_E1_5P_CEC, 2),
              `E2_2.5P_CEC vs. E1_5P_CEC` = round(lfc_E2_2.5P_CEC_E1_5P_CEC, 2),
              `E2_5P_CEC vs. E1_5P_CEC` = round(lfc_E2_5P_CEC_E1_5P_CEC, 2),
              `E2_2.5P_CEC vs. E2_10P_CEC` = round(lfc_E2_2.5P_CEC_E2_10P_CEC, 2),
              `E2_5P_CEC vs. E2_10P_CEC` = round(lfc_E2_5P_CEC_E2_10P_CEC, 2),
              `E2_5P_CEC vs. E2_2.5P_CEC` = round(lfc_E2_5P_CEC_E2_2.5P_CEC, 2)
              ) %>%
    pivot_longer(cols = `E1_2.5P_CEC vs. E1_10P_CEC`:`E2_5P_CEC vs. E2_2.5P_CEC`, 
                 names_to = "group", values_to = "value") %>%
    arrange(taxon)
df_fig_pair$group = factor(df_fig_pair$group, 
                           levels = c("E1_2.5P_CEC vs. E1_10P_CEC",
                                      "E1_5P_CEC vs. E1_10P_CEC",
                                      "E2_10P_CEC vs. E1_10P_CEC",
                                      "E2_5P_CEC vs. E1_10P_CEC",
                                      "E1_5P_CEC vs. E1_2.5P_CEC",
                                      "E2_10P_CEC vs. E1_2.5P_CEC",
                                      "E2_2.5P_CEC vs. E1_2.5P_CEC",
                                      "E2_5P_CEC vs. E1_2.5P_CEC",
                                      "E2_10P_CEC vs. E1_5P_CEC",
                                      "E2_2.5P_CEC vs. E1_5P_CEC",
                                      "E2_5P_CEC vs. E1_5P_CEC",
                                      "E2_2.5P_CEC vs. E2_10P_CEC",
                                      "E2_5P_CEC vs. E2_10P_CEC",
                                      "E2_5P_CEC vs. E2_2.5P_CEC"))
  
lo = floor(min(df_fig_pair$value))
up = ceiling(max(df_fig_pair$value))
mid = (lo + up)/2
fig_pair = df_fig_pair %>%
  ggplot(aes(x = group, y = taxon, fill = value)) + 
  geom_tile(color = "black") +
  scale_fill_gradient2(low = "blue", high = "red", mid = "white", 
                       na.value = "white", midpoint = mid, limit = c(lo, up),
                       name = NULL) +
  geom_text(aes(group, taxon, label = value), color = "black", size = 4) +
  labs(x = NULL, y = NULL, title = "Log fold change of pairwise comparisons in the Cecum E1 vs E2 by % Protein") +
  theme_minimal() +
  theme(plot.title = element_text(hjust = 0.5))
fig_pair

#results- no differentially abundant taxa identified
```


Colors!
```{r}
#first you want to factor your items. These are the names of the category in the order I want them to be every time. 
Meta$Exp_Diet_Location <-factor(Meta$Exp_Diet_Location, levels=c("E1_2.5P_CEC", "E1_5P_CEC", "E1_10P_CEC","E2_2.5P_CEC", "E2_5P_CEC", "E2_10P_CEC"))

#now create an object that is a list of colors in a certain order.The colors I chose are "color-blind friendly":
Exp_Diet_Scale <- c("#e6550d", "#fdae6b","#fee6ce","#3182bd", "#9ecae1","#deebf7")
#scales::show_col(DietScale)

#now we want to merge the two. We are taking the order from the locations and telling it that it needs to match the order from the cbPalette.
names(Exp_Diet_Scale) <- levels(Meta$Exp_Diet_Location)
#Have you ever used the scale_colour_manual command? Now we can combine that command and all of the stuff we want in it into a new object, LocationColors. At the end of your ggplot code, just add a +DietColor (or whatever you decide to name it), and it should pull the colors you made! 
Exp_DietColors<- scale_colour_manual(name="Exp_Diet_Location", values=Exp_Diet_Scale)

```





################
###############
###############


############
October 2025 CMM added additional ANCOMBC2 analyses to address R1 comments. Here we performed with gut section analysis by diet type and percent protein. 

ANCOM
Analysis of Compositions of Microbiomes with Bias Correction (ANCOM-BC) (Lin and Peddada 2020) is a methodology of differential abundance (DA) analysis for microbial absolute abundances. ANCOM-BC estimates the unknown sampling fractions, corrects the bias induced by their differences through a log linear regression model including the estimated sampling fraction as an offset terms, and identifies taxa that are differentially abundant according to the variable of interest. For more details, please refer to the ANCOM-BC paper.
official tutorial: https://bioconductor.org/packages/release/bioc/vignettes/ANCOMBC/inst/doc/ANCOMBC2.html

https://bioconductor.org/packages/devel/bioc/vignettes/ANCOMBC/inst/doc/ANCOMBC.html

a helpful tutorial: https://www.yanh.org/2021/01/01/microbiome-r/#differential-abundance-analysis

ANCOM manual: https://bioconductor.org/packages/release/bioc/manuals/ANCOMBC/man/ANCOMBC.pdf

E1_gut section by percnt protein
```{r setup, include=FALSE}
knitr::opts_chunk$set(message = FALSE, warning = FALSE, comment = NA, 
                      fig.width = 6.25, fig.height = 5)
library(ANCOMBC)
library(tidyverse)
library(caret)
library(DT)
options(DT.options = list(
  initComplete = JS("function(settings, json) {",
  "$(this.api().table().header()).css({'background-color': 
  '#000', 'color': '#fff'});","}")))
```

#LOAD OTHER PACKAGES THAT MIGHT BE USEFUL
```{r}
#load packages
# = this package is definitely needed
library(BiocManager)
library(phyloseq) #
library(ggplot2) #
library(randomForest)
library(vegan)
library(caret)
library(gridExtra) #
library(ggpubr) 
library(ggthemes) #
library(vegan) #
library(MASS)
library(cluster)
library(RColorBrewer) #
library(grid)
library(cooccur)
library(igraph)
library(Hmisc)
#library(minerva)
library(parallel)
library(corrplot)
library(tidyverse) #
library(plyr);library(dplyr) #
library(tidyr) #
library(usdm) #
library(decontam) #
library(ecodist) #
library(betapart) #
#library(psych)
#library(ggbiplot)
library(ANCOMBC) #
library(DT) #
library(gplots) #
library(remotes)
library(metagMisc)
library(iNEXT)
library("DESeq2")
packageVersion("DESeq2")
library("reshape2")
library("remotes")
library("phyloseq.extended")
#library(ggvegan) 
#library(ggpubfigs)
#library(agricolae) #this package is not compatable with this version of R, if we really need it we will need to install new version of package
library(RColorBrewer)
library(scales)
library("devtools")

theme_set(theme_bw()) #sets ggplot theme
```


#load metadata
```{r}
Meta <- read.csv("/Users/connermertz/Dropbox (vlab)/Conner Mertz/UNM biology/Chapter 1 E1 vs E2/Methods/16S rRNA gene sequencing/Sequencing_workup/E1_E2_Meta_ALL.csv") #you should have 333 total samples
row.names(Meta) <- Meta$Sample_ID
```

#read in ps objects
```{r}
ps_raw <- readRDS("/Users/connermertz/Dropbox (vlab)/Conner Mertz/UNM biology/Chapter 1 E1 vs E2/Methods/16S rRNA gene sequencing/Sequencing_workup/DADA2_output/clean_ps.rds") #raw but with choloro,mito, low reads (less than 2000), and sample duplicates removed
ps_all <- readRDS("/Users/connermertz/Dropbox (vlab)/Conner Mertz/UNM biology/Chapter 1 E1 vs E2/Methods/16S rRNA gene sequencing/Sequencing_workup/RarefyOutput/ps_rare.rds") #this is my phyloseq object I will be using for the majority of my analysis. It contains all samples from E1 and E2 (duplicates not included) and has been rarefied at lib size of 8000.
sample_data(ps_raw) <- Meta #If you beed to update your meta sheet, this is how to do it without dunning dada2 again to create a whole new ps
sample_data(ps_all) <- Meta #If you beed to update your meta sheet, this is how to do it without dunning dada2 again to create a whole new ps
```


```{r}
#read in un-normalized- unrarefied phyloseq object from rds (ANCOM normalizes for you, and if you give it normalized data it will give bias)
#make relevant subsets

### Make new phyloseq object of just E1, no duplicates
E1_ps_raw <- subset_samples(ps_raw, Experiment=="E1")


#Subset by E1 2.5P,5P, 10P only to compare to E2; no duplicates
ps_raw_E1_forcomparison <- subset_samples(E1_ps_raw, Percent_Protein %in% c("2.5P", "5P", "10P"))
ps_raw_E1_forcomparison <- subset_samples(ps_raw_E1_forcomparison, Location %in% c("Duodenum", "Ileum", "Cecum"))
ps_raw_E1_forcomparison

#Subset by E1 and E2 2.5P,5P, 10P; no duplicates
ps_raw_E1_E2_forcomparison <- subset_samples(ps_raw, Percent_Protein %in% c("2.5P", "5P", "10P"))
ps_raw_E1_E2_forcomparison <- subset_samples(ps_raw_E1_E2_forcomparison, Location %in% c("Duodenum", "Ileum", "Cecum"))
ps_raw_E1_E2_forcomparison


### Make new phyloseq object of just E2, no duplicates
E2_ps_raw <- subset_samples(ps_raw, Experiment=="E2")   


#Subset by E1_E2_Location
ps_raw_E1_E2_duod <- subset_samples(ps_raw_E1_E2_forcomparison, Location %in% c("Duodenum"))
ps_raw_E1_E2_ile <- subset_samples(ps_raw_E1_E2_forcomparison, Location %in% c("Ileum"))
ps_raw_E1_E2_cec <- subset_samples(ps_raw_E1_E2_forcomparison, Location %in% c("Cecum"))

#Subset by Experiment and by Location
ps_raw_E1_duod <- subset_samples(ps_raw_E1_forcomparison, Location %in% c("Duodenum"))
ps_raw_E1_ile <- subset_samples(ps_raw_E1_forcomparison, Location %in% c("Ileum"))
ps_raw_E1_cec <- subset_samples(ps_raw_E1_forcomparison, Location %in% c("Cecum"))
ps_raw_E2_duod <- subset_samples(E2_ps_raw, Location %in% c("Duodenum"))
ps_raw_E2_ile <- subset_samples(E2_ps_raw, Location %in% c("Ileum"))
ps_raw_E2_cec <- subset_samples(E2_ps_raw, Location %in% c("Cecum"))

```

Now lets look at whether we can see %protein differences within each gut section
```{r}
output = ancombc2(
  ps_raw_E1_duod,
  assay_name = NULL,
  tax_level = NULL, #keep null if you want at ASV level
  fix_formula= "Exp_Diet",
  rand_formula = NULL,
  p_adj_method = "holm",
  prv_cut = 0.1,
  lib_cut = 0,
  s0_perc = 0.05,
  group = "Exp_Diet", #leave as null if group variable of interest contains only two categories 
  struc_zero = FALSE,
  neg_lb = FALSE,
  alpha = 0.05,
  n_cl = 1,
  verbose = FALSE,
  global = TRUE,
  pairwise = TRUE,
  dunnet = TRUE,
  trend = FALSE,
  iter_control = list(tol = 0.01, max_iter = 20, verbose = FALSE),
  em_control = list(tol = 1e-05, max_iter = 100),
  lme_control = lme4::lmerControl(),
  mdfdr_control = list(fwer_ctrl_method = "holm", B = 100),
  trend_control = list(contrast = NULL, node = NULL, solver = "ECOS", B = 100)
)
```

Save the output
```{r}
#The primary output of the ANCOM-BC2 methodology (res) identifies taxa with differential abundance based on the chosen covariate. The results include: 1) log fold changes, 2) standard errors, 3) test statistics, 4) p-values, 5) adjusted p-values, 6) indicators denoting whether the taxon is differentially abundant (TRUE) or not (FALSE), and 7) indicators denoting whether the taxon passed the sensitivity analysis (TRUE) or not (FALSE).

res_pair = output$res_pair
saveRDS(res_pair, "/Users/connermertz/vlab Dropbox/Vesbach Lab/Conner Mertz/UNM biology/Chapter 1 E1 vs E2/Methods/16S rRNA gene sequencing/Sequencing_workup/ANCOMoutput/ISMERevisions/E1_duod_by_Percent_Protein_res_pair.rds")
E1_duod_by_Percent_Protein_res_pair <- readRDS("/Users/connermertz/vlab Dropbox/Vesbach Lab/Conner Mertz/UNM biology/Chapter 1 E1 vs E2/Methods/16S rRNA gene sequencing/Sequencing_workup/ANCOMoutput/ISMERevisions/E1_duod_by_Percent_Protein_res_pair.rds")
```


##ANCOM-BC2 pairwise directional test
ANCOM-BC2 pairwise directional test aims to determine taxa that are differentially abundant between any pair of two groups across three or more experimental groups, while controlling the mdFDR.

```{r}
#save as csv so that you can interjoin with metadata
E1_duod_by_Percent_Protein_res_pair

write.csv(E1_duod_by_Percent_Protein_res_pair, "/Users/connermertz/vlab Dropbox/Vesbach Lab/Conner Mertz/UNM biology/Chapter 1 E1 vs E2/Methods/16S rRNA gene sequencing/Sequencing_workup/ANCOMoutput/ISMERevisions/E1_duod_by_Percent_Protein_res_pair_edit.csv")

#read in your res_pair csv: 
E1_duod_by_Percent_Protein_res_pair2 <- read.csv("/Users/connermertz/vlab Dropbox/Vesbach Lab/Conner Mertz/UNM biology/Chapter 1 E1 vs E2/Methods/16S rRNA gene sequencing/Sequencing_workup/ANCOMoutput/ISMERevisions/E1_duod_by_Percent_Protein_res_pair_edit.csv")
#change name so target_ids match on both files
names(E1_duod_by_Percent_Protein_res_pair2)[2] ="taxon" 

#read in your Taxonomy classification file 
Taxonomy_classification <- read.csv("/Users/connermertz/Dropbox (vlab)/Conner Mertz/UNM biology/Chapter 1 E1 vs E2/Methods/16S rRNA gene sequencing/Sequencing_workup/DADA2_output/ALL_taxa.csv") #if this doesn't load be sure to download the file from the cloud, then it will work
#change name so target_ids match on both files
names(Taxonomy_classification)[1] ="taxon"

#now we need to get our gapmind results and join it with our Taxonomy results from CAT (combining two data frames)
#we only want to keep the genes that showed up with gapmind, we don't need the other genes that showed up
Part_1 <- Taxonomy_classification %>% 
  inner_join (E1_duod_by_Percent_Protein_res_pair2, by = "taxon")

write.csv(Part_1,"/Users/connermertz/vlab Dropbox/Vesbach Lab/Conner Mertz/UNM biology/Chapter 1 E1 vs E2/Methods/16S rRNA gene sequencing/Sequencing_workup/ANCOMoutput/ISMERevisions/E1vsE2_Duod_Exp_Diet_res_pair_AND_Taxonomy.csv") # went in here and edited file to my liking
```





#E1_PROTEIN_in duod
ANCOMBC2
pairwise comparisons by experiment and PROTEIN
```{r}
#create your data frame for log fold change of pairwise comparisons for dot plot! We will use this to graph dot plots of bargraphs
df_res_pair = E1_duod_by_Percent_Protein_res_pair %>%
  dplyr::filter(diff_Exp_DietE1_2.5P == 1) %>%
  mutate(lfc_E1_2.5P = ifelse(diff_Exp_DietE1_2.5P == 1, 
                             lfc_Exp_DietE1_2.5P, 0)) %>%
              transmute(taxon,
              lfc = lfc_E1_2.5P,
              se = se_Exp_DietE1_2.5P,
              p_val= p_Exp_DietE1_2.5P,
              group = "E1_2.5P - E1_10P") %>%
              
  bind_rows(
  E1_duod_by_Percent_Protein_res_pair %>%
  dplyr::filter(diff_Exp_DietE1_5P == 1) %>%
  mutate(lfc_E1_5P = ifelse(diff_Exp_DietE1_5P == 1, 
                             lfc_Exp_DietE1_5P, 0)) %>%
            transmute(taxon,
                      lfc = lfc_E1_5P,
                      se = se_Exp_DietE1_5P,
                      p_val = p_Exp_DietE1_5P,
                      group = "E1_5P - E1_10P")
    ) %>%                 
    bind_rows(
    E1_duod_by_Percent_Protein_res_pair %>%
    dplyr::filter(diff_Exp_DietE1_5P_Exp_DietE1_2.5P == 1) %>%
    mutate(lfc_E1_5P_E1_2.5P = ifelse(diff_Exp_DietE1_5P_Exp_DietE1_2.5P == 1, 
                                   lfc_Exp_DietE1_5P_Exp_DietE1_2.5P, 0)) %>%
                    transmute(taxon,
                      lfc = lfc_E1_5P_E1_2.5P,
                      se = se_Exp_DietE1_5P_Exp_DietE1_2.5P,
                      p_val = p_Exp_DietE1_5P_Exp_DietE1_2.5P,
                      group = "E1_5P - E1_2.5P")
    )
      df_res_pair$group = factor(df_res_pair$group, 
                            levels = c("E1_2.5P - E1_10P",
                                      "E1_5P - E1_10P",
                                      "E1_5P - E1_2.5P"
                                      ))


```

Save the df you created
```{r}
write.csv(df_res_pair, "/Users/connermertz/vlab Dropbox/Vesbach Lab/Conner Mertz/UNM biology/Chapter 1 E1 vs E2/Methods/16S rRNA gene sequencing/Sequencing_workup/ANCOMoutput/ISMERevisions/E1_duod_by_Percent_Protein_df_res_pair.csv") #ADD IN OTHER CATEGORIES YOU WANT TO GRAPH 
```

there were no ASVs that are differentially abundant in the duodenum of mice fed the synthetic diet by percent protein. 



Now lets look at whether we can see %protein differences in the Ileum

```{r}
output = ancombc2(
  ps_raw_E1_ile,
  assay_name = NULL,
  tax_level = NULL, #keep null if you want at ASV level
  fix_formula= "Exp_Diet",
  rand_formula = NULL,
  p_adj_method = "holm",
  prv_cut = 0.1,
  lib_cut = 0,
  s0_perc = 0.05,
  group = "Exp_Diet", #leave as null if group variable of interest contains only two categories 
  struc_zero = FALSE,
  neg_lb = FALSE,
  alpha = 0.05,
  n_cl = 1,
  verbose = FALSE,
  global = TRUE,
  pairwise = TRUE,
  dunnet = TRUE,
  trend = FALSE,
  iter_control = list(tol = 0.01, max_iter = 20, verbose = FALSE),
  em_control = list(tol = 1e-05, max_iter = 100),
  lme_control = lme4::lmerControl(),
  mdfdr_control = list(fwer_ctrl_method = "holm", B = 100),
  trend_control = list(contrast = NULL, node = NULL, solver = "ECOS", B = 100)
)
```

Save the output
```{r}
#The primary output of the ANCOM-BC2 methodology (res) identifies taxa with differential abundance based on the chosen covariate. The results include: 1) log fold changes, 2) standard errors, 3) test statistics, 4) p-values, 5) adjusted p-values, 6) indicators denoting whether the taxon is differentially abundant (TRUE) or not (FALSE), and 7) indicators denoting whether the taxon passed the sensitivity analysis (TRUE) or not (FALSE).

res_pair = output$res_pair
saveRDS(res_pair, "/Users/connermertz/vlab Dropbox/Vesbach Lab/Conner Mertz/UNM biology/Chapter 1 E1 vs E2/Methods/16S rRNA gene sequencing/Sequencing_workup/ANCOMoutput/ISMERevisions/E1_ile_by_Percent_Protein_res_pair.rds")
E1_ile_by_Percent_Protein_res_pair <- readRDS("/Users/connermertz/vlab Dropbox/Vesbach Lab/Conner Mertz/UNM biology/Chapter 1 E1 vs E2/Methods/16S rRNA gene sequencing/Sequencing_workup/ANCOMoutput/ISMERevisions/E1_ile_by_Percent_Protein_res_pair.rds")
```


##ANCOM-BC2 pairwise directional test
ANCOM-BC2 pairwise directional test aims to determine taxa that are differentially abundant between any pair of two groups across three or more experimental groups, while controlling the mdFDR.


#E1_PROTEIN_in ile
ANCOMBC2 pairwise comparisons 
```{r}
#create your data frame for log fold change of pairwise comparisons for dot plot! We will use this to graph dot plots of bargraphs
df_res_pair = E1_ile_by_Percent_Protein_res_pair %>%
  dplyr::filter(diff_Exp_DietE1_2.5P == 1) %>%
  mutate(lfc_E1_2.5P = ifelse(diff_Exp_DietE1_2.5P == 1, 
                              lfc_Exp_DietE1_2.5P, 0)) %>%
  transmute(taxon,
            lfc = lfc_E1_2.5P,
            se = se_Exp_DietE1_2.5P,
            p_val= p_Exp_DietE1_2.5P,
            group = "E1_2.5P - E1_10P") %>%
  
  bind_rows(
    E1_ile_by_Percent_Protein_res_pair %>%
      dplyr::filter(diff_Exp_DietE1_5P == 1) %>%
      mutate(lfc_E1_5P = ifelse(diff_Exp_DietE1_5P == 1, 
                                lfc_Exp_DietE1_5P, 0)) %>%
      transmute(taxon,
                lfc = lfc_E1_5P,
                se = se_Exp_DietE1_5P,
                p_val = p_Exp_DietE1_5P,
                group = "E1_5P - E1_10P")
  ) %>%                 
  bind_rows(
    E1_ile_by_Percent_Protein_res_pair %>%
      dplyr::filter(diff_Exp_DietE1_5P_Exp_DietE1_2.5P == 1) %>%
      mutate(lfc_E1_5P_E1_2.5P = ifelse(diff_Exp_DietE1_5P_Exp_DietE1_2.5P == 1, 
                                        lfc_Exp_DietE1_5P_Exp_DietE1_2.5P, 0)) %>%
      transmute(taxon,
                lfc = lfc_E1_5P_E1_2.5P,
                se = se_Exp_DietE1_5P_Exp_DietE1_2.5P,
                p_val = p_Exp_DietE1_5P_Exp_DietE1_2.5P,
                group = "E1_5P - E1_2.5P")
  )
df_res_pair$group = factor(df_res_pair$group, 
                           levels = c("E1_2.5P - E1_10P",
                                      "E1_5P - E1_10P",
                                      "E1_5P - E1_2.5P"
                           ))


```

Save the df you created
```{r}
#write.csv(df_res_pair, "/Users/connermertz/vlab Dropbox/Vesbach Lab/Conner Mertz/UNM biology/Chapter 1 E1 vs E2/Methods/16S rRNA gene sequencing/Sequencing_workup/ANCOMoutput/ISMERevisions/E1_ile_by_Percent_Protein_df_res_pair.csv") #ADD IN OTHER CATEGORIES YOU WANT TO GRAPH 
```

there were 2 ASVs that are differentially abundant in the ileum of mice fed the synthetic diet by percent protein.
E1_ile
E1_2.5P vs E1_10P: ASV80 is differentially abundant in E1_10P 
E1_5P vs E1_2.5P: ASV80 is differentially abundant in E1_5P
E1_5P vs E1_10P: no ASVs differentially abundant identified

Save the df you created
```{r}
#read in your EDITED DF csv: 
E1_ile_by_Percent_Protein_df_res_pair <- read.csv("/Users/connermertz/vlab Dropbox/Vesbach Lab/Conner Mertz/UNM biology/Chapter 1 E1 vs E2/Methods/16S rRNA gene sequencing/Sequencing_workup/ANCOMoutput/ISMERevisions/E1_ile_by_Percent_Protein_df_res_pair_edit.csv")
#change name so target_ids match on both files
names(E1_ile_by_Percent_Protein_df_res_pair)[2] ="taxon" 

#match to taxonomy
Taxonomy_classification <- read.csv("/Users/connermertz/Dropbox (vlab)/Conner Mertz/UNM biology/Chapter 1 E1 vs E2/Methods/16S rRNA gene sequencing/Sequencing_workup/DADA2_output/ALL_taxa.csv") #if this doesn't load be sure to dowload the file from the cloud, then it will work
#change name so target_ids match on both files
names(Taxonomy_classification)[1] ="taxon"

#now we need to get our DF results and join it with our Taxonomy results from DADA2 (combining two data frames)
#we only want to keep the genes that showed up with gapmind, we don't need the other genes that showed up
Part_DF <- Taxonomy_classification %>% 
  inner_join (E1_ile_by_Percent_Protein_df_res_pair, by = "taxon")

write.csv(Part_DF,"/Users/connermertz/vlab Dropbox/Vesbach Lab/Conner Mertz/UNM biology/Chapter 1 E1 vs E2/Methods/16S rRNA gene sequencing/Sequencing_workup/ANCOMoutput/ISMERevisions/E1_ILE_PROTEIN_DF_RES_PAIR_AND_Taxonomy.csv") # went in here and edited file to my liking. That is what I will be using for the figures.  

```



#E1_PROTEIN_in cec
ANCOMBC2 pairwise comparisons
```{r}
output = ancombc2(
  ps_raw_E1_cec,
  assay_name = NULL,
  tax_level = NULL, #keep null if you want at ASV level
  fix_formula= "Exp_Diet",
  rand_formula = NULL,
  p_adj_method = "holm",
  prv_cut = 0.1,
  lib_cut = 0,
  s0_perc = 0.05,
  group = "Exp_Diet", #leave as null if group variable of interest contains only two categories 
  struc_zero = FALSE,
  neg_lb = FALSE,
  alpha = 0.05,
  n_cl = 1,
  verbose = FALSE,
  global = TRUE,
  pairwise = TRUE,
  dunnet = TRUE,
  trend = FALSE,
  iter_control = list(tol = 0.01, max_iter = 20, verbose = FALSE),
  em_control = list(tol = 1e-05, max_iter = 100),
  lme_control = lme4::lmerControl(),
  mdfdr_control = list(fwer_ctrl_method = "holm", B = 100),
  trend_control = list(contrast = NULL, node = NULL, solver = "ECOS", B = 100)
)
```

Save the output
```{r}
#The primary output of the ANCOM-BC2 methodology (res) identifies taxa with differential abundance based on the chosen covariate. The results include: 1) log fold changes, 2) standard errors, 3) test statistics, 4) p-values, 5) adjusted p-values, 6) indicators denoting whether the taxon is differentially abundant (TRUE) or not (FALSE), and 7) indicators denoting whether the taxon passed the sensitivity analysis (TRUE) or not (FALSE).

res_pair = output$res_pair
saveRDS(res_pair, "/Users/connermertz/vlab Dropbox/Vesbach Lab/Conner Mertz/UNM biology/Chapter 1 E1 vs E2/Methods/16S rRNA gene sequencing/Sequencing_workup/ANCOMoutput/ISMERevisions/E1_cec_by_Percent_Protein_res_pair.rds")
E1_cec_by_Percent_Protein_res_pair <- readRDS("/Users/connermertz/vlab Dropbox/Vesbach Lab/Conner Mertz/UNM biology/Chapter 1 E1 vs E2/Methods/16S rRNA gene sequencing/Sequencing_workup/ANCOMoutput/ISMERevisions/E1_cec_by_Percent_Protein_res_pair.rds")
```


##ANCOM-BC2 pairwise directional test
ANCOM-BC2 pairwise directional test aims to determine taxa that are differentially abundant between any pair of two groups across three or more experimental groups, while controlling the mdFDR.


#E1_PROTEIN_in cecum
ANCOMBC2
pairwise comparisons by experiment and PROTEIN
```{r}
#create your data frame for log fold change of pairwise comparisons for dot plot! We will use this to graph dot plots of bargraphs
df_res_pair = E1_cec_by_Percent_Protein_res_pair %>%
  dplyr::filter(diff_Exp_DietE1_2.5P == 1) %>%
  mutate(lfc_E1_2.5P = ifelse(diff_Exp_DietE1_2.5P == 1, 
                              lfc_Exp_DietE1_2.5P, 0)) %>%
  transmute(taxon,
            lfc = lfc_E1_2.5P,
            se = se_Exp_DietE1_2.5P,
            p_val= p_Exp_DietE1_2.5P,
            group = "E1_2.5P - E1_10P") %>%
  
  bind_rows(
    E1_cec_by_Percent_Protein_res_pair %>%
      dplyr::filter(diff_Exp_DietE1_5P == 1) %>%
      mutate(lfc_E1_5P = ifelse(diff_Exp_DietE1_5P == 1, 
                                lfc_Exp_DietE1_5P, 0)) %>%
      transmute(taxon,
                lfc = lfc_E1_5P,
                se = se_Exp_DietE1_5P,
                p_val = p_Exp_DietE1_5P,
                group = "E1_5P - E1_10P")
  ) %>%                 
  bind_rows(
    E1_cec_by_Percent_Protein_res_pair %>%
      dplyr::filter(diff_Exp_DietE1_5P_Exp_DietE1_2.5P == 1) %>%
      mutate(lfc_E1_5P_E1_2.5P = ifelse(diff_Exp_DietE1_5P_Exp_DietE1_2.5P == 1, 
                                        lfc_Exp_DietE1_5P_Exp_DietE1_2.5P, 0)) %>%
      transmute(taxon,
                lfc = lfc_E1_5P_E1_2.5P,
                se = se_Exp_DietE1_5P_Exp_DietE1_2.5P,
                p_val = p_Exp_DietE1_5P_Exp_DietE1_2.5P,
                group = "E1_5P - E1_2.5P")
  )
df_res_pair$group = factor(df_res_pair$group, 
                           levels = c("E1_2.5P - E1_10P",
                                      "E1_5P - E1_10P",
                                      "E1_5P - E1_2.5P"
                           ))


```

Save the df you created
```{r}
write.csv(df_res_pair, "/Users/connermertz/vlab Dropbox/Vesbach Lab/Conner Mertz/UNM biology/Chapter 1 E1 vs E2/Methods/16S rRNA gene sequencing/Sequencing_workup/ANCOMoutput/ISMERevisions/E1_cec_by_Percent_Protein_df_res_pair.csv") #ADD IN OTHER CATEGORIES YOU WANT TO GRAPH 
```

there were 10 ASVs that are differentially abundant in the cecum of mice fed the synthetic diet by percent protein.
E1_cec
E1_2.5P vs E1_10P: ASV35, ASV115, ASV149, ASV224 is differentially abundant in E1_10P; ASV90, ASV160 is differentially abundant in E1_2.5P
E1_5P vs E1_2.5P: ASV20, ASV54, ASV115 is differentially abundant in E1_5P; ASV 199 is differentially 
abundant in E1_2.5P
E1_5P vs E1_10P: no ASVs differentially abundant identified



Save the df you created
```{r}
#read in your EDITED DF csv: 
E1_cec_by_Percent_Protein_df_res_pair <- read.csv("/Users/connermertz/vlab Dropbox/Vesbach Lab/Conner Mertz/UNM biology/Chapter 1 E1 vs E2/Methods/16S rRNA gene sequencing/Sequencing_workup/ANCOMoutput/ISMERevisions/E1_cec_by_Percent_Protein_df_res_pair_edit.csv")
#change name so target_ids match on both files
names(E1_cec_by_Percent_Protein_df_res_pair)[2] ="taxon" 

#match to taxonomy
Taxonomy_classification <- read.csv("/Users/connermertz/Dropbox (vlab)/Conner Mertz/UNM biology/Chapter 1 E1 vs E2/Methods/16S rRNA gene sequencing/Sequencing_workup/DADA2_output/ALL_taxa.csv") #if this doesn't load be sure to dowload the file from the cloud, then it will work
#change name so target_ids match on both files
names(Taxonomy_classification)[1] ="taxon"

#now we need to get our DF results and join it with our Taxonomy results from DADA2 (combining two data frames)
#we only want to keep the genes that showed up with gapmind, we don't need the other genes that showed up
Part_DF <- Taxonomy_classification %>% 
  inner_join (E1_cec_by_Percent_Protein_df_res_pair, by = "taxon")

write.csv(Part_DF,"/Users/connermertz/vlab Dropbox/Vesbach Lab/Conner Mertz/UNM biology/Chapter 1 E1 vs E2/Methods/16S rRNA gene sequencing/Sequencing_workup/ANCOMoutput/ISMERevisions/E1_CEC_PROTEIN_DF_RES_PAIR_AND_Taxonomy.csv") # went in here and edited file to my liking. That is what I will be using for the figures.  

```

Now lets repeat but for E2 tests:

ANCOM
Analysis of Compositions of Microbiomes with Bias Correction (ANCOM-BC) (Lin and Peddada 2020) is a methodology of differential abundance (DA) analysis for microbial absolute abundances. ANCOM-BC estimates the unknown sampling fractions, corrects the bias induced by their differences through a log linear regression model including the estimated sampling fraction as an offset terms, and identifies taxa that are differentially abundant according to the variable of interest. For more details, please refer to the ANCOM-BC paper.
official tutorial: https://bioconductor.org/packages/release/bioc/vignettes/ANCOMBC/inst/doc/ANCOMBC2.html

https://bioconductor.org/packages/devel/bioc/vignettes/ANCOMBC/inst/doc/ANCOMBC.html

a helpful tutorial: https://www.yanh.org/2021/01/01/microbiome-r/#differential-abundance-analysis

ANCOM manual: https://bioconductor.org/packages/release/bioc/manuals/ANCOMBC/man/ANCOMBC.pdf

```{r setup, include=FALSE}
knitr::opts_chunk$set(message = FALSE, warning = FALSE, comment = NA, 
                      fig.width = 6.25, fig.height = 5)
library(ANCOMBC)
library(tidyverse)
library(caret)
library(DT)
options(DT.options = list(
  initComplete = JS("function(settings, json) {",
  "$(this.api().table().header()).css({'background-color': 
  '#000', 'color': '#fff'});","}")))
```

#LOAD OTHER PACKAGES THAT MIGHT BE USEFUL
```{r}
#load packages
# = this package is definitely needed
library(BiocManager)
library(phyloseq) #
library(ggplot2) #
library(randomForest)
library(vegan)
library(caret)
library(gridExtra) #
library(ggpubr) 
library(ggthemes) #
library(vegan) #
library(MASS)
library(cluster)
library(RColorBrewer) #
library(grid)
library(cooccur)
library(igraph)
library(Hmisc)
#library(minerva)
library(parallel)
library(corrplot)
library(tidyverse) #
library(plyr);library(dplyr) #
library(tidyr) #
library(usdm) #
library(decontam) #
library(ecodist) #
library(betapart) #
#library(psych)
#library(ggbiplot)
library(ANCOMBC) #
library(DT) #
library(gplots) #
library(remotes)
library(metagMisc)
library(iNEXT)
library("DESeq2")
packageVersion("DESeq2")
library("reshape2")
library("remotes")
library("phyloseq.extended")
#library(ggvegan) 
#library(ggpubfigs)
#library(agricolae) #this package is not compatable with this version of R, if we really need it we will need to install new version of package
library(RColorBrewer)
library(scales)
library("devtools")

theme_set(theme_bw()) #sets ggplot theme
```


#load metadata
```{r}
Meta <- read.csv("/Users/connermertz/Dropbox (vlab)/Conner Mertz/UNM biology/Chapter 1 E1 vs E2/Methods/16S rRNA gene sequencing/Sequencing_workup/E1_E2_Meta_ALL.csv") #you should have 333 total samples
row.names(Meta) <- Meta$Sample_ID
```

#read in ps objects
```{r}
ps_raw <- readRDS("/Users/connermertz/Dropbox (vlab)/Conner Mertz/UNM biology/Chapter 1 E1 vs E2/Methods/16S rRNA gene sequencing/Sequencing_workup/DADA2_output/clean_ps.rds") #raw but with choloro,mito, low reads (less than 2000), and sample duplicates removed
ps_all <- readRDS("/Users/connermertz/Dropbox (vlab)/Conner Mertz/UNM biology/Chapter 1 E1 vs E2/Methods/16S rRNA gene sequencing/Sequencing_workup/RarefyOutput/ps_rare.rds") #this is my phyloseq object I will be using for the majority of my analysis. It contains all samples from E1 and E2 (duplicates not included) and has been rarefied at lib size of 8000.
sample_data(ps_raw) <- Meta #If you beed to update your meta sheet, this is how to do it without dunning dada2 again to create a whole new ps
sample_data(ps_all) <- Meta #If you beed to update your meta sheet, this is how to do it without dunning dada2 again to create a whole new ps
```


```{r}
#read in un-normalized- unrarefied phyloseq object from rds (ANCOM normalizes for you, and if you give it normalized data it will give bias)
#make relevant subsets

### Make new phyloseq object of just E1, no duplicates
E1_ps_raw <- subset_samples(ps_raw, Experiment=="E1")


#Subset by E1 2.5P,5P, 10P only to compare to E2; no duplicates
ps_raw_E1_forcomparison <- subset_samples(E1_ps_raw, Percent_Protein %in% c("2.5P", "5P", "10P"))
ps_raw_E1_forcomparison <- subset_samples(ps_raw_E1_forcomparison, Location %in% c("Duodenum", "Ileum", "Cecum"))
ps_raw_E1_forcomparison

#Subset by E1 and E2 2.5P,5P, 10P; no duplicates
ps_raw_E1_E2_forcomparison <- subset_samples(ps_raw, Percent_Protein %in% c("2.5P", "5P", "10P"))
ps_raw_E1_E2_forcomparison <- subset_samples(ps_raw_E1_E2_forcomparison, Location %in% c("Duodenum", "Ileum", "Cecum"))
ps_raw_E1_E2_forcomparison


### Make new phyloseq object of just E2, no duplicates
E2_ps_raw <- subset_samples(ps_raw, Experiment=="E2")   


#Subset by E1_E2_Location
ps_raw_E1_E2_duod <- subset_samples(ps_raw_E1_E2_forcomparison, Location %in% c("Duodenum"))
ps_raw_E1_E2_ile <- subset_samples(ps_raw_E1_E2_forcomparison, Location %in% c("Ileum"))
ps_raw_E1_E2_cec <- subset_samples(ps_raw_E1_E2_forcomparison, Location %in% c("Cecum"))

#Subset by Experiment and by Location
ps_raw_E1_duod <- subset_samples(ps_raw_E1_forcomparison, Location %in% c("Duodenum"))
ps_raw_E1_ile <- subset_samples(ps_raw_E1_forcomparison, Location %in% c("Ileum"))
ps_raw_E1_cec <- subset_samples(ps_raw_E1_forcomparison, Location %in% c("Cecum"))
ps_raw_E2_duod <- subset_samples(E2_ps_raw, Location %in% c("Duodenum"))
ps_raw_E2_ile <- subset_samples(E2_ps_raw, Location %in% c("Ileum"))
ps_raw_E2_cec <- subset_samples(E2_ps_raw, Location %in% c("Cecum"))

```

Now lets look at whether we can see %protein differences within each gut section
```{r}
output = ancombc2(
  ps_raw_E2_duod,
  assay_name = NULL,
  tax_level = NULL, #keep null if you want at ASV level
  fix_formula= "Exp_Diet",
  rand_formula = NULL,
  p_adj_method = "holm",
  prv_cut = 0.1,
  lib_cut = 0,
  s0_perc = 0.05,
  group = "Exp_Diet", #leave as null if group variable of interest contains only two categories 
  struc_zero = FALSE,
  neg_lb = FALSE,
  alpha = 0.05,
  n_cl = 1,
  verbose = FALSE,
  global = TRUE,
  pairwise = TRUE,
  dunnet = TRUE,
  trend = FALSE,
  iter_control = list(tol = 0.01, max_iter = 20, verbose = FALSE),
  em_control = list(tol = 1e-05, max_iter = 100),
  lme_control = lme4::lmerControl(),
  mdfdr_control = list(fwer_ctrl_method = "holm", B = 100),
  trend_control = list(contrast = NULL, node = NULL, solver = "ECOS", B = 100)
)
```

Save the output
```{r}
#The primary output of the ANCOM-BC2 methodology (res) identifies taxa with differential abundance based on the chosen covariate. The results include: 1) log fold changes, 2) standard errors, 3) test statistics, 4) p-values, 5) adjusted p-values, 6) indicators denoting whether the taxon is differentially abundant (TRUE) or not (FALSE), and 7) indicators denoting whether the taxon passed the sensitivity analysis (TRUE) or not (FALSE).

res_pair = output$res_pair
saveRDS(res_pair, "/Users/connermertz/vlab Dropbox/Vesbach Lab/Conner Mertz/UNM biology/Chapter 1 E1 vs E2/Methods/16S rRNA gene sequencing/Sequencing_workup/ANCOMoutput/ISMERevisions/E2_duod_by_Percent_Protein_res_pair.rds")
E2_duod_by_Percent_Protein_res_pair <- readRDS("/Users/connermertz/vlab Dropbox/Vesbach Lab/Conner Mertz/UNM biology/Chapter 1 E1 vs E2/Methods/16S rRNA gene sequencing/Sequencing_workup/ANCOMoutput/ISMERevisions/E2_duod_by_Percent_Protein_res_pair.rds")
```


##ANCOM-BC2 pairwise directional test
ANCOM-BC2 pairwise directional test aims to determine taxa that are differentially abundant between any pair of two groups across three or more experimental groups, while controlling the mdFDR.

```{r}
#save as csv 
write.csv(E2_duod_by_Percent_Protein_res_pair, "/Users/connermertz/vlab Dropbox/Vesbach Lab/Conner Mertz/UNM biology/Chapter 1 E1 vs E2/Methods/16S rRNA gene sequencing/Sequencing_workup/ANCOMoutput/ISMERevisions/E2_duod_by_Percent_Protein_res_pair_edit.csv")

```





#E2_PROTEIN_in duod
ANCOMBC2
pairwise comparisons by experiment and PROTEIN
```{r}
#create your data frame for log fold change of pairwise comparisons for dot plot! We will use this to graph dot plots of bargraphs
df_res_pair = E2_duod_by_Percent_Protein_res_pair %>%
  dplyr::filter(diff_Exp_DietE2_2.5P == 1) %>%
  mutate(lfc_E2_2.5P = ifelse(diff_Exp_DietE2_2.5P == 1, 
                              lfc_Exp_DietE2_2.5P, 0)) %>%
  transmute(taxon,
            lfc = lfc_E2_2.5P,
            se = se_Exp_DietE2_2.5P,
            p_val= p_Exp_DietE2_2.5P,
            group = "E2_2.5P - E2_10P") %>%
  
  bind_rows(
    E2_duod_by_Percent_Protein_res_pair %>%
      dplyr::filter(diff_Exp_DietE2_5P == 1) %>%
      mutate(lfc_E2_5P = ifelse(diff_Exp_DietE2_5P == 1, 
                                lfc_Exp_DietE2_5P, 0)) %>%
      transmute(taxon,
                lfc = lfc_E2_5P,
                se = se_Exp_DietE2_5P,
                p_val = p_Exp_DietE2_5P,
                group = "E2_5P - E2_10P")
  ) %>%                 
  bind_rows(
    E2_duod_by_Percent_Protein_res_pair %>%
      dplyr::filter(diff_Exp_DietE2_5P_Exp_DietE2_2.5P == 1) %>%
      mutate(lfc_E2_5P_E2_2.5P = ifelse(diff_Exp_DietE2_5P_Exp_DietE2_2.5P == 1, 
                                        lfc_Exp_DietE2_5P_Exp_DietE2_2.5P, 0)) %>%
      transmute(taxon,
                lfc = lfc_E2_5P_E2_2.5P,
                se = se_Exp_DietE2_5P_Exp_DietE2_2.5P,
                p_val = p_Exp_DietE2_5P_Exp_DietE2_2.5P,
                group = "E2_5P - E2_2.5P")
  )
df_res_pair$group = factor(df_res_pair$group, 
                           levels = c("E2_2.5P - E2_10P",
                                      "E2_5P - E2_10P",
                                      "E2_5P - E2_2.5P"
                           ))


```

Save the df you created
```{r}
write.csv(df_res_pair, "/Users/connermertz/vlab Dropbox/Vesbach Lab/Conner Mertz/UNM biology/Chapter 1 E1 vs E2/Methods/16S rRNA gene sequencing/Sequencing_workup/ANCOMoutput/ISMERevisions/E2_duod_by_Percent_Protein_df_res_pair.csv") #ADD IN OTHER CATEGORIES YOU WANT TO GRAPH 
```

there was 1 ASV that was differentially abundant in the duodenum of mice fed the seminatural diet by percent protein.
E2_duod
E2_2.5P vs E2_10P: ASV5 is differentially abundant in E2_2.5P
E2_5P vs E2_2.5P: no ASVs differentially abundant identified
E2_5P vs E2_10P: no ASVs differentially abundant identified

merge with DADA2 taxonomy assignment
```{r}
#read in your EDITED DF csv: 
E2_duod_by_Percent_Protein_df_res_pair <- read.csv("/Users/connermertz/vlab Dropbox/Vesbach Lab/Conner Mertz/UNM biology/Chapter 1 E1 vs E2/Methods/16S rRNA gene sequencing/Sequencing_workup/ANCOMoutput/ISMERevisions/E2_duod_by_Percent_Protein_df_res_pair_edit.csv")
#change name so target_ids match on both fduods
names(E2_duod_by_Percent_Protein_df_res_pair)[2] ="taxon" 

#match to taxonomy
Taxonomy_classification <- read.csv("/Users/connermertz/Dropbox (vlab)/Conner Mertz/UNM biology/Chapter 1 E1 vs E2/Methods/16S rRNA gene sequencing/Sequencing_workup/DADA2_output/ALL_taxa.csv") #if this doesn't load be sure to dowload the file from the cloud, then it will work
#change name so target_ids match on both files
names(Taxonomy_classification)[1] ="taxon"

#now we need to get our DF results and join it with our Taxonomy results from DADA2 (combining two data frames)
#we only want to keep the genes that showed up with gapmind, we don't need the other genes that showed up
Part_DF <- Taxonomy_classification %>% 
  inner_join (E2_duod_by_Percent_Protein_df_res_pair, by = "taxon")

write.csv(Part_DF,"/Users/connermertz/vlab Dropbox/Vesbach Lab/Conner Mertz/UNM biology/Chapter 1 E1 vs E2/Methods/16S rRNA gene sequencing/Sequencing_workup/ANCOMoutput/ISMERevisions/E2_DUOD_PROTEIN_DF_RES_PAIR_AND_Taxonomy.csv") # went in here and edited file to my liking. That is what I will be using for the figures.  

``` 



Now lets look at whether we can see %protein differences in the Ileum

```{r}
output = ancombc2(
  ps_raw_E2_ile,
  assay_name = NULL,
  tax_level = NULL, #keep null if you want at ASV level
  fix_formula= "Exp_Diet",
  rand_formula = NULL,
  p_adj_method = "holm",
  prv_cut = 0.1,
  lib_cut = 0,
  s0_perc = 0.05,
  group = "Exp_Diet", #leave as null if group variable of interest contains only two categories 
  struc_zero = FALSE,
  neg_lb = FALSE,
  alpha = 0.05,
  n_cl = 1,
  verbose = FALSE,
  global = TRUE,
  pairwise = TRUE,
  dunnet = TRUE,
  trend = FALSE,
  iter_control = list(tol = 0.01, max_iter = 20, verbose = FALSE),
  em_control = list(tol = 1e-05, max_iter = 100),
  lme_control = lme4::lmerControl(),
  mdfdr_control = list(fwer_ctrl_method = "holm", B = 100),
  trend_control = list(contrast = NULL, node = NULL, solver = "ECOS", B = 100)
)
```

Save the output
```{r}
#The primary output of the ANCOM-BC2 methodology (res) identifies taxa with differential abundance based on the chosen covariate. The results include: 1) log fold changes, 2) standard errors, 3) test statistics, 4) p-values, 5) adjusted p-values, 6) indicators denoting whether the taxon is differentially abundant (TRUE) or not (FALSE), and 7) indicators denoting whether the taxon passed the sensitivity analysis (TRUE) or not (FALSE).

res_pair = output$res_pair
saveRDS(res_pair, "/Users/connermertz/vlab Dropbox/Vesbach Lab/Conner Mertz/UNM biology/Chapter 1 E1 vs E2/Methods/16S rRNA gene sequencing/Sequencing_workup/ANCOMoutput/ISMERevisions/E2_ile_by_Percent_Protein_res_pair.rds")
E2_ile_by_Percent_Protein_res_pair <- readRDS("/Users/connermertz/vlab Dropbox/Vesbach Lab/Conner Mertz/UNM biology/Chapter 1 E1 vs E2/Methods/16S rRNA gene sequencing/Sequencing_workup/ANCOMoutput/ISMERevisions/E2_ile_by_Percent_Protein_res_pair.rds")
```


##ANCOM-BC2 pairwise directional test
ANCOM-BC2 pairwise directional test aims to determine taxa that are differentially abundant between any pair of two groups across three or more experimental groups, while controlling the mdFDR.


#E2_PROTEIN_in ile
ANCOMBC2 pairwise comparisons 
```{r}
#create your data frame for log fold change of pairwise comparisons for dot plot! We will use this to graph dot plots of bargraphs
df_res_pair = E2_ile_by_Percent_Protein_res_pair %>%
  dplyr::filter(diff_Exp_DietE2_2.5P == 1) %>%
  mutate(lfc_E2_2.5P = ifelse(diff_Exp_DietE2_2.5P == 1, 
                              lfc_Exp_DietE2_2.5P, 0)) %>%
  transmute(taxon,
            lfc = lfc_E2_2.5P,
            se = se_Exp_DietE2_2.5P,
            p_val= p_Exp_DietE2_2.5P,
            group = "E2_2.5P - E2_10P") %>%
  
  bind_rows(
    E2_ile_by_Percent_Protein_res_pair %>%
      dplyr::filter(diff_Exp_DietE2_5P == 1) %>%
      mutate(lfc_E2_5P = ifelse(diff_Exp_DietE2_5P == 1, 
                                lfc_Exp_DietE2_5P, 0)) %>%
      transmute(taxon,
                lfc = lfc_E2_5P,
                se = se_Exp_DietE2_5P,
                p_val = p_Exp_DietE2_5P,
                group = "E2_5P - E2_10P")
  ) %>%                 
  bind_rows(
    E2_ile_by_Percent_Protein_res_pair %>%
      dplyr::filter(diff_Exp_DietE2_5P_Exp_DietE2_2.5P == 1) %>%
      mutate(lfc_E2_5P_E2_2.5P = ifelse(diff_Exp_DietE2_5P_Exp_DietE2_2.5P == 1, 
                                        lfc_Exp_DietE2_5P_Exp_DietE2_2.5P, 0)) %>%
      transmute(taxon,
                lfc = lfc_E2_5P_E2_2.5P,
                se = se_Exp_DietE2_5P_Exp_DietE2_2.5P,
                p_val = p_Exp_DietE2_5P_Exp_DietE2_2.5P,
                group = "E2_5P - E2_2.5P")
  )
df_res_pair$group = factor(df_res_pair$group, 
                           levels = c("E2_2.5P - E2_10P",
                                      "E2_5P - E2_10P",
                                      "E2_5P - E2_2.5P"
                           ))


```

Save the df you created
```{r}
write.csv(df_res_pair, "/Users/connermertz/vlab Dropbox/Vesbach Lab/Conner Mertz/UNM biology/Chapter 1 E1 vs E2/Methods/16S rRNA gene sequencing/Sequencing_workup/ANCOMoutput/ISMERevisions/E2_ile_by_Percent_Protein_df_res_pair.csv") #ADD IN OTHER CATEGORIES YOU WANT TO GRAPH 
```

there were 31 ASVs that are differentially abundant in the ileum of mice fed the seminatural diet by percent protein.
E2_ile
E2_2.5P vs E2_10P: ASV5, ASV6, ASV9, ASV11, ASV14, ASV16, ASV17, ASV25, ASV27, ASV30, ASV37, ASV43, ASV64, ASV76, ASV189, ASV198, ASV265, ASV267 are differentially abundant in E2_2.5P; and ASV96, ASV153, ASV112 are differentially abundant in E2_10P
E2_5P vs E2_2.5P: ASV11, ASV25, ASV39, ASV86 are differentially abundant in E2_5P; ASV17
ASV64, ASV66, ASV189, ASV267 is differentially abundant in E2_5P
E2_5P vs E2_10P: no ASVs differentially abundant identified

merge with DADA2 taxonomy assignment
```{r}
#read in your EDITED DF csv: 
E2_ile_by_Percent_Protein_df_res_pair <- read.csv("/Users/connermertz/vlab Dropbox/Vesbach Lab/Conner Mertz/UNM biology/Chapter 1 E1 vs E2/Methods/16S rRNA gene sequencing/Sequencing_workup/ANCOMoutput/ISMERevisions/E2_ile_by_Percent_Protein_df_res_pair_edit.csv")
#change name so target_ids match on both files
names(E2_ile_by_Percent_Protein_df_res_pair)[2] ="taxon" 

#match to taxonomy
Taxonomy_classification <- read.csv("/Users/connermertz/Dropbox (vlab)/Conner Mertz/UNM biology/Chapter 1 E1 vs E2/Methods/16S rRNA gene sequencing/Sequencing_workup/DADA2_output/ALL_taxa.csv") #if this doesn't load be sure to dowload the file from the cloud, then it will work
#change name so target_ids match on both files
names(Taxonomy_classification)[1] ="taxon"

#now we need to get our DF results and join it with our Taxonomy results from DADA2 (combining two data frames)
#we only want to keep the genes that showed up with gapmind, we don't need the other genes that showed up
Part_DF <- Taxonomy_classification %>% 
  inner_join (E2_ile_by_Percent_Protein_df_res_pair, by = "taxon")

write.csv(Part_DF,"/Users/connermertz/vlab Dropbox/Vesbach Lab/Conner Mertz/UNM biology/Chapter 1 E1 vs E2/Methods/16S rRNA gene sequencing/Sequencing_workup/ANCOMoutput/ISMERevisions/E2_ILE_PROTEIN_DF_RES_PAIR_AND_Taxonomy.csv") # went in here and edited file to my liking. That is what I will be using for the figures.  

```



#E2_PROTEIN_in cec
ANCOMBC2 pairwise comparisons
```{r}
output = ancombc2(
  ps_raw_E2_cec,
  assay_name = NULL,
  tax_level = NULL, #keep null if you want at ASV level
  fix_formula= "Exp_Diet",
  rand_formula = NULL,
  p_adj_method = "holm",
  prv_cut = 0.1,
  lib_cut = 0,
  s0_perc = 0.05,
  group = "Exp_Diet", #leave as null if group variable of interest contains only two categories 
  struc_zero = FALSE,
  neg_lb = FALSE,
  alpha = 0.05,
  n_cl = 1,
  verbose = FALSE,
  global = TRUE,
  pairwise = TRUE,
  dunnet = TRUE,
  trend = FALSE,
  iter_control = list(tol = 0.01, max_iter = 20, verbose = FALSE),
  em_control = list(tol = 1e-05, max_iter = 100),
  lme_control = lme4::lmerControl(),
  mdfdr_control = list(fwer_ctrl_method = "holm", B = 100),
  trend_control = list(contrast = NULL, node = NULL, solver = "ECOS", B = 100)
)
```

Save the output
```{r}
#The primary output of the ANCOM-BC2 methodology (res) identifies taxa with differential abundance based on the chosen covariate. The results include: 1) log fold changes, 2) standard errors, 3) test statistics, 4) p-values, 5) adjusted p-values, 6) indicators denoting whether the taxon is differentially abundant (TRUE) or not (FALSE), and 7) indicators denoting whether the taxon passed the sensitivity analysis (TRUE) or not (FALSE).

res_pair = output$res_pair
saveRDS(res_pair, "/Users/connermertz/vlab Dropbox/Vesbach Lab/Conner Mertz/UNM biology/Chapter 1 E1 vs E2/Methods/16S rRNA gene sequencing/Sequencing_workup/ANCOMoutput/ISMERevisions/E2_cec_by_Percent_Protein_res_pair.rds")
E2_cec_by_Percent_Protein_res_pair <- readRDS("/Users/connermertz/vlab Dropbox/Vesbach Lab/Conner Mertz/UNM biology/Chapter 1 E1 vs E2/Methods/16S rRNA gene sequencing/Sequencing_workup/ANCOMoutput/ISMERevisions/E2_cec_by_Percent_Protein_res_pair.rds")
```


##ANCOM-BC2 pairwise directional test
ANCOM-BC2 pairwise directional test aims to determine taxa that are differentially abundant between any pair of two groups across three or more experimental groups, while controlling the mdFDR.


#E2_PROTEIN_in cecum
ANCOMBC2
pairwise comparisons by experiment and PROTEIN
```{r}
#create your data frame for log fold change of pairwise comparisons for dot plot! We will use this to graph dot plots of bargraphs
df_res_pair = E2_cec_by_Percent_Protein_res_pair %>%
  dplyr::filter(diff_Exp_DietE2_2.5P == 1) %>%
  mutate(lfc_E2_2.5P = ifelse(diff_Exp_DietE2_2.5P == 1, 
                              lfc_Exp_DietE2_2.5P, 0)) %>%
  transmute(taxon,
            lfc = lfc_E2_2.5P,
            se = se_Exp_DietE2_2.5P,
            p_val= p_Exp_DietE2_2.5P,
            group = "E2_2.5P - E2_10P") %>%
  
  bind_rows(
    E2_cec_by_Percent_Protein_res_pair %>%
      dplyr::filter(diff_Exp_DietE2_5P == 1) %>%
      mutate(lfc_E2_5P = ifelse(diff_Exp_DietE2_5P == 1, 
                                lfc_Exp_DietE2_5P, 0)) %>%
      transmute(taxon,
                lfc = lfc_E2_5P,
                se = se_Exp_DietE2_5P,
                p_val = p_Exp_DietE2_5P,
                group = "E2_5P - E2_10P")
  ) %>%                 
  bind_rows(
    E2_cec_by_Percent_Protein_res_pair %>%
      dplyr::filter(diff_Exp_DietE2_5P_Exp_DietE2_2.5P == 1) %>%
      mutate(lfc_E2_5P_E2_2.5P = ifelse(diff_Exp_DietE2_5P_Exp_DietE2_2.5P == 1, 
                                        lfc_Exp_DietE2_5P_Exp_DietE2_2.5P, 0)) %>%
      transmute(taxon,
                lfc = lfc_E2_5P_E2_2.5P,
                se = se_Exp_DietE2_5P_Exp_DietE2_2.5P,
                p_val = p_Exp_DietE2_5P_Exp_DietE2_2.5P,
                group = "E2_5P - E2_2.5P")
  )
df_res_pair$group = factor(df_res_pair$group, 
                           levels = c("E2_2.5P - E2_10P",
                                      "E2_5P - E2_10P",
                                      "E2_5P - E2_2.5P"
                           ))


```

Save the df you created
```{r}
write.csv(df_res_pair, "/Users/connermertz/vlab Dropbox/Vesbach Lab/Conner Mertz/UNM biology/Chapter 1 E1 vs E2/Methods/16S rRNA gene sequencing/Sequencing_workup/ANCOMoutput/ISMERevisions/E2_cec_by_Percent_Protein_df_res_pair.csv") #ADD IN OTHER CATEGORIES YOU WANT TO GRAPH 
```

there were 8 ASVs that are differentially abundant in the cecum of mice fed the semi-natural diet by percent protein.
E2_cec

E2_2.5P vs E2_10P: ASV193, ASV294, ASV521, ASV828 is differentially abundant in E2_10P; ASV5, ASV17 is differentially abundant in E2_2.5P
E2_5P vs E2_2.5P: ASV18 is differentially abundant in E2_5P
E2_5P vs E2_10P: ASV5 is differentially abundant in E2_5P



Save the df you created
```{r}
#read in your EDITED DF csv: 
E2_cec_by_Percent_Protein_df_res_pair <- read.csv("/Users/connermertz/vlab Dropbox/Vesbach Lab/Conner Mertz/UNM biology/Chapter 1 E1 vs E2/Methods/16S rRNA gene sequencing/Sequencing_workup/ANCOMoutput/ISMERevisions/E2_cec_by_Percent_Protein_df_res_pair_edit.csv")
#change name so target_ids match on both files
names(E2_cec_by_Percent_Protein_df_res_pair)[2] ="taxon" 

#match to taxonomy
Taxonomy_classification <- read.csv("/Users/connermertz/Dropbox (vlab)/Conner Mertz/UNM biology/Chapter 1 E1 vs E2/Methods/16S rRNA gene sequencing/Sequencing_workup/DADA2_output/ALL_taxa.csv") #if this doesn't load be sure to dowload the file from the cloud, then it will work
#change name so target_ids match on both files
names(Taxonomy_classification)[1] ="taxon"

#now we need to get our DF results and join it with our Taxonomy results from DADA2 (combining two data frames)
#we only want to keep the genes that showed up with gapmind, we don't need the other genes that showed up
Part_DF <- Taxonomy_classification %>% 
  inner_join (E2_cec_by_Percent_Protein_df_res_pair, by = "taxon")

write.csv(Part_DF,"/Users/connermertz/vlab Dropbox/Vesbach Lab/Conner Mertz/UNM biology/Chapter 1 E1 vs E2/Methods/16S rRNA gene sequencing/Sequencing_workup/ANCOMoutput/ISMERevisions/E2_CEC_PROTEIN_DF_RES_PAIR_AND_Taxonomy.csv") # went in here and edited file to my liking. That is what I will be using for the figures.  

```



Yay! you did it, now lets make some figures to visualize our results:

lets see if we can't join E1 and E2 Protein results together!
DUOD
```{r}
E1vsE2_DUOD_Protein_df <- read.csv("/Users/connermertz/vlab Dropbox/Vesbach Lab/Conner Mertz/UNM biology/Chapter 1 E1 vs E2/Methods/16S rRNA gene sequencing/Sequencing_workup/ANCOMoutput/ISMERevisions/combined_DF_figs/edit_E1_E2_DUOD_PROTEIN_DF.csv") %>%
  filter(group ==  "E1_2.5P - E1_5P" |
           group == "E1_5P - E1_2.5P" |
           group == "E1_2.5P - E1_10P" |
           group == "E1_10P - E1_2.5P" |
           group == "E1_5P - E1_10P" |
           group == "E1_10P - E1_5P" |
           group ==  "E2_2.5P - E2_5P" |
           group == "E2_5P - E2_2.5P" |
           group == "E2_2.5P - E2_10P" |
           group == "E2_10P - E2_2.5P" |
           group == "E2_5P - E2_10P" |
           group == "E2_10P - E2_5P")%>%
  mutate(Differentially_Abundant_Group = factor(Differentially_Abundant_Group, levels = c("E1_2.5P","E1_5P", "E1_10P", "E2_2.5P","E2_5P", "E2_10P"))) %>%
  mutate(Differentially_Abundant_Protein = factor(Differentially_Abundant_Protein, levels = c("2.5P","5P", "10P"))) %>%
  mutate(taxon = factor(taxon, levels = c("ASV5")))

```

Set up colors
```{r}
#first you want to factor your items. These are the names of the Location in the order I want them to be every time. 
E1vsE2_DUOD_Protein_df$Differentially_Abundant_Protein <-factor(E1vsE2_DUOD_Protein_df$Differentially_Abundant_Protein, levels = c("2.5P","5P", "10P"))

#now create an object that is a list of colors in a certain order.The colors I chose are "color-blind friendly":
Protein_Scale <- c("#cb146e", "#f7ce45","#1983fc")
#scales::show_col(DietScale)

#now we want to merge the two. We are taking the order from the locations and telling it that it needs to match the order from the cbPalette.
names(Protein_Scale) <- levels(E1vsE2_DUOD_Protein_df$Differentially_Protein)
#Have you ever used the scale_colour_manual command? Now we can combine that command and all of the stuff we want in it into a new object, LocationColors. At the end of your ggplot code, just add a +DietColor (or whatever you decide to name it), and it should pull the colors you made! 
ProteinColors.5<- scale_colour_manual(name="Protein", values= Protein_Scale)
```

DUOD by percent protein
```{r}
fig_trend = E1vsE2_DUOD_Protein_df %>% 
  ggplot(aes(x = lfc, y = taxon, color = Differentially_Abundant_Protein, shape = Differentially_Abundant_Experiment)) + 
  geom_point(size = 4, position = position_dodge()) +
  geom_errorbar(
    aes(xmin = lfc - se, xmax = lfc + se), 
    width = .2) +
  geom_vline(aes(xintercept = 0), linetype = "dashed")+
  facet_wrap(~PercentProteinvsPercentProtein, strip.position="top") +
  labs(x = "Log Fold Change", y = "ASV", title = "Differentially Abundant Taxa_DUOD_") +
  ProteinColors.5 +
  scale_x_continuous(limits=c(-7, 7)) +
  theme_bw() +
  theme(
    plot.title = element_text(hjust = 0.5),
    title =element_text(size=18, face='bold'),
    axis.text.x = element_text(color="black", size=12, angle = 90),
    axis.text.y = element_text(color="black", size=12),
    strip.text = element_text(size=6),
    strip.text.y = element_text(color="black", size=12, angle=90),
    legend.title = element_text(color="black", size=14),
    legend.text = element_text(color="black", size=12),
    legend.key.size = unit(1, 'cm'),
    legend.position = "bottom",
    strip.background = element_blank(), #to remove facet wrap title
    axis.title.x = element_text(color="black", size=12),
    axis.title.y = element_text(color="black", size=12, angle=90),
    panel.grid.major = element_blank(), # switch off major gridlines
    panel.grid.minor = element_blank()# switch off minor gridlines
  )
#CREATE A PDF AND SAVES IT TO THE RIGHT DIRECTORY/FOLDER. 
pdf("/Users/connermertz/vlab Dropbox/Vesbach Lab/Conner Mertz/UNM biology/Chapter 1 E1 vs E2/Methods/16S rRNA gene sequencing/Sequencing_workup/ANCOMoutput/ISMERevisions/combined_DF_figs/Plot/E1vsE2_DUOD_Protein.pdf", height = 6, width = 10 )
fig_trend
dev.off()
```






Cec
```{r}
E1vsE2_CEC_Protein_df <- read.csv("/Users/connermertz/vlab Dropbox/Vesbach Lab/Conner Mertz/UNM biology/Chapter 1 E1 vs E2/Methods/16S rRNA gene sequencing/Sequencing_workup/ANCOMoutput/ISMERevisions/combined_DF_figs/edit_E1_E2_CECUM_PROTEIN_DF.csv") %>%
  filter(group ==  "E1_2.5P - E1_5P" |
           group == "E1_5P - E1_2.5P" |
           group == "E1_2.5P - E1_10P" |
           group == "E1_10P - E1_2.5P" |
           group == "E1_5P - E1_10P" |
           group == "E1_10P - E1_5P" |
         group ==  "E2_2.5P - E2_5P" |
           group == "E2_5P - E2_2.5P" |
           group == "E2_2.5P - E2_10P" |
           group == "E2_10P - E2_2.5P" |
           group == "E2_5P - E2_10P" |
           group == "E2_10P - E2_5P")%>%
 mutate(Differentially_Abundant_Group = factor(Differentially_Abundant_Group, levels = c("E1_2.5P","E1_5P", "E1_10P", "E2_2.5P","E2_5P", "E2_10P"))) %>%
 mutate(Differentially_Abundant_Protein = factor(Differentially_Abundant_Protein, levels = c("2.5P","5P", "10P"))) %>%
 mutate(taxon = factor(taxon, levels = c("ASV828",
"ASV115",
"ASV20",
"ASV35",
"ASV521",
"ASV54",
"ASV193",
"ASV90",
"ASV294",
"ASV17",
"ASV5",
"ASV199",
"ASV224",
"ASV160",
"ASV18",
"ASV149")))

```

Set up colors
```{r}
#first you want to factor your items. These are the names of the Location in the order I want them to be every time. 
E1vsE2_CEC_Protein_df$Differentially_Abundant_Protein <-factor(E1vsE2_CEC_Protein_df$Differentially_Abundant_Protein, levels = c("2.5P","5P", "10P"))

#now create an object that is a list of colors in a certain order.The colors I chose are "color-blind friendly":
Protein_Scale <- c("#cb146e", "#f7ce45","#1983fc")
#scales::show_col(DietScale)

#now we want to merge the two. We are taking the order from the locations and telling it that it needs to match the order from the cbPalette.
names(Protein_Scale) <- levels(E1vsE2_CEC_Protein_df$Differentially_Protein)
#Have you ever used the scale_colour_manual command? Now we can combine that command and all of the stuff we want in it into a new object, LocationColors. At the end of your ggplot code, just add a +DietColor (or whatever you decide to name it), and it should pull the colors you made! 
ProteinColors.3<- scale_colour_manual(name="Protein", values= Protein_Scale)
```

CEC by percent protein
```{r}
fig_trend = E1vsE2_CEC_Protein_df %>% 
  ggplot(aes(x = lfc, y = taxon, color = Differentially_Abundant_Protein, shape = Differentially_Abundant_Experiment)) + 
  geom_point(size = 4, position = position_dodge()) +
  geom_errorbar(
    aes(xmin = lfc - se, xmax = lfc + se), 
    width = .2) +
  geom_vline(aes(xintercept = 0), linetype = "dashed")+
  facet_wrap(~PercentProteinvsPercentProtein, strip.position="top") +
  labs(x = "Log Fold Change", y = "ASV", title = "Differentially Abundant Taxa_CEC_") +
  ProteinColors.3 +
  scale_x_continuous(limits=c(-7, 7)) +
  theme_bw() +
  theme(
    plot.title = element_text(hjust = 0.5),
    title =element_text(size=18, face='bold'),
    axis.text.x = element_text(color="black", size=12, angle = 90),
    axis.text.y = element_text(color="black", size=12),
    strip.text = element_text(size=6),
    strip.text.y = element_text(color="black", size=12, angle=90),
    legend.title = element_text(color="black", size=14),
    legend.text = element_text(color="black", size=12),
    legend.key.size = unit(1, 'cm'),
    legend.position = "bottom",
     strip.background = element_blank(), #to remove facet wrap title
    axis.title.x = element_text(color="black", size=12),
    axis.title.y = element_text(color="black", size=12, angle=90),
    panel.grid.major = element_blank(), # switch off major gridlines
    panel.grid.minor = element_blank()# switch off minor gridlines
  )
#CREATE A PDF AND SAVES IT TO THE RIGHT DIRECTORY/FOLDER. 
pdf("/Users/connermertz/vlab Dropbox/Vesbach Lab/Conner Mertz/UNM biology/Chapter 1 E1 vs E2/Methods/16S rRNA gene sequencing/Sequencing_workup/ANCOMoutput/ISMERevisions/combined_DF_figs/Plot/E1vsE2_CEC_Protein.pdf", height = 6, width = 10)
fig_trend
dev.off()
```





ILE
```{r}
E1vsE2_ILE_Protein_df <- read.csv("/Users/connermertz/vlab Dropbox/Vesbach Lab/Conner Mertz/UNM biology/Chapter 1 E1 vs E2/Methods/16S rRNA gene sequencing/Sequencing_workup/ANCOMoutput/ISMERevisions/combined_DF_figs/edit_E1_E2_ILE_PROTEIN_DF.csv") %>%
  filter(group ==  "E1_2.5P - E1_5P" |
           group == "E1_5P - E1_2.5P" |
           group == "E1_2.5P - E1_10P" |
           group == "E1_10P - E1_2.5P" |
           group == "E1_5P - E1_10P" |
           group == "E1_10P - E1_5P" |
           group ==  "E2_2.5P - E2_5P" |
           group == "E2_5P - E2_2.5P" |
           group == "E2_2.5P - E2_10P" |
           group == "E2_10P - E2_2.5P" |
           group == "E2_5P - E2_10P" |
           group == "E2_10P - E2_5P")%>%
  mutate(Differentially_Abundant_Group = factor(Differentially_Abundant_Group, levels = c("E1_2.5P","E1_5P", "E1_10P", "E2_2.5P","E2_5P", "E2_10P"))) %>%
  mutate(Differentially_Abundant_Protein = factor(Differentially_Abundant_Protein, levels = c("2.5P","5P", "10P"))) %>%
  mutate(taxon = factor(taxon, levels = c("ASV25",
"ASV267",
"ASV66",
"ASV80",
"ASV96",
"ASV14",
"ASV112",
"ASV265",
"ASV11",
"ASV43",
"ASV6",
"ASV189",
"ASV39",
"ASV27",
"ASV17",
"ASV5",
"ASV64",
"ASV153",
"ASV9",
"ASV30",
"ASV16",
"ASV37",
"ASV76",
"ASV86",
"ASV168",
"ASV198")))

```

Set up colors
```{r}
#first you want to factor your items. These are the names of the Location in the order I want them to be every time. 
E1vsE2_ILE_Protein_df$Differentially_Abundant_Protein <-factor(E1vsE2_ILE_Protein_df$Differentially_Abundant_Protein, levels = c("2.5P","5P", "10P"))

#now create an object that is a list of colors in a certain order.The colors I chose are "color-blind friendly":
Protein_Scale <- c("#cb146e", "#f7ce45","#1983fc")
#scales::show_col(DietScale)

#now we want to merge the two. We are taking the order from the locations and telling it that it needs to match the order from the cbPalette.
names(Protein_Scale) <- levels(E1vsE2_ILE_Protein_df$Differentially_Protein)
#Have you ever used the scale_colour_manual command? Now we can combine that command and all of the stuff we want in it into a new object, LocationColors. At the end of your ggplot code, just add a +DietColor (or whatever you decide to name it), and it should pull the colors you made! 
ProteinColors.4<- scale_colour_manual(name="Protein", values= Protein_Scale)
```

ILE by percent protein
```{r}
fig_trend = E1vsE2_ILE_Protein_df %>% 
  ggplot(aes(x = lfc, y = taxon, color = Differentially_Abundant_Protein, shape = Differentially_Abundant_Experiment)) + 
  geom_point(size = 4, position = position_dodge()) +
  geom_errorbar(
    aes(xmin = lfc - se, xmax = lfc + se), 
    width = .2) +
  geom_vline(aes(xintercept = 0), linetype = "dashed")+
  facet_wrap(~PercentProteinvsPercentProtein, strip.position="top") +
  labs(x = "Log Fold Change", y = "ASV", title = "Differentially Abundant Taxa_ILE_") +
  ProteinColors.4 +
  scale_x_continuous(limits=c(-7, 7)) +
  theme_bw() +
  theme(
    plot.title = element_text(hjust = 0.5),
    title =element_text(size=18, face='bold'),
    axis.text.x = element_text(color="black", size=12, angle = 90),
    axis.text.y = element_text(color="black", size=12),
    strip.text = element_text(size=6),
    strip.text.y = element_text(color="black", size=12, angle=90),
    legend.title = element_text(color="black", size=14),
    legend.text = element_text(color="black", size=12),
    legend.key.size = unit(1, 'cm'),
    legend.position = "bottom",
    strip.background = element_blank(), #to remove facet wrap title
    axis.title.x = element_text(color="black", size=12),
    axis.title.y = element_text(color="black", size=12, angle=90),
    panel.grid.major = element_blank(), # switch off major gridlines
    panel.grid.minor = element_blank()# switch off minor gridlines
  )
#CREATE A PDF AND SAVES IT TO THE RIGHT DIRECTORY/FOLDER. 
pdf("/Users/connermertz/vlab Dropbox/Vesbach Lab/Conner Mertz/UNM biology/Chapter 1 E1 vs E2/Methods/16S rRNA gene sequencing/Sequencing_workup/ANCOMoutput/ISMERevisions/combined_DF_figs/Plot/E1vsE2_ILE_Protein.pdf", height = 6, width = 10)
fig_trend
dev.off()
```






